---
title: "AMR"
output: html_document
---

```{r}
library(readr)
library(phyloseq)
library(dplyr)
library(tidyr)
library(ggplot2)
library(phyloseq)
library(tidyverse)
library(microbiome)
library(readr)
library("ggpubr")
library("tidyverse")
library(vegan)
```

```{r}
AMR_counts2 <- read_delim("AMR_counts2.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
annotation_file <- megares_full_annotations_v2_00 <- read_csv("megares_full_annotations_v2.00.csv")
```
```{r}
AMR_counts <- AMR_counts2 %>% separate(contig, c("OTU_n","type","x1","x2"), sep = "\\|")
AMR_counts <- subset(AMR_counts, select = -c(type,x1,x2,length))
AMR_counts$OTU_n <- gsub('MEG', 'OTU', AMR_counts$OTU_n)
write_csv(AMR_counts, 'AMR_counts.csv')
OTU_counts <- read.csv("AMR_counts.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
OTU <- otu_table(OTU_counts, taxa_are_rows = TRUE)
```

```{r}
annotation_file <- annotation_file %>% separate(header, c("OTU_n","x1","x2"), sep = "\\|")
annotation_file <- subset(annotation_file, select = -c(x1,x2))
annotation_file$OTU_n <- gsub('MEG', 'OTU', annotation_file$OTU_n)
write_csv(annotation_file, 'annotation_file.csv')
TAX_pre <- read.csv("annotation_file.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
TAX_pre<- as.matrix(TAX_pre)
TAX <- tax_table(TAX_pre)
```

```{r}
Metadata2 <- read_csv("~/Desktop/Current_work/OTU_Builber/Sequencing_Data_Info_Sheet3.csv")
#Metadata2= as.matrix(Metadata)
META <- sample_data(Metadata2)
sample_names(META)= META$SampleID
```

```{r}
phy1 <- phyloseq(OTU, TAX, META)
```
```{r}
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21")
phy1.2 <- subset_samples(phy1, !(SampleID %in% Samples_toRemove))
```
```{r}
# get read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(phy1.2))
```

```{r}
# histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) +
  geom_histogram(color = "black", fill = "indianred", binwidth = 1000) +
  ggtitle("Distribution of sample sequencing depth") +
  xlab("Read counts") +
  theme(axis.title.y = element_blank())
```

```{r}
# mean, max and min of sample read counts
smin <- min(sample_sums(phy1.2))
smean <- mean(sample_sums(phy1.2))
smax <- max
```

```{r}
soil_study_phylum <- phy1.2 %>%
  tax_glom(taxrank = "group") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa
  arrange(group) 
```

```{r}
# Plot
ggplot(soil_study_phylum, aes(x = Sample, y = Abundance, fill = group)) +
  geom_bar(stat = "identity") + theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

```{r}
# plot richness
plot_richness(phy1.2, x = "Time_Collected", color = "Subject") + geom_boxplot()
```

```{r}
# Ordination Plot
# Ordinate
soil_study_pcoa <- ordinate(
  physeq = phy1.2,
  method = "PCoA",
  distance = "bray"
)

```

```{r}
# Plot PCoA
plot_ordination(
  physeq = phy1.2,
  ordination = soil_study_pcoa,
  color = "Subject",
  title = "PCoA of Soil Bacterial Communities") +
  geom_point(aes(color = Subject), alpha = 0.7, size = 4) +
  geom_point(colour = "black", size = 1.5)  + theme_minimal()

```


```{r}
# NMDS
set.seed(1)
soil_study_nmds <- ordinate(
  physeq = phy1.2,
  method = "NMDS",
  distance = "bray"
)
```
```{r}
# Plot NMDS
plot_ordination(
  physeq = phy1.2,
  ordination = soil_study_nmds,
  color = "Subject",
  title = "PCoA of Soil Bacterial Communities") +
  geom_point(aes(color = Subject), alpha = 0.7, size = 4) +
  geom_point(colour = "grey90", size = 1.5)  + theme_minimal()
```

```{r}
#Run some stats
set.seed(1)
soil_study_bray <- phyloseq::distance(phy1.2, method = "bray")
```

```{r}
# make a data frame from the sample_data
sampled_soil_study <- data.frame(sample_data(phy1.2))
```

```{r}
# Adonis test
adonis(soil_study_bray ~ Subject, data = sampled_soil_study)
```

```{r}
# Homogeneity of dispersion test
beta <- betadisper(soil_study_bray, sampled_soil_study$Subject)
permutest(beta)
```

```{r}
# Remove data points with missing metadata
soil_study_not_na <- phy1.2 %>%
  subset_samples(
    !is.na(Subject) &
      !is.na(Time_Collected) &
      !is.na(Location) &
      !is.na(Gender) &
      !is.na(Antibacterial_Soap_Used) &
      !is.na(Gel_Sanitizer_Use)
  )
bray_not_na <- phyloseq::distance(physeq = soil_study_not_na, method = "bray")
```

```{r}
# CAP ordinate
cap_ord <- ordinate(
  physeq = soil_study_not_na,
  method = "CAP",
  distance = bray_not_na,
  formula = ~ Subject + Time_Collected + Location + Gender + Antibacterial_Soap_Used + Gel_Sanitizer_Use
)
```

```{r}
# CAP plot
cap_plot <- plot_ordination(
  physeq = soil_study_not_na,
  ordination = cap_ord,
  color = "Subject",
  axes = c(1,2)) +
  geom_point(aes(colour = Subject), alpha = 0.4, size = 4) +
  geom_point(colour = "grey", size = 1.5)  + theme_minimal()
```

```{r}
# Now add the environmental variables as arrows
arrowmat <- vegan::scores(cap_ord, display = "bp")
```

```{r}
# Add labels, make a data.frame
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)

```

```{r}
# Define the arrow aesthetic mapping
arrow_map <- aes(xend = CAP1,
                 yend = CAP2,
                 x = 0,
                 y = 0,
                 shape = NULL,
                 color = NULL,
                 label = labels)

label_map <- aes(x = 1.3 * CAP1,
                 y = 1.3 * CAP2,
                 shape = NULL,
                 color = NULL,
                 label = labels)

arrowhead = arrow(length = unit(0.02, "npc"))
```

```{r}
# Make a new graphic
cap_plot +
  geom_segment(
    mapping = arrow_map,
    size = .5,
    data = arrowdf,
    color = "gray",
    arrow = arrowhead
  ) +
  geom_text(
    mapping = label_map,
    size = 4,
    data = arrowdf,
    show.legend = FALSE
  ) + theme_minimal()

```

```{r}
# Ordinate
soil_study_pcoa <- ordinate(
  physeq = phy1.2,
  method = "PCoA",
  distance = "bray"
)
```

```{r}
# ANOVA on permutation
anova(cap_ord)
```

```{r}
physeq<- filter_taxa(phy1.2, function(x) sum(x > 0) >= 1, TRUE)


phylum.sum<- tapply(taxa_sums(physeq), tax_table(physeq)[, "group"], sum, na.rm=TRUE)
top10phyla <-names(sort(phylum.sum, TRUE))[1:10]
physeq_top10 <- prune_taxa((tax_table(physeq)[, "group"] %in% top10phyla), physeq)

plot_bar(physeq_top10, x="Subject", fill = "group")
```
```{r}
physeq_drug<- subset_taxa(physeq, type =="Drugs")
phylum.sum_drug<- tapply(taxa_sums(physeq_drug), tax_table(physeq_drug)[, "group"], sum, na.rm=TRUE)
top10phyla_drug <-names(sort(phylum.sum_drug, TRUE))[1:10]
physeq_drug_top10 <- prune_taxa((tax_table(physeq_drug)[, "group"] %in% top10phyla_drug), physeq_drug)

plot_bar(physeq_drug_top10, x="Subject", fill = "group")

```


```{r}
physeq2 <- subset_taxa(physeq2, gene!="erm(C)" )
plot_bar(physeq2, x="gene")
plot_bar(physeq, x="gene")
```

```{r}
P25<- subset_samples(physeq, Subject=="P25")
P25<- subset_taxa(P25, group="ERMC")
```
#Used to evaluate person 25 just to see if there are some observed changes over time
```{r}
prevdf_ps= apply(X = otu_table(physeq_drug), 
                       MARGIN = ifelse(taxa_are_rows(physeq_drug), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(physeq_drug))
#View(prevdf_ps)
ps_IBK_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_IBK_run
ps_IBK_analyze <- prune_taxa(ps_IBK_run, physeq_drug)
ps_IBK_analyze
sum(otu_table(ps_IBK_analyze))/sum(otu_table(physeq_drug)) 
save(ps_IBK_analyze, file = "ps_IBK_analyze.rds")
load("ps_IBK_analyze.rds")
sum(otu_table(ps_IBK_analyze)) 
```
```{r}
norm_IBK  <-  transform_sample_counts(ps_IBK_analyze, function(x) x / sum(x) )
save(norm_IBK, file= "norm_IBK.rds")
load("norm_IBK.rds")
```




```{r, results='hide'}
prevdf_ps= apply(X = otu_table(physeq_drug), 
                       MARGIN = ifelse(taxa_are_rows(physeq_drug), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(physeq_drug))
#View(prevdf_ps)
ps_IBK_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_IBK_run
ps_IBK_analyze <- prune_taxa(ps_IBK_run, physeq_drug)
ps_IBK_analyze
sum(otu_table(ps_IBK_analyze))/sum(otu_table(physeq_drug)) 
save(ps_IBK_analyze, file = "ps_IBK_analyze.rds")
load("ps_IBK_analyze.rds")
sum(otu_table(ps_IBK_analyze)) 
```
```{r}
norm_IBK  <-  transform_sample_counts(ps_IBK_analyze, function(x) x / sum(x) )
save(norm_IBK, file= "norm_IBK.rds")
load("norm_IBK.rds")
```

```{r}
ps_rarefy <- rarefy_even_depth(ps_IBK_analyze, sample.size = min(sample_sums(ps_IBK_analyze)),
  rngseed = T, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
Chao1_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Chao1"))
head(Chao1_obser_rarefy)
obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Observed"))
head(obser_rarefy)
#write.table(obser_rarefy, "a.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
Chao1_obser_rarefy_metadata= merge(META,Chao1_obser_rarefy, by= "row.names")
head(Chao1_obser_rarefy_metadata)
#rarefy with t test
```

```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Location", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Location, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Location)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Location)
chao1_pairwise 
```


```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Time_Collected", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Time_Collected, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Time_Collected)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Time_Collected)
chao1_pairwise 
```


```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Gel_Sanitizer_Use", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Gel_Sanitizer_Use, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Gel_Sanitizer_Use)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Gel_Sanitizer_Use, correct=FALSE, p.adjust="none")
chao1_pairwise
```

```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Subject", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Subject, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Subject)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Subject, correct=FALSE, p.adjust="none")
chao1_pairwise
apl <- as.list(chao1_pairwise)
view(apl)
```

```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Antibacterial_Soap_Used", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Antibacterial_Soap_Used, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Antibacterial_Soap_Used)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Antibacterial_Soap_Used)
chao1_pairwise 
```

```{r}
norm_soap<- transform_sample_counts(physeq, function(x) x / sum(x) )
ord.soap.pcoa <- ordinate(norm_soap, method="PCoA", distance="bray")
plot_ordination(norm_soap, ord.soap.pcoa, color = "Subject") +
  geom_point(size=5) 
```



```{r}
soil_study_not_na <- physeq_drug %>%
  subset_samples(
    !is.na(Gel_Sanitizer_Use)
  )

```

```{r, results='hide'}
prevdf_ps= apply(X = otu_table(soil_study_not_na), 
                       MARGIN = ifelse(taxa_are_rows(soil_study_not_na), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(soil_study_not_na))
#View(prevdf_ps)
ps_IBK_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_IBK_run
ps_IBK_analyze <- prune_taxa(ps_IBK_run, soil_study_not_na)
ps_IBK_analyze
sum(otu_table(ps_IBK_analyze))/sum(otu_table(soil_study_not_na)) 
save(ps_IBK_analyze, file = "ps_IBK_analyze.rds")
load("ps_IBK_analyze.rds")
sum(otu_table(ps_IBK_analyze)) 
```
```{r}
norm_IBK  <-  transform_sample_counts(ps_IBK_analyze, function(x) x / sum(x) )
save(norm_IBK, file= "norm_IBK.rds")
load("norm_IBK.rds")
```

```{r}
ps_rarefy <- rarefy_even_depth(ps_IBK_analyze, sample.size = min(sample_sums(ps_IBK_analyze)),
  rngseed = T, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
Chao1_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Chao1"))
head(Chao1_obser_rarefy)
obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Observed"))
head(obser_rarefy)
#write.table(obser_rarefy, "a.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
Chao1_obser_rarefy_metadata= merge(META,Chao1_obser_rarefy, by= "row.names")
head(Chao1_obser_rarefy_metadata)
#rarefy with t test
```

```{r}
alpha_meas = c("Observed", "Chao1")
(p <- plot_richness(ps_rarefy, "Gel_Sanitizer_Use", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Gel_Sanitizer_Use, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Gel_Sanitizer_Use)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Gel_Sanitizer_Use, correct=FALSE, p.adjust="none")
chao1_pairwise
```
