---
title: "CheckV_analysis"
output: html_document
---



#Use for 1400 viral sequences

#read in OTU table
```{r}
#bwa
counts <- read_delim("counts2_checkv.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
#bowtie
counts <- read_delim("Final_Counts/counts2_no1.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
```
#read in annotations file and generate tax_table for phyloseq
```{r}
#vc_tax<-read.csv("vc_tax.txt", "\t", header = TRUE, stringsAsFactors = FALSE)
vc_tax <- read_csv("vc_tax.csv")
taxe2 <- vc_tax
taxe2$contig_id2 <- taxe2$contig_id
tax2 <- subset(taxe2, select = c(contig_id,contig_id2))
write_csv(tax2, 'VCAnntations.csv')
VCTAX_pre <- read.csv("VCAnntations.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
#VCTAX_pre<-read.csv("vc_tax.txt", "\t", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
VCTAX_pre<- as.matrix(VCTAX_pre)
VCTAX <- tax_table(VCTAX_pre)

tax_ann <- read.csv("vc_tax.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
#tax_ann<-read.csv("vc_tax.txt", "\t", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
tax_ann <- as.matrix(tax_ann)
#tax_ann <- subset(tax_ann, select = -c(X))
ATAX <- tax_table(tax_ann)
```
#remove contigs that are not in the annotation file and generate Otu table for phyloseq
```{r}
counts2 <- counts
contig_keep <- vc_tax$contig_id
counts2 <- counts2[(counts2$contig %in% contig_keep), ]
write_csv(counts2, 'VCcounts.csv')
VCOTU_counts <- read.csv("VCcounts.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
VCOTU <- otu_table(VCOTU_counts, taxa_are_rows = TRUE)


#counts3 <- counts2 %>% mutate_if(is.numeric, ~1 * (. != 0))
#write_csv(counts3, 'VCcounts_test.csv')
```
#read in metadata, edit column names, and generate sample data file for phyloseq
```{r}
Metadata2 <- read_csv("~/Desktop/Current_work/OTU_Builber/Sequencing_Data_Info_Sheet3.csv")
Metadata3 <- Metadata2
Metadata3$Time <- Metadata3$Time_Collected
Metadata3$Time[which(Metadata3$Time == "T_0")] <- "00_Weeks"
Metadata3$Time[which(Metadata3$Time == "2_week")] <- "02_Weeks"
Metadata3$Time[which(Metadata3$Time == "1_month")] <- "04_Weeks"
Metadata3$Time[which(Metadata3$Time == "3_month")] <- "12_Weeks"
Metadata3$Time[which(Metadata3$Time == "6_month")] <- "24_Weeks"

META <- sample_data(Metadata3)
sample_names(META)= META$SampleID
```
#make phyloseq object and remove control/repeat samples from phyloseq object
```{r}
VCPHY <- phyloseq(VCOTU, VCTAX, META)
ANPHY <- phyloseq(VCOTU, ATAX, META)
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01","HV_003_15")

VCPHY2 <- subset_samples(VCPHY, !(SampleID %in% Samples_toRemove))
VCPHY1 <- subset_samples(VCPHY, !(SampleID %in% Samples_toRemove))
ANPHY1 <- subset_samples(ANPHY, !(SampleID %in% Samples_toRemove))
```
#Rarefy data for diversity analysis
```{r}
prevdf_ps= apply(X = otu_table(ANPHY1), 
                       MARGIN = ifelse(taxa_are_rows(ANPHY1), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(ANPHY1))
#View(prevdf_ps)
ps_virome_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_virome_run
ps_virome_analyze <- prune_taxa(ps_virome_run, ANPHY1)
ps_virome_analyze
sum(otu_table(ps_virome_analyze))/sum(otu_table(ANPHY1)) 
save(ps_virome_analyze, file = "ps_virome_analyze.rds")
load("ps_virome_analyze.rds")
sum(otu_table(ps_virome_analyze)) 
ps_rarefy <- rarefy_even_depth(ps_virome_analyze, sample.size = min(sample_sums(ps_virome_analyze)), rngseed = T, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)

Chao1_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Chao1"))
head(Chao1_obser_rarefy)
obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Observed"))
head(obser_rarefy)
shannon_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Shannon"))
Chao1_obser_rarefy_metadata= merge(META,Chao1_obser_rarefy, by= "row.names")
head(Chao1_obser_rarefy_metadata)
shannon_obser_rarefy_metadata= merge(META,shannon_obser_rarefy, by= "row.names")
```
#run alpha diversity measures for Month_Sampled
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Month_Sampled", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Month_Sampled, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Month_Sampled, correct=FALSE, p.adjust="none")
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Month_Sampled,correct=FALSE, p.adjust="none")
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Month_Sampled, correct=FALSE, p.adjust="none")
shannon_pairwise

```
#Rarefy data for diversity analysis
```{r}
prevdf_ps= apply(X = otu_table(VCPHY1), 
                       MARGIN = ifelse(taxa_are_rows(VCPHY1), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(VCPHY1))
#View(prevdf_ps)
ps_virome_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_virome_run
ps_virome_analyze <- prune_taxa(ps_virome_run, VCPHY1)
ps_virome_analyze
sum(otu_table(ps_virome_analyze))/sum(otu_table(VCPHY1)) 
save(ps_virome_analyze, file = "ps_virome_analyze.rds")
load("ps_virome_analyze.rds")
sum(otu_table(ps_virome_analyze)) 
ps_rarefy <- rarefy_even_depth(ps_virome_analyze, sample.size = min(sample_sums(ps_virome_analyze)), rngseed = T, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)

Chao1_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Chao1"))
head(Chao1_obser_rarefy)
obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Observed"))
head(obser_rarefy)
shannon_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Shannon"))
Chao1_obser_rarefy_metadata= merge(META,Chao1_obser_rarefy, by= "row.names")
head(Chao1_obser_rarefy_metadata)
shannon_obser_rarefy_metadata= merge(META,shannon_obser_rarefy, by= "row.names")
```
#run alpha diversity measures for Month_Sampled
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Month_Sampled", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Month_Sampled, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Month_Sampled)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Month_Sampled)
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Month_Sampled)
shannon_pairwise
```

#run alpha diversity measures for Time_Collected
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Time_Collected", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Time_Collected, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Time_Collected)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Time_Collected)
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Time_Collected)
shannon_pairwise
```

#run alpha diversity measures for Gender
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Gender", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Gender, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Gender)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Gender)
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Gender)
shannon_pairwise
```

#run alpha diversity measures for Subject
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Subject", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Subject, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Subject)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Subject)
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Subject)
shannon_pairwise
```

#run alpha diversity measures for Location
```{r}
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Location", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Location, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Location)
observed_pairwise
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Location)
chao1_pairwise 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Location)
shannon_pairwise
```

##############_______CORE____###################


```{r}
#https://microbiome.github.io/tutorials/Core.html
#library(microbiome)

pseq.rel <- microbiome::transform(VCPHY1, "compositional")
head(prevalence(pseq.rel, detection = 1/100, sort = TRUE))
head(prevalence(pseq.rel, detection = 1/100, sort = TRUE, count = TRUE))


```







#Generate core taxa that are present across 80% of time points
```{r}
#Core taxa across 80% of all samples
CORE<- filter_taxa(ANPHY1, function(x) sum(x > 0) >= (0.8*length(x)), TRUE)

CORE.sum<- tapply(taxa_sums(CORE), tax_table(CORE)[, "family"], sum, na.rm=TRUE)

```
#barplot of top 10 family level taxa in core taxa 
```{r}
CORE.sum<- tapply(taxa_sums(CORE), tax_table(CORE)[, "family"], sum, na.rm=TRUE)
COREtop10phyla <-names(sort(CORE.sum, TRUE))[1:10]
CORE_top10 <- prune_taxa((tax_table(CORE)[, "family"] %in% COREtop10phyla), CORE)
plot_bar(CORE_top10, x="Subject", fill = "family")
```

#CORE BY LOCATION
```{R}
#scalp
CORE_SC <- subset_samples(ANPHY1, Location=="scalp")
CORE_SC<- filter_taxa(CORE_SC, function(x) sum(x > 0) >= (0.8*length(x)), TRUE)
#left
CORE_LH <- subset_samples(ANPHY1, Location=="left")
CORE_LH<- filter_taxa(CORE_LH, function(x) sum(x > 0) >= (0.8*length(x)), TRUE)
#right
CORE_RH <- subset_samples(ANPHY1, Location=="right")
CORE_RH<- filter_taxa(CORE_RH, function(x) sum(x > 0) >= (0.8*length(x)), TRUE)
#hands
CORE_LH2 <- subset_samples(ANPHY1, Location=="left")
CORE_RH2 <- subset_samples(ANPHY1, Location=="right")
CORE_HANDS <- merge_phyloseq(CORE_RH2 , CORE_LH2)
CORE_HANDS<- filter_taxa(CORE_HANDS, function(x) sum(x > 0) >= (0.8*length(x)), TRUE)
```
#Person 01 Core
```{r}
#Subset by person 01
P_01 <- subset_samples(ANPHY1, Subject=="P01")
otu_table(P_01)
#Subset person 01 by locations
P_01_LH<- subset_samples(P_01, Location=="left")
#P_01_LHf <- tax_glom(P_01_LH, "family")
P_01_RH<- subset_samples(P_01, Location=="right")
#P_01_RHf <- tax_glom(P_01_RH, "family")
P_01_SC<- subset_samples(P_01, Location=="scalp")
#P_01_SCf <- tax_glom(P_01_SC, "family")
#Find taxa present across 80% of the repeated time points for each location for person 01 
core_P01LH <- filter_taxa(P_01_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
#core_P01LHf <- filter_taxa(P_01_LHf, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P01RH <- filter_taxa(P_01_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
#core_P01RHf <- filter_taxa(P_01_RHf, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P01SC <- filter_taxa(P_01_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
#core_P01SCf <- filter_taxa(P_01_SCf, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
#merge core locations for person 01 together to make an overall core for person 01
core_P01 <- merge_phyloseq(core_P01LH , core_P01RH, core_P01SC)
#core_P01f <- merge_phyloseq(core_P01LHf , core_P01RHf, core_P01SCf)
#Subtract out overall CORE to get just unique core taxa for person 01 
core_rm<- rownames(otu_table(CORE))
core_P01_rm <- taxa_names(core_P01)
core_P01_rm <- core_P01_rm[!(core_P01_rm %in% core_rm)]
core_P01_rm <- prune_taxa(core_P01_rm, core_P01)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P01_rm <- taxa_names(core_P01SC)
core_SC_P01_rm <- core_SC_P01_rm[!(core_SC_P01_rm %in% core_SC_rm)]
core_SC_P01_rm <- prune_taxa(core_SC_P01_rm, core_P01SC)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P01_rm <- taxa_names(core_P01LH)
core_LH_P01_rm <- core_LH_P01_rm[!(core_LH_P01_rm %in% core_LH_rm)]
core_LH_P01_rm <- prune_taxa(core_LH_P01_rm, core_P01LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P01_rm <- taxa_names(core_P01RH)
core_RH_P01_rm <- core_RH_P01_rm[!(core_RH_P01_rm %in% core_RH_rm)]
core_RH_P01_rm <- prune_taxa(core_RH_P01_rm, core_P01RH)


core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P01_hands <- merge_phyloseq(core_P01LH , core_P01RH)
core_HANDS_P01_rm <- taxa_names(core_P01_hands)
core_HANDS_P01_rm <- core_HANDS_P01_rm[!(core_HANDS_P01_rm %in% core_HANDS_rm)]
core_HANDS_P01_rm <- prune_taxa(core_HANDS_P01_rm, core_P01_hands)
```

#Beta diversity for person 01 core 
```{r}
#P01 Core
norm_core_P01 <- transform_sample_counts(core_P01, function(x) x / sum(x) )
ord.core_01.pcoa <- ordinate(norm_core_P01, method="PCoA", distance="bray")
plot_ordination(norm_core_P01, ord.core_01.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
#P01 unique core 
norm_core_P01_rm <- transform_sample_counts(core_P01_rm, function(x) x / sum(x) )
ord.core_01_rm.pcoa <- ordinate(norm_core_P01_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P01_rm, ord.core_01_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

```{r}
# Extract abundance matrix from the phyloseq object
Total_OTU = as(otu_table(ANPHY1), "matrix")
# transpose if necessary
Total_OTU <- t(Total_OTU)
# Coerce to data.frame
Total_OTUdf = as.data.frame(Total_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01", "HV_003_15")

Metadata_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_002_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_03", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_04", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_06", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_07", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_021_13", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_01", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_020_10", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_015_21", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_05", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_020_22", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_02", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_015_20", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_003_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_004_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_005_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_006_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_007_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_008_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_009_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_010_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_011_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_012_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_013_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_014_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_015_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_016_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_017_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_018_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_019_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_020_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_021_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_022_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_023_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_024_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_025_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_026_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_027_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_028_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_029_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_030_27", Metadata_stat$SampleID),]
Metadata_stat <- Metadata_stat[!grepl("HV_003_15", Metadata_stat$SampleID),]


#Total_OTUdf2<- Total_OTUdf
#Total_OTUdf2$SampleID <- row.names(Total_OTUdf2)
#Total_OTUdf2 <- subset(Total_OTUdf2, select = c(SampleID))
#Metadata_stat2 <- subset(Metadata_stat, select = c(SampleID))
#anti_join(Metadata_stat2, Total_OTUdf2)

#alpha diversity measures:
Metadata_stat$shannon.vegan <- diversity(Total_OTUdf, index = "shannon")
Metadata_stat$simpson.vegan <- diversity(Total_OTUdf, index = "simpson")
Metadata_stat$invsimpson.vegan <- diversity(Total_OTUdf, index = "invsimpson")
```
```{r}
model_AOV <- aov(shannon.vegan~Gender+Subject/Location+ Subject/Time_Collected+Subject, data = Metadata_stat)
summary(model_AOV)
```

```{r}
norm_groups <- transform_sample_counts(ANPHY1, function(x) x / sum(x) )
ord.core_groups.pcoa <- ordinate(norm_groups, method="PCoA", distance="jaccard", na.rm = TRUE)
plot_ordination(norm_groups, ord.core_groups.pcoa, color = "Subject", shape = "Location") +
  geom_point(size=1)
```

```{r}
metadata <- as(sample_data(norm_groups ), "data.frame")
adonis(phyloseq::distance(norm_groups, method="jaccard") ~  Gender+Subject/Location+Subject,
       data = metadata)
```
```{r}
ANPHY1_LH<- subset_samples(ANPHY1, Location=="left")
# Extract abundance matrix from the phyloseq object
Total_OTU_LH = as(otu_table(ANPHY1_LH), "matrix")
# transpose if necessary
Total_OTU_LH <- t(Total_OTU_LH)
# Coerce to data.frame
Total_OTUdf_LH = as.data.frame(Total_OTU_LH)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01", "HV_003_15")

Metadata_stat_LH <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_002_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_03", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_04", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_06", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_07", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_021_13", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_01", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_020_10", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_015_21", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_05", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_020_22", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_02", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_015_20", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_003_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_004_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_005_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_006_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_007_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_008_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_009_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_010_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_011_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_012_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_013_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_014_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_015_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_016_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_017_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_018_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_019_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_020_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_021_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_022_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_023_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_024_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_025_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_026_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_027_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_028_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_029_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_030_27", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("HV_003_15", Metadata_stat_LH$SampleID),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("right", Metadata_stat_LH$Location),]
Metadata_stat_LH <- Metadata_stat_LH[!grepl("scalp", Metadata_stat_LH$Location),]


#Total_OTUdf2<- Total_OTUdf
#Total_OTUdf2$SampleID <- row.names(Total_OTUdf2)
#Total_OTUdf2 <- subset(Total_OTUdf2, select = c(SampleID))
#Metadata_stat2 <- subset(Metadata_stat, select = c(SampleID))
#anti_join(Metadata_stat2, Total_OTUdf2)

#alpha diversity measures:
Metadata_stat_LH$shannon.vegan <- diversity(Total_OTUdf_LH, index = "shannon")
Metadata_stat_LH$simpson.vegan <- diversity(Total_OTUdf_LH, index = "simpson")
Metadata_stat_LH$invsimpson.vegan <- diversity(Total_OTUdf_LH, index = "invsimpson")
```
```{r}
model_AOV_LH <- aov(shannon.vegan~Gender+Subject/Time_Collected+Subject, data = Metadata_stat_LH)
summary(model_AOV_LH)
```
```{r}
norm_groups_LH <- transform_sample_counts(ANPHY1_LH, function(x) x / sum(x) )
ord.LH.pcoa <- ordinate(norm_groups_LH, method="PCoA", distance="jaccard", na.rm = TRUE)
plot_ordination(norm_groups_LH, ord.LH.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```
```{r}
metadata <- as(sample_data(norm_groups_LH), "data.frame")
adonis(phyloseq::distance(norm_groups_LH, method="jaccard") ~  Gender+Subject/Time_Collected+Subject,
       data = metadata)
```
######################################
```{r}
#Subset by person 01
P_01 <- subset_samples(ANPHY1, Subject=="P01")
#Subset person 01 by locations
P_01_LH<- subset_samples(P_01, Location=="left")
P_01_RH<- subset_samples(P_01, Location=="right")
P_01_SC<- subset_samples(P_01, Location=="scalp")

prevelancedf <- apply(X = otu_table(P_01_LH),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf <- data.frame(Prevalence = prevelancedf,
                          TotalAbundance = taxa_sums(P_01_LH),
                          tax_table(P_01_LH))
prevelancedf_stable <- subset(prevelancedf, prevelancedf$Prevalence >= 4)  
```
```{r}
#Subset by person 01
P_01 <- subset_samples(ANPHY1, Subject=="P01")
P_01_RH<- subset_samples(P_01, Location=="right")

prevelancedf_P_01_RH <- apply(X = otu_table(P_01_RH),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf_P_01_RH <- data.frame(Prevalence = prevelancedf_P_01_RH,
                          TotalAbundance = taxa_sums(P_01_RH),
                          tax_table(P_01_RH))
prevelancedf_P_01_RH_stable <- subset(prevelancedf_P_01_RH, prevelancedf_P_01_RH$Prevalence >= 4)  
```
Metadata3$Time <- Metadata3$Time_Collected
Metadata3$Time[which(Metadata3$Time == "T_0")] <- "00_Weeks"
Metadata3$Time[which(Metadata3$Time == "2_week")] <- "02_Weeks"
Metadata3$Time[which(Metadata3$Time == "1_month")] <- "04_Weeks"
Metadata3$Time[which(Metadata3$Time == "3_month")] <- "12_Weeks"
Metadata3$Time[which(Metadata3$Time == "6_month")] <- "24_Weeks"
```{r}
#Subset by person 01
P_01 <- subset_samples(ANPHY1, Subject=="P01")
P_01_RH<- subset_samples(P_01, Location=="right")
P_01_RH_1<- subset_samples(P_01_RH, Time_Collected=="T_0")

prevelancedf_P_01_RH_1 <- apply(X = otu_table(P_01_RH_1),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf_P_01_RH_1 <- data.frame(Prevalence = prevelancedf_P_01_RH_1,
                          TotalAbundance = taxa_sums(P_01_RH_1),
                          tax_table(P_01_RH_1))

P01_Prev_RH_1<-plyr::ddply(prevelancedf_P_01_RH_1, "Species", function(df1){
  data.frame(mean_prevalence_T_0=sum(df1$Prevalence),total_abundance_T_0=sum(df1$TotalAbundance,na.rm = T),stringsAsFactors = F)
  })


P_01_RH_2<- subset_samples(P_01_RH, Time_Collected=="2_week")

prevelancedf_P_01_RH_2 <- apply(X = otu_table(P_01_RH_2),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf_P_01_RH_2 <- data.frame(Prevalence = prevelancedf_P_01_RH_2,
                          TotalAbundance = taxa_sums(P_01_RH_2),
                          tax_table(P_01_RH_2))

P01_Prev_RH_2<-plyr::ddply(prevelancedf_P_01_RH_2, "Species", function(df1){
  data.frame(mean_prevalence_2_week=sum(df1$Prevalence),total_abundance_2_week=sum(df1$TotalAbundance,na.rm = T),stringsAsFactors = F)
  })
 
P01_RH_prev <- merge(P01_Prev_RH_1,P01_Prev_RH_2,by="Species")

P01_RH_prev$mean_prevalence_T_0[P01_RH_prev$mean_prevalence_T_0 > 0] <- 1 
P01_RH_prev$mean_prevalence_2_week[P01_RH_prev$mean_prevalence_2_week > 0] <- 1 
P01_RH_prev$TOTAL_PREVALENCE <- P01_RH_prev$mean_prevalence_2_week+P01_RH_prev$mean_prevalence_T_0



#prevelancedf_P_01_RH_1_total<- subset(prevelancedf_P_01_RH_1, prevelancedf_P_01_RH_1$Prevalence >= 4)  
```
df.richness %>% mutate_if(is.numeric, ~1 * (. != 0))

```{r}
#Subset by person 01
P_01 <- subset_samples(VCPHY1, Subject=="P01")
P_01_RH<- subset_samples(P_01, Location=="right")

prevelancedf_P_01_RH <- apply(X = otu_table(P_01_RH),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf_P_01_RH <- data.frame(Prevalence = prevelancedf_P_01_RH,
                          TotalAbundance = taxa_sums(P_01_RH),
                          tax_table(P_01_RH))
prevelancedf_P_01_RH_stable <- subset(prevelancedf_P_01_RH, prevelancedf_P_01_RH$Prevalence >= 3)  
```

#The previous two coding chunks are then repeated for each person as seen below
#For my sample set I only have persons 1-43, make sure to add more for each additional person you have in your dataset 

#P02
```{r}
P_02 <- subset_samples(VCPHY1, Subject=="P02")
P_02_LH<- subset_samples(P_02, Location=="left")
P_02_RH<- subset_samples(P_02, Location=="right")
P_02_SC<- subset_samples(P_02, Location=="scalp")

core_P02LH <- filter_taxa(P_02_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P02RH <- filter_taxa(P_02_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P02SC <- filter_taxa(P_02_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P02 <- merge_phyloseq(core_P02LH , core_P02RH, core_P02SC)

core_rm<- rownames(otu_table(CORE))
core_P02_rm <- taxa_names(core_P02)
core_P02_rm <- core_P02_rm[!(core_P02_rm %in% core_rm)]
core_P02_rm <- prune_taxa(core_P02_rm, core_P02)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P02_rm <- taxa_names(core_P02SC)
core_SC_P02_rm <- core_SC_P02_rm[!(core_SC_P02_rm %in% core_SC_rm)]
core_SC_P02_rm <- prune_taxa(core_SC_P02_rm, core_P02SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P02_hands <- merge_phyloseq(core_P02LH , core_P02RH)
core_HANDS_P02_rm <- taxa_names(core_P02_hands)
core_HANDS_P02_rm <- core_HANDS_P02_rm[!(core_HANDS_P02_rm %in% core_HANDS_rm)]
core_HANDS_P02_rm <- prune_taxa(core_HANDS_P02_rm, core_P02_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P02_rm <- taxa_names(core_P02LH)
core_LH_P02_rm <- core_LH_P02_rm[!(core_LH_P02_rm %in% core_LH_rm)]
core_LH_P02_rm <- prune_taxa(core_LH_P02_rm, core_P02LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P02_rm <- taxa_names(core_P02RH)
core_RH_P02_rm <- core_RH_P02_rm[!(core_RH_P02_rm %in% core_RH_rm)]
core_RH_P02_rm <- prune_taxa(core_RH_P02_rm, core_P02RH)
```
```{r}
norm_core_P02 <- transform_sample_counts(core_P02, function(x) x / sum(x) )
ord.core_02.pcoa <- ordinate(norm_core_P02, method="PCoA", distance="bray")
plot_ordination(norm_core_P02, ord.core_02.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P02_rm <- transform_sample_counts(core_P02_rm, function(x) x / sum(x) )
ord.core_02_rm.pcoa <- ordinate(norm_core_P02_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P02_rm, ord.core_02_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P03
```{r}
P_03 <- subset_samples(VCPHY1, Subject=="P03")
P_03_LH<- subset_samples(P_03, Location=="left")
P_03_RH<- subset_samples(P_03, Location=="right")
P_03_SC<- subset_samples(P_03, Location=="scalp")
core_P03LH <- filter_taxa(P_03_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P03RH <- filter_taxa(P_03_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P03SC <- filter_taxa(P_03_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P03 <- merge_phyloseq(core_P03LH , core_P03RH, core_P03SC)

core_rm<- rownames(otu_table(CORE))
core_P03_rm <- taxa_names(core_P03)
core_P03_rm <- core_P03_rm[!(core_P03_rm %in% core_rm)]
core_P03_rm <- prune_taxa(core_P03_rm, core_P03)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P03_rm <- taxa_names(core_P03SC)
core_SC_P03_rm <- core_SC_P03_rm[!(core_SC_P03_rm %in% core_SC_rm)]
core_SC_P03_rm <- prune_taxa(core_SC_P03_rm, core_P03SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P03_hands <- merge_phyloseq(core_P03LH , core_P03RH)
core_HANDS_P03_rm <- taxa_names(core_P03_hands)
core_HANDS_P03_rm <- core_HANDS_P03_rm[!(core_HANDS_P03_rm %in% core_HANDS_rm)]
core_HANDS_P03_rm <- prune_taxa(core_HANDS_P03_rm, core_P03_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P03_rm <- taxa_names(core_P03LH)
core_LH_P03_rm <- core_LH_P03_rm[!(core_LH_P03_rm %in% core_LH_rm)]
core_LH_P03_rm <- prune_taxa(core_LH_P03_rm, core_P03LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P03_rm <- taxa_names(core_P03RH)
core_RH_P03_rm <- core_RH_P03_rm[!(core_RH_P03_rm %in% core_RH_rm)]
core_RH_P03_rm <- prune_taxa(core_RH_P03_rm, core_P03RH)
```
```{r}
norm_core_P03 <- transform_sample_counts(core_P03, function(x) x / sum(x) )
ord.core_03.pcoa <- ordinate(norm_core_P03, method="PCoA", distance="bray")
plot_ordination(norm_core_P03, ord.core_03.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P03_rm <- transform_sample_counts(core_P03_rm, function(x) x / sum(x) )
ord.core_03_rm.pcoa <- ordinate(norm_core_P03_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P03_rm, ord.core_03_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)

```

#P04
```{r}
P_04 <- subset_samples(VCPHY1, Subject=="P04")
P_04_LH<- subset_samples(P_04, Location=="left")
P_04_RH<- subset_samples(P_04, Location=="right")
P_04_SC<- subset_samples(P_04, Location=="scalp")
core_P04LH <- filter_taxa(P_04_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P04RH <- filter_taxa(P_04_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P04SC <- filter_taxa(P_04_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P04 <- merge_phyloseq(core_P04LH , core_P04RH, core_P04SC)

core_rm<- rownames(otu_table(CORE))
core_P04_rm <- taxa_names(core_P04)
core_P04_rm <- core_P04_rm[!(core_P04_rm %in% core_rm)]
core_P04_rm <- prune_taxa(core_P04_rm, core_P04)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P04_rm <- taxa_names(core_P04SC)
core_SC_P04_rm <- core_SC_P04_rm[!(core_SC_P04_rm %in% core_SC_rm)]
core_SC_P04_rm <- prune_taxa(core_SC_P04_rm, core_P04SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P04_hands <- merge_phyloseq(core_P04LH , core_P04RH)
core_HANDS_P04_rm <- taxa_names(core_P04_hands)
core_HANDS_P04_rm <- core_HANDS_P04_rm[!(core_HANDS_P04_rm %in% core_HANDS_rm)]
core_HANDS_P04_rm <- prune_taxa(core_HANDS_P04_rm, core_P04_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P04_rm <- taxa_names(core_P04LH)
core_LH_P04_rm <- core_LH_P04_rm[!(core_LH_P04_rm %in% core_LH_rm)]
core_LH_P04_rm <- prune_taxa(core_LH_P04_rm, core_P04LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P04_rm <- taxa_names(core_P04RH)
core_RH_P04_rm <- core_RH_P04_rm[!(core_RH_P04_rm %in% core_RH_rm)]
core_RH_P04_rm <- prune_taxa(core_RH_P04_rm, core_P04RH)
```
```{r}
norm_core_P04 <- transform_sample_counts(core_P04, function(x) x / sum(x) )
ord.core_04.pcoa <- ordinate(norm_core_P04, method="PCoA", distance="bray")
plot_ordination(norm_core_P04, ord.core_04.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P04_rm <- transform_sample_counts(core_P04_rm, function(x) x / sum(x) )
ord.core_04_rm.pcoa <- ordinate(norm_core_P04_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P04_rm, ord.core_04_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P05
```{r}
P_05 <- subset_samples(VCPHY1, Subject=="P05")
P_05_LH<- subset_samples(P_05, Location=="left")
P_05_RH<- subset_samples(P_05, Location=="right")
P_05_SC<- subset_samples(P_05, Location=="scalp")
core_P05LH <- filter_taxa(P_05_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P05RH <- filter_taxa(P_05_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P05SC <- filter_taxa(P_05_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P05 <- merge_phyloseq(core_P05LH , core_P05RH, core_P05SC)

core_rm<- rownames(otu_table(CORE))
core_P05_rm <- taxa_names(core_P05)
core_P05_rm <- core_P05_rm[!(core_P05_rm %in% core_rm)]
core_P05_rm <- prune_taxa(core_P05_rm, core_P05)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P05_rm <- taxa_names(core_P05SC)
core_SC_P05_rm <- core_SC_P05_rm[!(core_SC_P05_rm %in% core_SC_rm)]
core_SC_P05_rm <- prune_taxa(core_SC_P05_rm, core_P05SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P05_hands <- merge_phyloseq(core_P05LH , core_P05RH)
core_HANDS_P05_rm <- taxa_names(core_P05_hands)
core_HANDS_P05_rm <- core_HANDS_P05_rm[!(core_HANDS_P05_rm %in% core_HANDS_rm)]
core_HANDS_P05_rm <- prune_taxa(core_HANDS_P05_rm, core_P05_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P05_rm <- taxa_names(core_P05LH)
core_LH_P05_rm <- core_LH_P05_rm[!(core_LH_P05_rm %in% core_LH_rm)]
core_LH_P05_rm <- prune_taxa(core_LH_P05_rm, core_P05LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P05_rm <- taxa_names(core_P05RH)
core_RH_P05_rm <- core_RH_P05_rm[!(core_RH_P05_rm %in% core_RH_rm)]
core_RH_P05_rm <- prune_taxa(core_RH_P05_rm, core_P05RH)
```
```{r}
norm_core_P05 <- transform_sample_counts(core_P05, function(x) x / sum(x) )
ord.core_05.pcoa <- ordinate(norm_core_P05, method="PCoA", distance="bray")
plot_ordination(norm_core_P05, ord.core_05.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P05_rm <- transform_sample_counts(core_P05_rm, function(x) x / sum(x) )
ord.core_05_rm.pcoa <- ordinate(norm_core_P05_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P05_rm, ord.core_05_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P06
```{r}
P_06 <- subset_samples(VCPHY1, Subject=="P06")
P_06_LH<- subset_samples(P_06, Location=="left")
P_06_RH<- subset_samples(P_06, Location=="right")
P_06_SC<- subset_samples(P_06, Location=="scalp")
core_P06LH <- filter_taxa(P_06_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P06RH <- filter_taxa(P_06_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P06SC <- filter_taxa(P_06_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P06 <- merge_phyloseq(core_P06LH , core_P06RH, core_P06SC)

core_rm<- rownames(otu_table(CORE))
core_P06_rm <- taxa_names(core_P06)
core_P06_rm <- core_P06_rm[!(core_P06_rm %in% core_rm)]
core_P06_rm <- prune_taxa(core_P06_rm, core_P06)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P06_rm <- taxa_names(core_P06SC)
core_SC_P06_rm <- core_SC_P06_rm[!(core_SC_P06_rm %in% core_SC_rm)]
core_SC_P06_rm <- prune_taxa(core_SC_P06_rm, core_P06SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P06_hands <- merge_phyloseq(core_P06LH , core_P06RH)
core_HANDS_P06_rm <- taxa_names(core_P06_hands)
core_HANDS_P06_rm <- core_HANDS_P06_rm[!(core_HANDS_P06_rm %in% core_HANDS_rm)]
core_HANDS_P06_rm <- prune_taxa(core_HANDS_P06_rm, core_P06_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P06_rm <- taxa_names(core_P06LH)
core_LH_P06_rm <- core_LH_P06_rm[!(core_LH_P06_rm %in% core_LH_rm)]
core_LH_P06_rm <- prune_taxa(core_LH_P06_rm, core_P06LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P06_rm <- taxa_names(core_P06RH)
core_RH_P06_rm <- core_RH_P06_rm[!(core_RH_P06_rm %in% core_RH_rm)]
core_RH_P06_rm <- prune_taxa(core_RH_P06_rm, core_P06RH)
```
```{r}
norm_core_P06 <- transform_sample_counts(core_P06, function(x) x / sum(x) )
ord.core_06.pcoa <- ordinate(norm_core_P06, method="PCoA", distance="bray")
plot_ordination(norm_core_P06, ord.core_06.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P06_rm <- transform_sample_counts(core_P06_rm, function(x) x / sum(x) )
ord.core_06_rm.pcoa <- ordinate(norm_core_P06_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P06_rm, ord.core_06_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P07
```{r}
P_07 <- subset_samples(VCPHY1, Subject=="P07")
P_07_LH<- subset_samples(P_07, Location=="left")
P_07_RH<- subset_samples(P_07, Location=="right")
P_07_SC<- subset_samples(P_07, Location=="scalp")
core_P07LH <- filter_taxa(P_07_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P07RH <- filter_taxa(P_07_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P07SC <- filter_taxa(P_07_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P07 <- merge_phyloseq(core_P07LH , core_P07RH, core_P07SC)

core_rm<- rownames(otu_table(CORE))
core_P07_rm <- taxa_names(core_P07)
core_P07_rm <- core_P07_rm[!(core_P07_rm %in% core_rm)]
core_P07_rm <- prune_taxa(core_P07_rm, core_P07)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P07_rm <- taxa_names(core_P07SC)
core_SC_P07_rm <- core_SC_P07_rm[!(core_SC_P07_rm %in% core_SC_rm)]
core_SC_P07_rm <- prune_taxa(core_SC_P07_rm, core_P07SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P07_hands <- merge_phyloseq(core_P07LH , core_P07RH)
core_HANDS_P07_rm <- taxa_names(core_P07_hands)
core_HANDS_P07_rm <- core_HANDS_P07_rm[!(core_HANDS_P07_rm %in% core_HANDS_rm)]
core_HANDS_P07_rm <- prune_taxa(core_HANDS_P07_rm, core_P07_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P07_rm <- taxa_names(core_P07LH)
core_LH_P07_rm <- core_LH_P07_rm[!(core_LH_P07_rm %in% core_LH_rm)]
core_LH_P07_rm <- prune_taxa(core_LH_P07_rm, core_P07LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P07_rm <- taxa_names(core_P07RH)
core_RH_P07_rm <- core_RH_P07_rm[!(core_RH_P07_rm %in% core_RH_rm)]
core_RH_P07_rm <- prune_taxa(core_RH_P07_rm, core_P07RH)
```
```{r}
norm_core_P07 <- transform_sample_counts(core_P07, function(x) x / sum(x) )
ord.core_07.pcoa <- ordinate(norm_core_P07, method="PCoA", distance="bray")
plot_ordination(norm_core_P07, ord.core_07.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P07_rm <- transform_sample_counts(core_P07_rm, function(x) x / sum(x) )
ord.core_07_rm.pcoa <- ordinate(norm_core_P07_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P07_rm, ord.core_07_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P08
```{r}
P_08 <- subset_samples(VCPHY1, Subject=="P08")
P_08_LH<- subset_samples(P_08, Location=="left")
P_08_RH<- subset_samples(P_08, Location=="right")
P_08_SC<- subset_samples(P_08, Location=="scalp")
core_P08LH <- filter_taxa(P_08_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P08RH <- filter_taxa(P_08_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P08SC <- filter_taxa(P_08_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P08 <- merge_phyloseq(core_P08LH , core_P08RH, core_P08SC)

core_rm<- rownames(otu_table(CORE))
core_P08_rm <- taxa_names(core_P08)
core_P08_rm <- core_P08_rm[!(core_P08_rm %in% core_rm)]
core_P08_rm <- prune_taxa(core_P08_rm, core_P08)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P08_rm <- taxa_names(core_P08SC)
core_SC_P08_rm <- core_SC_P08_rm[!(core_SC_P08_rm %in% core_SC_rm)]
core_SC_P08_rm <- prune_taxa(core_SC_P08_rm, core_P08SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P08_hands <- merge_phyloseq(core_P08LH , core_P08RH)
core_HANDS_P08_rm <- taxa_names(core_P08_hands)
core_HANDS_P08_rm <- core_HANDS_P08_rm[!(core_HANDS_P08_rm %in% core_HANDS_rm)]
core_HANDS_P08_rm <- prune_taxa(core_HANDS_P08_rm, core_P08_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P08_rm <- taxa_names(core_P08LH)
core_LH_P08_rm <- core_LH_P08_rm[!(core_LH_P08_rm %in% core_LH_rm)]
core_LH_P08_rm <- prune_taxa(core_LH_P08_rm, core_P08LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P08_rm <- taxa_names(core_P08RH)
core_RH_P08_rm <- core_RH_P08_rm[!(core_RH_P08_rm %in% core_RH_rm)]
core_RH_P08_rm <- prune_taxa(core_RH_P08_rm, core_P08RH)
```
```{r}
norm_core_P08 <- transform_sample_counts(core_P08, function(x) x / sum(x) )
ord.core_08.pcoa <- ordinate(norm_core_P08, method="PCoA", distance="bray")
plot_ordination(norm_core_P08, ord.core_08.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P08_rm <- transform_sample_counts(core_P08_rm, function(x) x / sum(x) )
ord.core_08_rm.pcoa <- ordinate(norm_core_P08_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P08_rm, ord.core_08_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P09
```{r}
P_09 <- subset_samples(VCPHY1, Subject=="P09")
P_09_LH<- subset_samples(P_09, Location=="left")
P_09_RH<- subset_samples(P_09, Location=="right")
P_09_SC<- subset_samples(P_09, Location=="scalp")
core_P09LH <- filter_taxa(P_09_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P09RH <- filter_taxa(P_09_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P09SC <- filter_taxa(P_09_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P09 <- merge_phyloseq(core_P09LH , core_P09RH, core_P09SC)

core_rm<- rownames(otu_table(CORE))
core_P09_rm <- taxa_names(core_P09)
core_P09_rm <- core_P09_rm[!(core_P09_rm %in% core_rm)]
core_P09_rm <- prune_taxa(core_P09_rm, core_P09)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P09_rm <- taxa_names(core_P09SC)
core_SC_P09_rm <- core_SC_P09_rm[!(core_SC_P09_rm %in% core_SC_rm)]
core_SC_P09_rm <- prune_taxa(core_SC_P09_rm, core_P09SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P09_hands <- merge_phyloseq(core_P09LH , core_P09RH)
core_HANDS_P09_rm <- taxa_names(core_P09_hands)
core_HANDS_P09_rm <- core_HANDS_P09_rm[!(core_HANDS_P09_rm %in% core_HANDS_rm)]
core_HANDS_P09_rm <- prune_taxa(core_HANDS_P09_rm, core_P09_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P09_rm <- taxa_names(core_P09LH)
core_LH_P09_rm <- core_LH_P09_rm[!(core_LH_P09_rm %in% core_LH_rm)]
core_LH_P09_rm <- prune_taxa(core_LH_P09_rm, core_P09LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P09_rm <- taxa_names(core_P09RH)
core_RH_P09_rm <- core_RH_P09_rm[!(core_RH_P09_rm %in% core_RH_rm)]
core_RH_P09_rm <- prune_taxa(core_RH_P09_rm, core_P09RH)
```
```{r}
norm_core_P09 <- transform_sample_counts(core_P09, function(x) x / sum(x) )
ord.core_09.pcoa <- ordinate(norm_core_P09, method="PCoA", distance="bray")
plot_ordination(norm_core_P09, ord.core_09.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P09_rm <- transform_sample_counts(core_P09_rm, function(x) x / sum(x) )
ord.core_09_rm.pcoa <- ordinate(norm_core_P09_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P09_rm, ord.core_09_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P10
```{r}
P_10 <- subset_samples(VCPHY1, Subject=="P10")
P_10_LH<- subset_samples(P_10, Location=="left")
P_10_RH<- subset_samples(P_10, Location=="right")
P_10_SC<- subset_samples(P_10, Location=="scalp")
core_P10LH <- filter_taxa(P_10_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P10RH <- filter_taxa(P_10_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P10SC <- filter_taxa(P_10_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P10 <- merge_phyloseq(core_P10LH , core_P10RH, core_P10SC)

core_rm<- rownames(otu_table(CORE))
core_P10_rm <- taxa_names(core_P10)
core_P10_rm <- core_P10_rm[!(core_P10_rm %in% core_rm)]
core_P10_rm <- prune_taxa(core_P10_rm, core_P10)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P10_rm <- taxa_names(core_P10SC)
core_SC_P10_rm <- core_SC_P10_rm[!(core_SC_P10_rm %in% core_SC_rm)]
core_SC_P10_rm <- prune_taxa(core_SC_P10_rm, core_P10SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P10_hands <- merge_phyloseq(core_P10LH , core_P10RH)
core_HANDS_P10_rm <- taxa_names(core_P10_hands)
core_HANDS_P10_rm <- core_HANDS_P10_rm[!(core_HANDS_P10_rm %in% core_HANDS_rm)]
core_HANDS_P10_rm <- prune_taxa(core_HANDS_P10_rm, core_P10_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P10_rm <- taxa_names(core_P10LH)
core_LH_P10_rm <- core_LH_P10_rm[!(core_LH_P10_rm %in% core_LH_rm)]
core_LH_P10_rm <- prune_taxa(core_LH_P10_rm, core_P10LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P10_rm <- taxa_names(core_P10RH)
core_RH_P10_rm <- core_RH_P10_rm[!(core_RH_P10_rm %in% core_RH_rm)]
core_RH_P10_rm <- prune_taxa(core_RH_P10_rm, core_P10RH)
```
```{r}
norm_core_P10 <- transform_sample_counts(core_P10, function(x) x / sum(x) )
ord.core_10.pcoa <- ordinate(norm_core_P10, method="PCoA", distance="bray")
plot_ordination(norm_core_P10, ord.core_10.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P10_rm <- transform_sample_counts(core_P10_rm, function(x) x / sum(x) )
ord.core_10_rm.pcoa <- ordinate(norm_core_P10_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P10_rm, ord.core_10_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P11
```{r}
P_11 <- subset_samples(VCPHY1, Subject=="P11")
P_11_LH<- subset_samples(P_11, Location=="left")
P_11_RH<- subset_samples(P_11, Location=="right")
P_11_SC<- subset_samples(P_11, Location=="scalp")
core_P11LH <- filter_taxa(P_11_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P11RH <- filter_taxa(P_11_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P11SC <- filter_taxa(P_11_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P11 <- merge_phyloseq(core_P11LH , core_P11RH, core_P11SC)

core_rm<- rownames(otu_table(CORE))
core_P11_rm <- taxa_names(core_P11)
core_P11_rm <- core_P11_rm[!(core_P11_rm %in% core_rm)]
core_P11_rm <- prune_taxa(core_P11_rm, core_P11)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P11_rm <- taxa_names(core_P11SC)
core_SC_P11_rm <- core_SC_P11_rm[!(core_SC_P11_rm %in% core_SC_rm)]
core_SC_P11_rm <- prune_taxa(core_SC_P11_rm, core_P11SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P11_hands <- merge_phyloseq(core_P11LH , core_P11RH)
core_HANDS_P11_rm <- taxa_names(core_P11_hands)
core_HANDS_P11_rm <- core_HANDS_P11_rm[!(core_HANDS_P11_rm %in% core_HANDS_rm)]
core_HANDS_P11_rm <- prune_taxa(core_HANDS_P11_rm, core_P11_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P11_rm <- taxa_names(core_P11LH)
core_LH_P11_rm <- core_LH_P11_rm[!(core_LH_P11_rm %in% core_LH_rm)]
core_LH_P11_rm <- prune_taxa(core_LH_P11_rm, core_P11LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P11_rm <- taxa_names(core_P11RH)
core_RH_P11_rm <- core_RH_P11_rm[!(core_RH_P11_rm %in% core_RH_rm)]
core_RH_P11_rm <- prune_taxa(core_RH_P11_rm, core_P11RH)
```
```{r}
norm_core_P11 <- transform_sample_counts(core_P11, function(x) x / sum(x) )
ord.core_11.pcoa <- ordinate(norm_core_P11, method="PCoA", distance="bray")
plot_ordination(norm_core_P11, ord.core_11.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P11_rm <- transform_sample_counts(core_P11_rm, function(x) x / sum(x) )
ord.core_11_rm.pcoa <- ordinate(norm_core_P11_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P11_rm, ord.core_11_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P12
```{r}
P_12 <- subset_samples(VCPHY1, Subject=="P12")
P_12_LH<- subset_samples(P_12, Location=="left")
P_12_RH<- subset_samples(P_12, Location=="right")
P_12_SC<- subset_samples(P_12, Location=="scalp")
core_P12LH <- filter_taxa(P_12_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P12RH <- filter_taxa(P_12_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P12SC <- filter_taxa(P_12_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P12 <- merge_phyloseq(core_P12LH , core_P12RH, core_P12SC)

core_rm<- rownames(otu_table(CORE))
core_P12_rm <- taxa_names(core_P12)
core_P12_rm <- core_P12_rm[!(core_P12_rm %in% core_rm)]
core_P12_rm <- prune_taxa(core_P12_rm, core_P12)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P12_rm <- taxa_names(core_P12SC)
core_SC_P12_rm <- core_SC_P12_rm[!(core_SC_P12_rm %in% core_SC_rm)]
core_SC_P12_rm <- prune_taxa(core_SC_P12_rm, core_P12SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P12_hands <- merge_phyloseq(core_P12LH , core_P12RH)
core_HANDS_P12_rm <- taxa_names(core_P12_hands)
core_HANDS_P12_rm <- core_HANDS_P12_rm[!(core_HANDS_P12_rm %in% core_HANDS_rm)]
core_HANDS_P12_rm <- prune_taxa(core_HANDS_P12_rm, core_P12_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P12_rm <- taxa_names(core_P12LH)
core_LH_P12_rm <- core_LH_P12_rm[!(core_LH_P12_rm %in% core_LH_rm)]
core_LH_P12_rm <- prune_taxa(core_LH_P12_rm, core_P12LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P12_rm <- taxa_names(core_P12RH)
core_RH_P12_rm <- core_RH_P12_rm[!(core_RH_P12_rm %in% core_RH_rm)]
core_RH_P12_rm <- prune_taxa(core_RH_P12_rm, core_P12RH)
```
```{r}
norm_core_P12 <- transform_sample_counts(core_P12, function(x) x / sum(x) )
ord.core_12.pcoa <- ordinate(norm_core_P12, method="PCoA", distance="bray")
plot_ordination(norm_core_P12, ord.core_12.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P12_rm <- transform_sample_counts(core_P12_rm, function(x) x / sum(x) )
ord.core_12_rm.pcoa <- ordinate(norm_core_P12_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P12_rm, ord.core_12_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P13
```{r}
P_13 <- subset_samples(VCPHY1, Subject=="P13")
P_13_LH<- subset_samples(P_13, Location=="left")
P_13_RH<- subset_samples(P_13, Location=="right")
P_13_SC<- subset_samples(P_13, Location=="scalp")
core_P13LH <- filter_taxa(P_13_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P13RH <- filter_taxa(P_13_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P13SC <- filter_taxa(P_13_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P13 <- merge_phyloseq(core_P13LH , core_P13RH, core_P13SC)

core_rm<- rownames(otu_table(CORE))
core_P13_rm <- taxa_names(core_P13)
core_P13_rm <- core_P13_rm[!(core_P13_rm %in% core_rm)]
core_P13_rm <- prune_taxa(core_P13_rm, core_P13)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P13_rm <- taxa_names(core_P13SC)
core_SC_P13_rm <- core_SC_P13_rm[!(core_SC_P13_rm %in% core_SC_rm)]
core_SC_P13_rm <- prune_taxa(core_SC_P13_rm, core_P13SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P13_hands <- merge_phyloseq(core_P13LH , core_P13RH)
core_HANDS_P13_rm <- taxa_names(core_P13_hands)
core_HANDS_P13_rm <- core_HANDS_P13_rm[!(core_HANDS_P13_rm %in% core_HANDS_rm)]
core_HANDS_P13_rm <- prune_taxa(core_HANDS_P13_rm, core_P13_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P13_rm <- taxa_names(core_P13LH)
core_LH_P13_rm <- core_LH_P13_rm[!(core_LH_P13_rm %in% core_LH_rm)]
core_LH_P13_rm <- prune_taxa(core_LH_P13_rm, core_P13LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P13_rm <- taxa_names(core_P13RH)
core_RH_P13_rm <- core_RH_P13_rm[!(core_RH_P13_rm %in% core_RH_rm)]
core_RH_P13_rm <- prune_taxa(core_RH_P13_rm, core_P13RH)
```
```{r}
norm_core_P13 <- transform_sample_counts(core_P13, function(x) x / sum(x) )
ord.core_13.pcoa <- ordinate(norm_core_P13, method="PCoA", distance="bray")
plot_ordination(norm_core_P13, ord.core_13.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P13_rm <- transform_sample_counts(core_P13_rm, function(x) x / sum(x) )
ord.core_13_rm.pcoa <- ordinate(norm_core_P13_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P13_rm, ord.core_13_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P14
```{r}
P_14 <- subset_samples(VCPHY1, Subject=="P14")
P_14_LH<- subset_samples(P_14, Location=="left")
P_14_RH<- subset_samples(P_14, Location=="right")
P_14_SC<- subset_samples(P_14, Location=="scalp")
core_P14LH <- filter_taxa(P_14_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P14RH <- filter_taxa(P_14_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P14SC <- filter_taxa(P_14_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P14 <- merge_phyloseq(core_P14LH , core_P14RH, core_P14SC)

core_rm<- rownames(otu_table(CORE))
core_P14_rm <- taxa_names(core_P14)
core_P14_rm <- core_P14_rm[!(core_P14_rm %in% core_rm)]
core_P14_rm <- prune_taxa(core_P14_rm, core_P14)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P14_rm <- taxa_names(core_P14SC)
core_SC_P14_rm <- core_SC_P14_rm[!(core_SC_P14_rm %in% core_SC_rm)]
core_SC_P14_rm <- prune_taxa(core_SC_P14_rm, core_P14SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P14_hands <- merge_phyloseq(core_P14LH , core_P14RH)
core_HANDS_P14_rm <- taxa_names(core_P14_hands)
core_HANDS_P14_rm <- core_HANDS_P14_rm[!(core_HANDS_P14_rm %in% core_HANDS_rm)]
core_HANDS_P14_rm <- prune_taxa(core_HANDS_P14_rm, core_P14_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P14_rm <- taxa_names(core_P14LH)
core_LH_P14_rm <- core_LH_P14_rm[!(core_LH_P14_rm %in% core_LH_rm)]
core_LH_P14_rm <- prune_taxa(core_LH_P14_rm, core_P14LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P14_rm <- taxa_names(core_P14RH)
core_RH_P14_rm <- core_RH_P14_rm[!(core_RH_P14_rm %in% core_RH_rm)]
core_RH_P14_rm <- prune_taxa(core_RH_P14_rm, core_P14RH)
```
```{r}
norm_core_P14 <- transform_sample_counts(core_P14, function(x) x / sum(x) )
ord.core_14.pcoa <- ordinate(norm_core_P14, method="PCoA", distance="bray")
plot_ordination(norm_core_P14, ord.core_14.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5) 
norm_core_P14_rm <- transform_sample_counts(core_P14_rm, function(x) x / sum(x) )
ord.core_14_rm.pcoa <- ordinate(norm_core_P14_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P14_rm, ord.core_14_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P15
```{r}
P_15 <- subset_samples(VCPHY1, Subject=="P15")
P_15_LH<- subset_samples(P_15, Location=="left")
P_15_RH<- subset_samples(P_15, Location=="right")
P_15_SC<- subset_samples(P_15, Location=="scalp")
core_P15LH <- filter_taxa(P_15_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P15RH <- filter_taxa(P_15_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P15SC <- filter_taxa(P_15_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P15 <- merge_phyloseq(core_P15LH , core_P15RH, core_P15SC)

core_rm<- rownames(otu_table(CORE))
core_P15_rm <- taxa_names(core_P15)
core_P15_rm <- core_P15_rm[!(core_P15_rm %in% core_rm)]
core_P15_rm <- prune_taxa(core_P15_rm, core_P15)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P15_rm <- taxa_names(core_P15SC)
core_SC_P15_rm <- core_SC_P15_rm[!(core_SC_P15_rm %in% core_SC_rm)]
core_SC_P15_rm <- prune_taxa(core_SC_P15_rm, core_P15SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P15_hands <- merge_phyloseq(core_P15LH , core_P15RH)
core_HANDS_P15_rm <- taxa_names(core_P15_hands)
core_HANDS_P15_rm <- core_HANDS_P15_rm[!(core_HANDS_P15_rm %in% core_HANDS_rm)]
core_HANDS_P15_rm <- prune_taxa(core_HANDS_P15_rm, core_P15_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P15_rm <- taxa_names(core_P15LH)
core_LH_P15_rm <- core_LH_P15_rm[!(core_LH_P15_rm %in% core_LH_rm)]
core_LH_P15_rm <- prune_taxa(core_LH_P15_rm, core_P15LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P15_rm <- taxa_names(core_P15RH)
core_RH_P15_rm <- core_RH_P15_rm[!(core_RH_P15_rm %in% core_RH_rm)]
core_RH_P15_rm <- prune_taxa(core_RH_P15_rm, core_P15RH)
```
```{r}
norm_core_P15 <- transform_sample_counts(core_P15, function(x) x / sum(x) )
ord.core_15.pcoa <- ordinate(norm_core_P15, method="PCoA", distance="bray")
plot_ordination(norm_core_P15, ord.core_15.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P15_rm <- transform_sample_counts(core_P15_rm, function(x) x / sum(x) )
ord.core_15_rm.pcoa <- ordinate(norm_core_P15_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P15_rm, ord.core_15_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P16
```{r}
P_16 <- subset_samples(VCPHY1, Subject=="P16")
P_16_LH<- subset_samples(P_16, Location=="left")
P_16_RH<- subset_samples(P_16, Location=="right")
P_16_SC<- subset_samples(P_16, Location=="scalp")
core_P16LH <- filter_taxa(P_16_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P16RH <- filter_taxa(P_16_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P16SC <- filter_taxa(P_16_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P16 <- merge_phyloseq(core_P16LH , core_P16RH, core_P16SC)

core_rm<- rownames(otu_table(CORE))
core_P16_rm <- taxa_names(core_P16)
core_P16_rm <- core_P16_rm[!(core_P16_rm %in% core_rm)]
core_P16_rm <- prune_taxa(core_P16_rm, core_P16)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P16_rm <- taxa_names(core_P16SC)
core_SC_P16_rm <- core_SC_P16_rm[!(core_SC_P16_rm %in% core_SC_rm)]
core_SC_P16_rm <- prune_taxa(core_SC_P16_rm, core_P16SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P16_hands <- merge_phyloseq(core_P16LH , core_P16RH)
core_HANDS_P16_rm <- taxa_names(core_P16_hands)
core_HANDS_P16_rm <- core_HANDS_P16_rm[!(core_HANDS_P16_rm %in% core_HANDS_rm)]
core_HANDS_P16_rm <- prune_taxa(core_HANDS_P16_rm, core_P16_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P16_rm <- taxa_names(core_P16LH)
core_LH_P16_rm <- core_LH_P16_rm[!(core_LH_P16_rm %in% core_LH_rm)]
core_LH_P16_rm <- prune_taxa(core_LH_P16_rm, core_P16LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P16_rm <- taxa_names(core_P16RH)
core_RH_P16_rm <- core_RH_P16_rm[!(core_RH_P16_rm %in% core_RH_rm)]
core_RH_P16_rm <- prune_taxa(core_RH_P16_rm, core_P16RH)
```
```{r}
norm_core_P16 <- transform_sample_counts(core_P16, function(x) x / sum(x) )
ord.core_16.pcoa <- ordinate(norm_core_P16, method="PCoA", distance="bray")
plot_ordination(norm_core_P16, ord.core_16.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P16_rm <- transform_sample_counts(core_P16_rm, function(x) x / sum(x) )
ord.core_16_rm.pcoa <- ordinate(norm_core_P16_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P16_rm, ord.core_16_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)


```

#P17
```{r}
P_17 <- subset_samples(VCPHY1, Subject=="P17")
P_17_LH<- subset_samples(P_17, Location=="left")
P_17_RH<- subset_samples(P_17, Location=="right")
P_17_SC<- subset_samples(P_17, Location=="scalp")
core_P17LH <- filter_taxa(P_17_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P17RH <- filter_taxa(P_17_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P17SC <- filter_taxa(P_17_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P17 <- merge_phyloseq(core_P17LH , core_P17RH, core_P17SC)

core_rm<- rownames(otu_table(CORE))
core_P17_rm <- taxa_names(core_P17)
core_P17_rm <- core_P17_rm[!(core_P17_rm %in% core_rm)]
core_P17_rm <- prune_taxa(core_P17_rm, core_P17)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P17_rm <- taxa_names(core_P17SC)
core_SC_P17_rm <- core_SC_P17_rm[!(core_SC_P17_rm %in% core_SC_rm)]
core_SC_P17_rm <- prune_taxa(core_SC_P17_rm, core_P17SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P17_hands <- merge_phyloseq(core_P17LH , core_P17RH)
core_HANDS_P17_rm <- taxa_names(core_P17_hands)
core_HANDS_P17_rm <- core_HANDS_P17_rm[!(core_HANDS_P17_rm %in% core_HANDS_rm)]
core_HANDS_P17_rm <- prune_taxa(core_HANDS_P17_rm, core_P17_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P17_rm <- taxa_names(core_P17LH)
core_LH_P17_rm <- core_LH_P17_rm[!(core_LH_P17_rm %in% core_LH_rm)]
core_LH_P17_rm <- prune_taxa(core_LH_P17_rm, core_P17LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P17_rm <- taxa_names(core_P17RH)
core_RH_P17_rm <- core_RH_P17_rm[!(core_RH_P17_rm %in% core_RH_rm)]
core_RH_P17_rm <- prune_taxa(core_RH_P17_rm, core_P17RH)
```
```{r}
norm_core_P17 <- transform_sample_counts(core_P17, function(x) x / sum(x) )
ord.core_17.pcoa <- ordinate(norm_core_P17, method="PCoA", distance="bray")
plot_ordination(norm_core_P17, ord.core_17.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P17_rm <- transform_sample_counts(core_P17_rm, function(x) x / sum(x) )
ord.core_17_rm.pcoa <- ordinate(norm_core_P17_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P17_rm, ord.core_17_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P18
```{r}
P_18 <- subset_samples(VCPHY1, Subject=="P18")
P_18_LH<- subset_samples(P_18, Location=="left")
P_18_RH<- subset_samples(P_18, Location=="right")
P_18_SC<- subset_samples(P_18, Location=="scalp")
core_P18LH <- filter_taxa(P_18_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P18RH <- filter_taxa(P_18_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P18SC <- filter_taxa(P_18_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P18 <- merge_phyloseq(core_P18LH , core_P18RH, core_P18SC)

core_rm<- rownames(otu_table(CORE))
core_P18_rm <- taxa_names(core_P18)
core_P18_rm <- core_P18_rm[!(core_P18_rm %in% core_rm)]
core_P18_rm <- prune_taxa(core_P18_rm, core_P18)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P18_rm <- taxa_names(core_P18SC)
core_SC_P18_rm <- core_SC_P18_rm[!(core_SC_P18_rm %in% core_SC_rm)]
core_SC_P18_rm <- prune_taxa(core_SC_P18_rm, core_P18SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P18_hands <- merge_phyloseq(core_P18LH , core_P18RH)
core_HANDS_P18_rm <- taxa_names(core_P18_hands)
core_HANDS_P18_rm <- core_HANDS_P18_rm[!(core_HANDS_P18_rm %in% core_HANDS_rm)]
core_HANDS_P18_rm <- prune_taxa(core_HANDS_P18_rm, core_P18_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P18_rm <- taxa_names(core_P18LH)
core_LH_P18_rm <- core_LH_P18_rm[!(core_LH_P18_rm %in% core_LH_rm)]
core_LH_P18_rm <- prune_taxa(core_LH_P18_rm, core_P18LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P18_rm <- taxa_names(core_P18RH)
core_RH_P18_rm <- core_RH_P18_rm[!(core_RH_P18_rm %in% core_RH_rm)]
core_RH_P18_rm <- prune_taxa(core_RH_P18_rm, core_P18RH)
```
```{r}
norm_core_P18 <- transform_sample_counts(core_P18, function(x) x / sum(x) )
ord.core_18.pcoa <- ordinate(norm_core_P18, method="PCoA", distance="bray")
plot_ordination(norm_core_P18, ord.core_18.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P18_rm <- transform_sample_counts(core_P18_rm, function(x) x / sum(x) )
ord.core_18_rm.pcoa <- ordinate(norm_core_P18_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P18_rm, ord.core_18_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P19
```{r}
P_19 <- subset_samples(VCPHY1, Subject=="P19")
P_19_LH<- subset_samples(P_19, Location=="left")
P_19_RH<- subset_samples(P_19, Location=="right")
P_19_SC<- subset_samples(P_19, Location=="scalp")
core_P19LH <- filter_taxa(P_19_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P19RH <- filter_taxa(P_19_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P19SC <- filter_taxa(P_19_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P19 <- merge_phyloseq(core_P19LH , core_P19RH, core_P19SC)

core_rm<- rownames(otu_table(CORE))
core_P19_rm <- taxa_names(core_P19)
core_P19_rm <- core_P19_rm[!(core_P19_rm %in% core_rm)]
core_P19_rm <- prune_taxa(core_P19_rm, core_P19)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P19_rm <- taxa_names(core_P19SC)
core_SC_P19_rm <- core_SC_P19_rm[!(core_SC_P19_rm %in% core_SC_rm)]
core_SC_P19_rm <- prune_taxa(core_SC_P19_rm, core_P19SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P19_hands <- merge_phyloseq(core_P19LH , core_P19RH)
core_HANDS_P19_rm <- taxa_names(core_P19_hands)
core_HANDS_P19_rm <- core_HANDS_P19_rm[!(core_HANDS_P19_rm %in% core_HANDS_rm)]
core_HANDS_P19_rm <- prune_taxa(core_HANDS_P19_rm, core_P19_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P19_rm <- taxa_names(core_P19LH)
core_LH_P19_rm <- core_LH_P19_rm[!(core_LH_P19_rm %in% core_LH_rm)]
core_LH_P19_rm <- prune_taxa(core_LH_P19_rm, core_P19LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P19_rm <- taxa_names(core_P19RH)
core_RH_P19_rm <- core_RH_P19_rm[!(core_RH_P19_rm %in% core_RH_rm)]
core_RH_P19_rm <- prune_taxa(core_RH_P19_rm, core_P19RH)
```
```{r}
norm_core_P19 <- transform_sample_counts(core_P19, function(x) x / sum(x) )
ord.core_19.pcoa <- ordinate(norm_core_P19, method="PCoA", distance="bray")
plot_ordination(norm_core_P19, ord.core_19.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P19_rm <- transform_sample_counts(core_P19_rm, function(x) x / sum(x) )
ord.core_19_rm.pcoa <- ordinate(norm_core_P19_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P19_rm, ord.core_19_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P20
```{r}
P_20 <- subset_samples(VCPHY1, Subject=="P20")
P_20_LH<- subset_samples(P_20, Location=="left")
P_20_RH<- subset_samples(P_20, Location=="right")
P_20_SC<- subset_samples(P_20, Location=="scalp")
core_P20LH <- filter_taxa(P_20_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P20RH <- filter_taxa(P_20_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P20SC <- filter_taxa(P_20_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P20 <- merge_phyloseq(core_P20LH , core_P20RH, core_P20SC)

core_rm<- rownames(otu_table(CORE))
core_P20_rm <- taxa_names(core_P20)
core_P20_rm <- core_P20_rm[!(core_P20_rm %in% core_rm)]
core_P20_rm <- prune_taxa(core_P20_rm, core_P20)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P20_rm <- taxa_names(core_P20SC)
core_SC_P20_rm <- core_SC_P20_rm[!(core_SC_P20_rm %in% core_SC_rm)]
core_SC_P20_rm <- prune_taxa(core_SC_P20_rm, core_P20SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P20_hands <- merge_phyloseq(core_P20LH , core_P20RH)
core_HANDS_P20_rm <- taxa_names(core_P20_hands)
core_HANDS_P20_rm <- core_HANDS_P20_rm[!(core_HANDS_P20_rm %in% core_HANDS_rm)]
core_HANDS_P20_rm <- prune_taxa(core_HANDS_P20_rm, core_P20_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P20_rm <- taxa_names(core_P20LH)
core_LH_P20_rm <- core_LH_P20_rm[!(core_LH_P20_rm %in% core_LH_rm)]
core_LH_P20_rm <- prune_taxa(core_LH_P20_rm, core_P20LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P20_rm <- taxa_names(core_P20RH)
core_RH_P20_rm <- core_RH_P20_rm[!(core_RH_P20_rm %in% core_RH_rm)]
core_RH_P20_rm <- prune_taxa(core_RH_P20_rm, core_P20RH)
```
```{r}
norm_core_P20 <- transform_sample_counts(core_P20, function(x) x / sum(x) )
ord.core_20.pcoa <- ordinate(norm_core_P20, method="PCoA", distance="bray")
plot_ordination(norm_core_P20, ord.core_20.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P20_rm <- transform_sample_counts(core_P20_rm, function(x) x / sum(x) )
ord.core_20_rm.pcoa <- ordinate(norm_core_P20_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P20_rm, ord.core_20_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P21
```{r}
P_21 <- subset_samples(VCPHY1, Subject=="P21")
P_21_LH<- subset_samples(P_21, Location=="left")
P_21_RH<- subset_samples(P_21, Location=="right")
P_21_SC<- subset_samples(P_21, Location=="scalp")
core_P21LH <- filter_taxa(P_21_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P21RH <- filter_taxa(P_21_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P21SC <- filter_taxa(P_21_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P21 <- merge_phyloseq(core_P21LH , core_P21RH, core_P21SC)

core_rm<- rownames(otu_table(CORE))
core_P21_rm <- taxa_names(core_P21)
core_P21_rm <- core_P21_rm[!(core_P21_rm %in% core_rm)]
core_P21_rm <- prune_taxa(core_P21_rm, core_P21)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P21_rm <- taxa_names(core_P21SC)
core_SC_P21_rm <- core_SC_P21_rm[!(core_SC_P21_rm %in% core_SC_rm)]
core_SC_P21_rm <- prune_taxa(core_SC_P21_rm, core_P21SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P21_hands <- merge_phyloseq(core_P21LH , core_P21RH)
core_HANDS_P21_rm <- taxa_names(core_P21_hands)
core_HANDS_P21_rm <- core_HANDS_P21_rm[!(core_HANDS_P21_rm %in% core_HANDS_rm)]
core_HANDS_P21_rm <- prune_taxa(core_HANDS_P21_rm, core_P21_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P21_rm <- taxa_names(core_P21LH)
core_LH_P21_rm <- core_LH_P21_rm[!(core_LH_P21_rm %in% core_LH_rm)]
core_LH_P21_rm <- prune_taxa(core_LH_P21_rm, core_P21LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P21_rm <- taxa_names(core_P21RH)
core_RH_P21_rm <- core_RH_P21_rm[!(core_RH_P21_rm %in% core_RH_rm)]
core_RH_P21_rm <- prune_taxa(core_RH_P21_rm, core_P21RH)
```
```{r}
norm_core_P21 <- transform_sample_counts(core_P21, function(x) x / sum(x) )
ord.core_21.pcoa <- ordinate(norm_core_P21, method="PCoA", distance="bray")
plot_ordination(norm_core_P21, ord.core_21.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P21_rm <- transform_sample_counts(core_P21_rm, function(x) x / sum(x) )
ord.core_21_rm.pcoa <- ordinate(norm_core_P21_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P21_rm, ord.core_21_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```


#P22
```{r}
P_22 <- subset_samples(VCPHY1, Subject=="P22")
P_22_LH<- subset_samples(P_22, Location=="left")
P_22_RH<- subset_samples(P_22, Location=="right")
P_22_SC<- subset_samples(P_22, Location=="scalp")
core_P22LH <- filter_taxa(P_22_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P22RH <- filter_taxa(P_22_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P22SC <- filter_taxa(P_22_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P22 <- merge_phyloseq(core_P22LH , core_P22RH, core_P22SC)

core_rm<- rownames(otu_table(CORE))
core_P22_rm <- taxa_names(core_P22)
core_P22_rm <- core_P22_rm[!(core_P22_rm %in% core_rm)]
core_P22_rm <- prune_taxa(core_P22_rm, core_P22)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P22_rm <- taxa_names(core_P22SC)
core_SC_P22_rm <- core_SC_P22_rm[!(core_SC_P22_rm %in% core_SC_rm)]
core_SC_P22_rm <- prune_taxa(core_SC_P22_rm, core_P22SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P22_hands <- merge_phyloseq(core_P22LH , core_P22RH)
core_HANDS_P22_rm <- taxa_names(core_P22_hands)
core_HANDS_P22_rm <- core_HANDS_P22_rm[!(core_HANDS_P22_rm %in% core_HANDS_rm)]
core_HANDS_P22_rm <- prune_taxa(core_HANDS_P22_rm, core_P22_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P22_rm <- taxa_names(core_P22LH)
core_LH_P22_rm <- core_LH_P22_rm[!(core_LH_P22_rm %in% core_LH_rm)]
core_LH_P22_rm <- prune_taxa(core_LH_P22_rm, core_P22LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P22_rm <- taxa_names(core_P22RH)
core_RH_P22_rm <- core_RH_P22_rm[!(core_RH_P22_rm %in% core_RH_rm)]
core_RH_P22_rm <- prune_taxa(core_RH_P22_rm, core_P22RH)
```
```{r}
norm_core_P22 <- transform_sample_counts(core_P22, function(x) x / sum(x) )
ord.core_22.pcoa <- ordinate(norm_core_P22, method="PCoA", distance="bray")
plot_ordination(norm_core_P22, ord.core_22.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P22_rm <- transform_sample_counts(core_P22_rm, function(x) x / sum(x) )
ord.core_22_rm.pcoa <- ordinate(norm_core_P22_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P22_rm, ord.core_22_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P23
```{r}
P_23 <- subset_samples(VCPHY1, Subject=="P23")
P_23_LH<- subset_samples(P_23, Location=="left")
P_23_RH<- subset_samples(P_23, Location=="right")
P_23_SC<- subset_samples(P_23, Location=="scalp")
core_P23LH <- filter_taxa(P_23_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P23RH <- filter_taxa(P_23_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P23SC <- filter_taxa(P_23_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P23 <- merge_phyloseq(core_P23LH , core_P23RH, core_P23SC)

core_rm<- rownames(otu_table(CORE))
core_P23_rm <- taxa_names(core_P23)
core_P23_rm <- core_P23_rm[!(core_P23_rm %in% core_rm)]
core_P23_rm <- prune_taxa(core_P23_rm, core_P23)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P23_rm <- taxa_names(core_P23SC)
core_SC_P23_rm <- core_SC_P23_rm[!(core_SC_P23_rm %in% core_SC_rm)]
core_SC_P23_rm <- prune_taxa(core_SC_P23_rm, core_P23SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P23_hands <- merge_phyloseq(core_P23LH , core_P23RH)
core_HANDS_P23_rm <- taxa_names(core_P23_hands)
core_HANDS_P23_rm <- core_HANDS_P23_rm[!(core_HANDS_P23_rm %in% core_HANDS_rm)]
core_HANDS_P23_rm <- prune_taxa(core_HANDS_P23_rm, core_P23_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P23_rm <- taxa_names(core_P23LH)
core_LH_P23_rm <- core_LH_P23_rm[!(core_LH_P23_rm %in% core_LH_rm)]
core_LH_P23_rm <- prune_taxa(core_LH_P23_rm, core_P23LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P23_rm <- taxa_names(core_P23RH)
core_RH_P23_rm <- core_RH_P23_rm[!(core_RH_P23_rm %in% core_RH_rm)]
core_RH_P23_rm <- prune_taxa(core_RH_P23_rm, core_P23RH)
```
```{r}
norm_core_P23 <- transform_sample_counts(core_P23, function(x) x / sum(x) )
ord.core_23.pcoa <- ordinate(norm_core_P23, method="PCoA", distance="bray")
plot_ordination(norm_core_P23, ord.core_23.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P23_rm <- transform_sample_counts(core_P23_rm, function(x) x / sum(x) )
ord.core_23_rm.pcoa <- ordinate(norm_core_P23_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P23_rm, ord.core_23_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P24
```{r}
P_24 <- subset_samples(VCPHY1, Subject=="P24")
P_24_LH<- subset_samples(P_24, Location=="left")
P_24_RH<- subset_samples(P_24, Location=="right")
P_24_SC<- subset_samples(P_24, Location=="scalp")
core_P24LH <- filter_taxa(P_24_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P24RH <- filter_taxa(P_24_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P24SC <- filter_taxa(P_24_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P24 <- merge_phyloseq(core_P24LH , core_P24RH, core_P24SC)
norm_core_P24 <- transform_sample_counts(core_P24, function(x) x / sum(x) )

core_rm<- rownames(otu_table(CORE))
core_P24_rm <- taxa_names(core_P24)
core_P24_rm <- core_P24_rm[!(core_P24_rm %in% core_rm)]
core_P24_rm <- prune_taxa(core_P24_rm, core_P24)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P24_rm <- taxa_names(core_P24SC)
core_SC_P24_rm <- core_SC_P24_rm[!(core_SC_P24_rm %in% core_SC_rm)]
core_SC_P24_rm <- prune_taxa(core_SC_P24_rm, core_P24SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P24_hands <- merge_phyloseq(core_P24LH , core_P24RH)
core_HANDS_P24_rm <- taxa_names(core_P24_hands)
core_HANDS_P24_rm <- core_HANDS_P24_rm[!(core_HANDS_P24_rm %in% core_HANDS_rm)]
core_HANDS_P24_rm <- prune_taxa(core_HANDS_P24_rm, core_P24_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P24_rm <- taxa_names(core_P24LH)
core_LH_P24_rm <- core_LH_P24_rm[!(core_LH_P24_rm %in% core_LH_rm)]
core_LH_P24_rm <- prune_taxa(core_LH_P24_rm, core_P24LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P24_rm <- taxa_names(core_P24RH)
core_RH_P24_rm <- core_RH_P24_rm[!(core_RH_P24_rm %in% core_RH_rm)]
core_RH_P24_rm <- prune_taxa(core_RH_P24_rm, core_P24RH)
```
```{r}
ord.core_24.pcoa <- ordinate(norm_core_P24, method="PCoA", distance="bray")
plot_ordination(norm_core_P24, ord.core_24.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P24_rm <- transform_sample_counts(core_P24_rm, function(x) x / sum(x) )
ord.core_24_rm.pcoa <- ordinate(norm_core_P24_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P24_rm, ord.core_24_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P25
```{r}
P_25 <- subset_samples(VCPHY1, Subject=="P25")
P_25_LH<- subset_samples(P_25, Location=="left")
P_25_RH<- subset_samples(P_25, Location=="right")
P_25_SC<- subset_samples(P_25, Location=="scalp")
core_P25LH <- filter_taxa(P_25_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P25RH <- filter_taxa(P_25_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P25SC <- filter_taxa(P_25_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P25 <- merge_phyloseq(core_P25LH , core_P25RH, core_P25SC)

core_rm<- rownames(otu_table(CORE))
core_P25_rm <- taxa_names(core_P25)
core_P25_rm <- core_P25_rm[!(core_P25_rm %in% core_rm)]
core_P25_rm <- prune_taxa(core_P25_rm, core_P25)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P25_rm <- taxa_names(core_P25SC)
core_SC_P25_rm <- core_SC_P25_rm[!(core_SC_P25_rm %in% core_SC_rm)]
core_SC_P25_rm <- prune_taxa(core_SC_P25_rm, core_P25SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P25_hands <- merge_phyloseq(core_P25LH , core_P25RH)
core_HANDS_P25_rm <- taxa_names(core_P25_hands)
core_HANDS_P25_rm <- core_HANDS_P25_rm[!(core_HANDS_P25_rm %in% core_HANDS_rm)]
core_HANDS_P25_rm <- prune_taxa(core_HANDS_P25_rm, core_P25_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P25_rm <- taxa_names(core_P25LH)
core_LH_P25_rm <- core_LH_P25_rm[!(core_LH_P25_rm %in% core_LH_rm)]
core_LH_P25_rm <- prune_taxa(core_LH_P25_rm, core_P25LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P25_rm <- taxa_names(core_P25RH)
core_RH_P25_rm <- core_RH_P25_rm[!(core_RH_P25_rm %in% core_RH_rm)]
core_RH_P25_rm <- prune_taxa(core_RH_P25_rm, core_P25RH)
```
```{r}
norm_core_P25 <- transform_sample_counts(core_P25, function(x) x / sum(x) )
ord.core_25.pcoa <- ordinate(norm_core_P25, method="PCoA", distance="bray")
plot_ordination(norm_core_P25, ord.core_25.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P25_rm <- transform_sample_counts(core_P25_rm, function(x) x / sum(x) )
ord.core_25_rm.pcoa <- ordinate(norm_core_P25_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P25_rm, ord.core_25_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P26
```{r}
P_26 <- subset_samples(VCPHY1, Subject=="P26")
P_26_LH<- subset_samples(P_26, Location=="left")
P_26_RH<- subset_samples(P_26, Location=="right")
P_26_SC<- subset_samples(P_26, Location=="scalp")
core_P26LH <- filter_taxa(P_26_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P26RH <- filter_taxa(P_26_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P26SC <- filter_taxa(P_26_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P26 <- merge_phyloseq(core_P26LH , core_P26RH, core_P26SC)

core_rm<- rownames(otu_table(CORE))
core_P26_rm <- taxa_names(core_P26)
core_P26_rm <- core_P26_rm[!(core_P26_rm %in% core_rm)]
core_P26_rm <- prune_taxa(core_P26_rm, core_P26)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P26_rm <- taxa_names(core_P26SC)
core_SC_P26_rm <- core_SC_P26_rm[!(core_SC_P26_rm %in% core_SC_rm)]
core_SC_P26_rm <- prune_taxa(core_SC_P26_rm, core_P26SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P26_hands <- merge_phyloseq(core_P26LH , core_P26RH)
core_HANDS_P26_rm <- taxa_names(core_P26_hands)
core_HANDS_P26_rm <- core_HANDS_P26_rm[!(core_HANDS_P26_rm %in% core_HANDS_rm)]
core_HANDS_P26_rm <- prune_taxa(core_HANDS_P26_rm, core_P26_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P26_rm <- taxa_names(core_P26LH)
core_LH_P26_rm <- core_LH_P26_rm[!(core_LH_P26_rm %in% core_LH_rm)]
core_LH_P26_rm <- prune_taxa(core_LH_P26_rm, core_P26LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P26_rm <- taxa_names(core_P26RH)
core_RH_P26_rm <- core_RH_P26_rm[!(core_RH_P26_rm %in% core_RH_rm)]
core_RH_P26_rm <- prune_taxa(core_RH_P26_rm, core_P26RH)
```
```{r}
norm_core_P26 <- transform_sample_counts(core_P26, function(x) x / sum(x) )
ord.core_26.pcoa <- ordinate(norm_core_P26, method="PCoA", distance="bray")
plot_ordination(norm_core_P26, ord.core_26.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P26_rm <- transform_sample_counts(core_P26_rm, function(x) x / sum(x) )
ord.core_26_rm.pcoa <- ordinate(norm_core_P26_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P26_rm, ord.core_26_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P27
```{r}
P_27 <- subset_samples(VCPHY1, Subject=="P27")
P_27_LH<- subset_samples(P_27, Location=="left")
P_27_RH<- subset_samples(P_27, Location=="right")
P_27_SC<- subset_samples(P_27, Location=="scalp")
core_P27LH <- filter_taxa(P_27_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P27RH <- filter_taxa(P_27_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P27SC <- filter_taxa(P_27_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P27 <- merge_phyloseq(core_P27LH , core_P27RH, core_P27SC)

core_rm<- rownames(otu_table(CORE))
core_P27_rm <- taxa_names(core_P27)
core_P27_rm <- core_P27_rm[!(core_P27_rm %in% core_rm)]
core_P27_rm <- prune_taxa(core_P27_rm, core_P27)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P27_rm <- taxa_names(core_P27SC)
core_SC_P27_rm <- core_SC_P27_rm[!(core_SC_P27_rm %in% core_SC_rm)]
core_SC_P27_rm <- prune_taxa(core_SC_P27_rm, core_P27SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P27_hands <- merge_phyloseq(core_P27LH , core_P27RH)
core_HANDS_P27_rm <- taxa_names(core_P27_hands)
core_HANDS_P27_rm <- core_HANDS_P27_rm[!(core_HANDS_P27_rm %in% core_HANDS_rm)]
core_HANDS_P27_rm <- prune_taxa(core_HANDS_P27_rm, core_P27_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P27_rm <- taxa_names(core_P27LH)
core_LH_P27_rm <- core_LH_P27_rm[!(core_LH_P27_rm %in% core_LH_rm)]
core_LH_P27_rm <- prune_taxa(core_LH_P27_rm, core_P27LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P27_rm <- taxa_names(core_P27RH)
core_RH_P27_rm <- core_RH_P27_rm[!(core_RH_P27_rm %in% core_RH_rm)]
core_RH_P27_rm <- prune_taxa(core_RH_P27_rm, core_P27RH)
```
```{r}
norm_core_P27 <- transform_sample_counts(core_P27, function(x) x / sum(x) )
ord.core_27.pcoa <- ordinate(norm_core_P27, method="PCoA", distance="bray")
plot_ordination(norm_core_P27, ord.core_27.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P27_rm <- transform_sample_counts(core_P27_rm, function(x) x / sum(x) )
ord.core_27_rm.pcoa <- ordinate(norm_core_P27_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P27_rm, ord.core_27_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P28
```{r}
P_28 <- subset_samples(VCPHY1, Subject=="P28")
P_28_LH<- subset_samples(P_28, Location=="left")
P_28_RH<- subset_samples(P_28, Location=="right")
P_28_SC<- subset_samples(P_28, Location=="scalp")
core_P28LH <- filter_taxa(P_28_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P28RH <- filter_taxa(P_28_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P28SC <- filter_taxa(P_28_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P28 <- merge_phyloseq(core_P28LH , core_P28RH, core_P28SC)

core_rm<- rownames(otu_table(CORE))
core_P28_rm <- taxa_names(core_P28)
core_P28_rm <- core_P28_rm[!(core_P28_rm %in% core_rm)]
core_P28_rm <- prune_taxa(core_P28_rm, core_P28)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P28_rm <- taxa_names(core_P28SC)
core_SC_P28_rm <- core_SC_P28_rm[!(core_SC_P28_rm %in% core_SC_rm)]
core_SC_P28_rm <- prune_taxa(core_SC_P28_rm, core_P28SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P28_hands <- merge_phyloseq(core_P28LH , core_P28RH)
core_HANDS_P28_rm <- taxa_names(core_P28_hands)
core_HANDS_P28_rm <- core_HANDS_P28_rm[!(core_HANDS_P28_rm %in% core_HANDS_rm)]
core_HANDS_P28_rm <- prune_taxa(core_HANDS_P28_rm, core_P28_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P28_rm <- taxa_names(core_P28LH)
core_LH_P28_rm <- core_LH_P28_rm[!(core_LH_P28_rm %in% core_LH_rm)]
core_LH_P28_rm <- prune_taxa(core_LH_P28_rm, core_P28LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P28_rm <- taxa_names(core_P28RH)
core_RH_P28_rm <- core_RH_P28_rm[!(core_RH_P28_rm %in% core_RH_rm)]
core_RH_P28_rm <- prune_taxa(core_RH_P28_rm, core_P28RH)
```
```{r}
norm_core_P28 <- transform_sample_counts(core_P28, function(x) x / sum(x) )
ord.core_28.pcoa <- ordinate(norm_core_P28, method="PCoA", distance="bray")
plot_ordination(norm_core_P28, ord.core_28.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P28_rm <- transform_sample_counts(core_P28_rm, function(x) x / sum(x) )
ord.core_28_rm.pcoa <- ordinate(norm_core_P28_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P28_rm, ord.core_28_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P29
```{r}
P_29 <- subset_samples(VCPHY1, Subject=="P29")
P_29_LH<- subset_samples(P_29, Location=="left")
P_29_RH<- subset_samples(P_29, Location=="right")
P_29_SC<- subset_samples(P_29, Location=="scalp")
core_P29LH <- filter_taxa(P_29_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P29RH <- filter_taxa(P_29_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P29SC <- filter_taxa(P_29_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P29 <- merge_phyloseq(core_P29LH , core_P29RH, core_P29SC)

core_rm<- rownames(otu_table(CORE))
core_P29_rm <- taxa_names(core_P29)
core_P29_rm <- core_P29_rm[!(core_P29_rm %in% core_rm)]
core_P29_rm <- prune_taxa(core_P29_rm, core_P29)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P29_rm <- taxa_names(core_P29SC)
core_SC_P29_rm <- core_SC_P29_rm[!(core_SC_P29_rm %in% core_SC_rm)]
core_SC_P29_rm <- prune_taxa(core_SC_P29_rm, core_P29SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P29_hands <- merge_phyloseq(core_P29LH , core_P29RH)
core_HANDS_P29_rm <- taxa_names(core_P29_hands)
core_HANDS_P29_rm <- core_HANDS_P29_rm[!(core_HANDS_P29_rm %in% core_HANDS_rm)]
core_HANDS_P29_rm <- prune_taxa(core_HANDS_P29_rm, core_P29_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P29_rm <- taxa_names(core_P29LH)
core_LH_P29_rm <- core_LH_P29_rm[!(core_LH_P29_rm %in% core_LH_rm)]
core_LH_P29_rm <- prune_taxa(core_LH_P29_rm, core_P29LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P29_rm <- taxa_names(core_P29RH)
core_RH_P29_rm <- core_RH_P29_rm[!(core_RH_P29_rm %in% core_RH_rm)]
core_RH_P29_rm <- prune_taxa(core_RH_P29_rm, core_P29RH)
```
```{r}
norm_core_P29 <- transform_sample_counts(core_P29, function(x) x / sum(x) )
ord.core_29.pcoa <- ordinate(norm_core_P29, method="PCoA", distance="bray")
plot_ordination(norm_core_P29, ord.core_29.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P29_rm <- transform_sample_counts(core_P29_rm, function(x) x / sum(x) )
ord.core_29_rm.pcoa <- ordinate(norm_core_P29_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P29_rm, ord.core_29_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P30
```{r}
P_30 <- subset_samples(VCPHY1, Subject=="P30")
P_30_LH<- subset_samples(P_30, Location=="left")
P_30_RH<- subset_samples(P_30, Location=="right")
P_30_SC<- subset_samples(P_30, Location=="scalp")
core_P30LH <- filter_taxa(P_30_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P30RH <- filter_taxa(P_30_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P30SC <- filter_taxa(P_30_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P30 <- merge_phyloseq(core_P30LH , core_P30RH, core_P30SC)

core_rm<- rownames(otu_table(CORE))
core_P30_rm <- taxa_names(core_P30)
core_P30_rm <- core_P30_rm[!(core_P30_rm %in% core_rm)]
core_P30_rm <- prune_taxa(core_P30_rm, core_P30)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P30_rm <- taxa_names(core_P30SC)
core_SC_P30_rm <- core_SC_P30_rm[!(core_SC_P30_rm %in% core_SC_rm)]
core_SC_P30_rm <- prune_taxa(core_SC_P30_rm, core_P30SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P30_hands <- merge_phyloseq(core_P30LH , core_P30RH)
core_HANDS_P30_rm <- taxa_names(core_P30_hands)
core_HANDS_P30_rm <- core_HANDS_P30_rm[!(core_HANDS_P30_rm %in% core_HANDS_rm)]
core_HANDS_P30_rm <- prune_taxa(core_HANDS_P30_rm, core_P30_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P30_rm <- taxa_names(core_P30LH)
core_LH_P30_rm <- core_LH_P30_rm[!(core_LH_P30_rm %in% core_LH_rm)]
core_LH_P30_rm <- prune_taxa(core_LH_P30_rm, core_P30LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P30_rm <- taxa_names(core_P30RH)
core_RH_P30_rm <- core_RH_P30_rm[!(core_RH_P30_rm %in% core_RH_rm)]
core_RH_P30_rm <- prune_taxa(core_RH_P30_rm, core_P30RH)
```
```{r}
norm_core_P30 <- transform_sample_counts(core_P30, function(x) x / sum(x) )
ord.core_30.pcoa <- ordinate(norm_core_P30, method="PCoA", distance="bray")
plot_ordination(norm_core_P30, ord.core_30.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P30_rm <- transform_sample_counts(core_P30_rm, function(x) x / sum(x) )
ord.core_30_rm.pcoa <- ordinate(norm_core_P30_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P30_rm, ord.core_30_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P31
```{r}
P_31 <- subset_samples(VCPHY1, Subject=="P31")
P_31_LH<- subset_samples(P_31, Location=="left")
P_31_RH<- subset_samples(P_31, Location=="right")
P_31_SC<- subset_samples(P_31, Location=="scalp")
core_P31LH <- filter_taxa(P_31_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P31RH <- filter_taxa(P_31_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P31SC <- filter_taxa(P_31_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P31 <- merge_phyloseq(core_P31LH , core_P31RH, core_P31SC)

core_rm<- rownames(otu_table(CORE))
core_P31_rm <- taxa_names(core_P31)
core_P31_rm <- core_P31_rm[!(core_P31_rm %in% core_rm)]
core_P31_rm <- prune_taxa(core_P31_rm, core_P31)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P31_rm <- taxa_names(core_P31SC)
core_SC_P31_rm <- core_SC_P31_rm[!(core_SC_P31_rm %in% core_SC_rm)]
core_SC_P31_rm <- prune_taxa(core_SC_P31_rm, core_P31SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P31_hands <- merge_phyloseq(core_P31LH , core_P31RH)
core_HANDS_P31_rm <- taxa_names(core_P31_hands)
core_HANDS_P31_rm <- core_HANDS_P31_rm[!(core_HANDS_P31_rm %in% core_HANDS_rm)]
core_HANDS_P31_rm <- prune_taxa(core_HANDS_P31_rm, core_P31_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P31_rm <- taxa_names(core_P31LH)
core_LH_P31_rm <- core_LH_P31_rm[!(core_LH_P31_rm %in% core_LH_rm)]
core_LH_P31_rm <- prune_taxa(core_LH_P31_rm, core_P31LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P31_rm <- taxa_names(core_P31RH)
core_RH_P31_rm <- core_RH_P31_rm[!(core_RH_P31_rm %in% core_RH_rm)]
core_RH_P31_rm <- prune_taxa(core_RH_P31_rm, core_P31RH)
```
```{r}
norm_core_P31 <- transform_sample_counts(core_P31, function(x) x / sum(x) )
ord.core_31.pcoa <- ordinate(norm_core_P31, method="PCoA", distance="bray")
plot_ordination(norm_core_P31, ord.core_31.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P31_rm <- transform_sample_counts(core_P31_rm, function(x) x / sum(x) )
ord.core_31_rm.pcoa <- ordinate(norm_core_P31_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P31_rm, ord.core_31_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P32
```{r}
P_32 <- subset_samples(VCPHY1, Subject=="P32")
P_32_LH<- subset_samples(P_32, Location=="left")
P_32_RH<- subset_samples(P_32, Location=="right")
P_32_SC<- subset_samples(P_32, Location=="scalp")
core_P32LH <- filter_taxa(P_32_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P32RH <- filter_taxa(P_32_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P32SC <- filter_taxa(P_32_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P32 <- merge_phyloseq(core_P32LH , core_P32RH, core_P32SC)

core_rm<- rownames(otu_table(CORE))
core_P32_rm <- taxa_names(core_P32)
core_P32_rm <- core_P32_rm[!(core_P32_rm %in% core_rm)]
core_P32_rm <- prune_taxa(core_P32_rm, core_P32)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P32_rm <- taxa_names(core_P32SC)
core_SC_P32_rm <- core_SC_P32_rm[!(core_SC_P32_rm %in% core_SC_rm)]
core_SC_P32_rm <- prune_taxa(core_SC_P32_rm, core_P32SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P32_hands <- merge_phyloseq(core_P32LH , core_P32RH)
core_HANDS_P32_rm <- taxa_names(core_P32_hands)
core_HANDS_P32_rm <- core_HANDS_P32_rm[!(core_HANDS_P32_rm %in% core_HANDS_rm)]
core_HANDS_P32_rm <- prune_taxa(core_HANDS_P32_rm, core_P32_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P32_rm <- taxa_names(core_P32LH)
core_LH_P32_rm <- core_LH_P32_rm[!(core_LH_P32_rm %in% core_LH_rm)]
core_LH_P32_rm <- prune_taxa(core_LH_P32_rm, core_P32LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P32_rm <- taxa_names(core_P32RH)
core_RH_P32_rm <- core_RH_P32_rm[!(core_RH_P32_rm %in% core_RH_rm)]
core_RH_P32_rm <- prune_taxa(core_RH_P32_rm, core_P32RH)
```
```{r}
norm_core_P32 <- transform_sample_counts(core_P32, function(x) x / sum(x) )
ord.core_32.pcoa <- ordinate(norm_core_P32, method="PCoA", distance="bray")
plot_ordination(norm_core_P32, ord.core_32.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P32_rm <- transform_sample_counts(core_P32_rm, function(x) x / sum(x) )
ord.core_32_rm.pcoa <- ordinate(norm_core_P32_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P32_rm, ord.core_32_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P34
```{r}
P_34 <- subset_samples(VCPHY1, Subject=="P34")
P_34_LH<- subset_samples(P_34, Location=="left")
P_34_RH<- subset_samples(P_34, Location=="right")
P_34_SC<- subset_samples(P_34, Location=="scalp")
core_P34LH <- filter_taxa(P_34_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P34RH <- filter_taxa(P_34_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P34SC <- filter_taxa(P_34_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P34 <- merge_phyloseq(core_P34LH , core_P34RH, core_P34SC)

core_rm<- rownames(otu_table(CORE))
core_P34_rm <- taxa_names(core_P34)
core_P34_rm <- core_P34_rm[!(core_P34_rm %in% core_rm)]
core_P34_rm <- prune_taxa(core_P34_rm, core_P34)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P34_rm <- taxa_names(core_P34SC)
core_SC_P34_rm <- core_SC_P34_rm[!(core_SC_P34_rm %in% core_SC_rm)]
core_SC_P34_rm <- prune_taxa(core_SC_P34_rm, core_P34SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P34_hands <- merge_phyloseq(core_P34LH , core_P34RH)
core_HANDS_P34_rm <- taxa_names(core_P34_hands)
core_HANDS_P34_rm <- core_HANDS_P34_rm[!(core_HANDS_P34_rm %in% core_HANDS_rm)]
core_HANDS_P34_rm <- prune_taxa(core_HANDS_P34_rm, core_P34_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P34_rm <- taxa_names(core_P34LH)
core_LH_P34_rm <- core_LH_P34_rm[!(core_LH_P34_rm %in% core_LH_rm)]
core_LH_P34_rm <- prune_taxa(core_LH_P34_rm, core_P34LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P34_rm <- taxa_names(core_P34RH)
core_RH_P34_rm <- core_RH_P34_rm[!(core_RH_P34_rm %in% core_RH_rm)]
core_RH_P34_rm <- prune_taxa(core_RH_P34_rm, core_P34RH)
```
```{r}
norm_core_P34 <- transform_sample_counts(core_P34, function(x) x / sum(x) )
ord.core_34.pcoa <- ordinate(norm_core_P34, method="PCoA", distance="bray")
plot_ordination(norm_core_P34, ord.core_34.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P34_rm <- transform_sample_counts(core_P34_rm, function(x) x / sum(x) )
ord.core_34_rm.pcoa <- ordinate(norm_core_P34_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P34_rm, ord.core_34_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P35
```{r}
P_35 <- subset_samples(VCPHY1, Subject=="P35")
P_35_LH<- subset_samples(P_35, Location=="left")
P_35_RH<- subset_samples(P_35, Location=="right")
P_35_SC<- subset_samples(P_35, Location=="scalp")
core_P35LH <- filter_taxa(P_35_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P35RH <- filter_taxa(P_35_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P35SC <- filter_taxa(P_35_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P35 <- merge_phyloseq(core_P35LH , core_P35RH, core_P35SC)

core_rm<- rownames(otu_table(CORE))
core_P35_rm <- taxa_names(core_P35)
core_P35_rm <- core_P35_rm[!(core_P35_rm %in% core_rm)]
core_P35_rm <- prune_taxa(core_P35_rm, core_P35)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P35_rm <- taxa_names(core_P35SC)
core_SC_P35_rm <- core_SC_P35_rm[!(core_SC_P35_rm %in% core_SC_rm)]
core_SC_P35_rm <- prune_taxa(core_SC_P35_rm, core_P35SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P35_hands <- merge_phyloseq(core_P35LH , core_P35RH)
core_HANDS_P35_rm <- taxa_names(core_P35_hands)
core_HANDS_P35_rm <- core_HANDS_P35_rm[!(core_HANDS_P35_rm %in% core_HANDS_rm)]
core_HANDS_P35_rm <- prune_taxa(core_HANDS_P35_rm, core_P35_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P35_rm <- taxa_names(core_P35LH)
core_LH_P35_rm <- core_LH_P35_rm[!(core_LH_P35_rm %in% core_LH_rm)]
core_LH_P35_rm <- prune_taxa(core_LH_P35_rm, core_P35LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P35_rm <- taxa_names(core_P35RH)
core_RH_P35_rm <- core_RH_P35_rm[!(core_RH_P35_rm %in% core_RH_rm)]
core_RH_P35_rm <- prune_taxa(core_RH_P35_rm, core_P35RH)
```
```{r}
norm_core_P35 <- transform_sample_counts(core_P35, function(x) x / sum(x) )
ord.core_35.pcoa <- ordinate(norm_core_P35, method="PCoA", distance="bray")
plot_ordination(norm_core_P35, ord.core_35.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P35_rm <- transform_sample_counts(core_P35_rm, function(x) x / sum(x) )
ord.core_35_rm.pcoa <- ordinate(norm_core_P35_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P35_rm, ord.core_35_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```


#P36
```{r}
P_36 <- subset_samples(VCPHY1, Subject=="P36")
P_36_LH<- subset_samples(P_36, Location=="left")
P_36_RH<- subset_samples(P_36, Location=="right")
P_36_SC<- subset_samples(P_36, Location=="scalp")
core_P36LH <- filter_taxa(P_36_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P36RH <- filter_taxa(P_36_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P36SC <- filter_taxa(P_36_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P36 <- merge_phyloseq(core_P36LH , core_P36RH, core_P36SC)

core_rm<- rownames(otu_table(CORE))
core_P36_rm <- taxa_names(core_P36)
core_P36_rm <- core_P36_rm[!(core_P36_rm %in% core_rm)]
core_P36_rm <- prune_taxa(core_P36_rm, core_P36)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P36_rm <- taxa_names(core_P36SC)
core_SC_P36_rm <- core_SC_P36_rm[!(core_SC_P36_rm %in% core_SC_rm)]
core_SC_P36_rm <- prune_taxa(core_SC_P36_rm, core_P36SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P36_hands <- merge_phyloseq(core_P36LH , core_P36RH)
core_HANDS_P36_rm <- taxa_names(core_P36_hands)
core_HANDS_P36_rm <- core_HANDS_P36_rm[!(core_HANDS_P36_rm %in% core_HANDS_rm)]
core_HANDS_P36_rm <- prune_taxa(core_HANDS_P36_rm, core_P36_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P36_rm <- taxa_names(core_P36LH)
core_LH_P36_rm <- core_LH_P36_rm[!(core_LH_P36_rm %in% core_LH_rm)]
core_LH_P36_rm <- prune_taxa(core_LH_P36_rm, core_P36LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P36_rm <- taxa_names(core_P36RH)
core_RH_P36_rm <- core_RH_P36_rm[!(core_RH_P36_rm %in% core_RH_rm)]
core_RH_P36_rm <- prune_taxa(core_RH_P36_rm, core_P36RH)
```
```{r}
norm_core_P36 <- transform_sample_counts(core_P36, function(x) x / sum(x) )
ord.core_36.pcoa <- ordinate(norm_core_P36, method="PCoA", distance="bray")
plot_ordination(norm_core_P36, ord.core_36.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P36_rm <- transform_sample_counts(core_P36_rm, function(x) x / sum(x) )
ord.core_36_rm.pcoa <- ordinate(norm_core_P36_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P36_rm, ord.core_36_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P37
```{r}
P_37 <- subset_samples(VCPHY1, Subject=="P37")
P_37_LH<- subset_samples(P_37, Location=="left")
P_37_RH<- subset_samples(P_37, Location=="right")
P_37_SC<- subset_samples(P_37, Location=="scalp")
core_P37LH <- filter_taxa(P_37_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P37RH <- filter_taxa(P_37_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P37SC <- filter_taxa(P_37_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P37 <- merge_phyloseq(core_P37LH , core_P37RH, core_P37SC)

core_rm<- rownames(otu_table(CORE))
core_P37_rm <- taxa_names(core_P37)
core_P37_rm <- core_P37_rm[!(core_P37_rm %in% core_rm)]
core_P37_rm <- prune_taxa(core_P37_rm, core_P37)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P37_rm <- taxa_names(core_P37SC)
core_SC_P37_rm <- core_SC_P37_rm[!(core_SC_P37_rm %in% core_SC_rm)]
core_SC_P37_rm <- prune_taxa(core_SC_P37_rm, core_P37SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P37_hands <- merge_phyloseq(core_P37LH , core_P37RH)
core_HANDS_P37_rm <- taxa_names(core_P37_hands)
core_HANDS_P37_rm <- core_HANDS_P37_rm[!(core_HANDS_P37_rm %in% core_HANDS_rm)]
core_HANDS_P37_rm <- prune_taxa(core_HANDS_P37_rm, core_P37_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P37_rm <- taxa_names(core_P37LH)
core_LH_P37_rm <- core_LH_P37_rm[!(core_LH_P37_rm %in% core_LH_rm)]
core_LH_P37_rm <- prune_taxa(core_LH_P37_rm, core_P37LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P37_rm <- taxa_names(core_P37RH)
core_RH_P37_rm <- core_RH_P37_rm[!(core_RH_P37_rm %in% core_RH_rm)]
core_RH_P37_rm <- prune_taxa(core_RH_P37_rm, core_P37RH)
```
```{r}
norm_core_P37 <- transform_sample_counts(core_P37, function(x) x / sum(x) )
ord.core_37.pcoa <- ordinate(norm_core_P37, method="PCoA", distance="bray")
plot_ordination(norm_core_P37, ord.core_37.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P37_rm <- transform_sample_counts(core_P37_rm, function(x) x / sum(x) )
ord.core_37_rm.pcoa <- ordinate(norm_core_P37_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P37_rm, ord.core_37_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```


#P38
```{r}
P_38 <- subset_samples(VCPHY1, Subject=="P38")
P_38_LH<- subset_samples(P_38, Location=="left")
P_38_RH<- subset_samples(P_38, Location=="right")
P_38_SC<- subset_samples(P_38, Location=="scalp")
core_P38LH <- filter_taxa(P_38_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P38RH <- filter_taxa(P_38_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P38SC <- filter_taxa(P_38_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P38 <- merge_phyloseq(core_P38LH , core_P38RH, core_P38SC)

core_rm<- rownames(otu_table(CORE))
core_P38_rm <- taxa_names(core_P38)
core_P38_rm <- core_P38_rm[!(core_P38_rm %in% core_rm)]
core_P38_rm <- prune_taxa(core_P38_rm, core_P38)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P38_rm <- taxa_names(core_P38SC)
core_SC_P38_rm <- core_SC_P38_rm[!(core_SC_P38_rm %in% core_SC_rm)]
core_SC_P38_rm <- prune_taxa(core_SC_P38_rm, core_P38SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P38_hands <- merge_phyloseq(core_P38LH , core_P38RH)
core_HANDS_P38_rm <- taxa_names(core_P38_hands)
core_HANDS_P38_rm <- core_HANDS_P38_rm[!(core_HANDS_P38_rm %in% core_HANDS_rm)]
core_HANDS_P38_rm <- prune_taxa(core_HANDS_P38_rm, core_P38_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P38_rm <- taxa_names(core_P38LH)
core_LH_P38_rm <- core_LH_P38_rm[!(core_LH_P38_rm %in% core_LH_rm)]
core_LH_P38_rm <- prune_taxa(core_LH_P38_rm, core_P38LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P38_rm <- taxa_names(core_P38RH)
core_RH_P38_rm <- core_RH_P38_rm[!(core_RH_P38_rm %in% core_RH_rm)]
core_RH_P38_rm <- prune_taxa(core_RH_P38_rm, core_P38RH)
```
```{r}
norm_core_P38 <- transform_sample_counts(core_P38, function(x) x / sum(x) )
ord.core_38.pcoa <- ordinate(norm_core_P38, method="PCoA", distance="bray")
plot_ordination(norm_core_P38, ord.core_38.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P38_rm <- transform_sample_counts(core_P38_rm, function(x) x / sum(x) )
ord.core_38_rm.pcoa <- ordinate(norm_core_P38_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P38_rm, ord.core_38_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P39
```{r}
P_39 <- subset_samples(VCPHY1, Subject=="P39")
P_39_LH<- subset_samples(P_39, Location=="left")
P_39_RH<- subset_samples(P_39, Location=="right")
P_39_SC<- subset_samples(P_39, Location=="scalp")
core_P39LH <- filter_taxa(P_39_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P39RH <- filter_taxa(P_39_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P39SC <- filter_taxa(P_39_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P39 <- merge_phyloseq(core_P39LH , core_P39RH, core_P39SC)

core_rm<- rownames(otu_table(CORE))
core_P39_rm <- taxa_names(core_P39)
core_P39_rm <- core_P39_rm[!(core_P39_rm %in% core_rm)]
core_P39_rm <- prune_taxa(core_P39_rm, core_P39)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P39_rm <- taxa_names(core_P39SC)
core_SC_P39_rm <- core_SC_P39_rm[!(core_SC_P39_rm %in% core_SC_rm)]
core_SC_P39_rm <- prune_taxa(core_SC_P39_rm, core_P39SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P39_hands <- merge_phyloseq(core_P39LH , core_P39RH)
core_HANDS_P39_rm <- taxa_names(core_P39_hands)
core_HANDS_P39_rm <- core_HANDS_P39_rm[!(core_HANDS_P39_rm %in% core_HANDS_rm)]
core_HANDS_P39_rm <- prune_taxa(core_HANDS_P39_rm, core_P39_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P39_rm <- taxa_names(core_P39LH)
core_LH_P39_rm <- core_LH_P39_rm[!(core_LH_P39_rm %in% core_LH_rm)]
core_LH_P39_rm <- prune_taxa(core_LH_P39_rm, core_P39LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P39_rm <- taxa_names(core_P39RH)
core_RH_P39_rm <- core_RH_P39_rm[!(core_RH_P39_rm %in% core_RH_rm)]
core_RH_P39_rm <- prune_taxa(core_RH_P39_rm, core_P39RH)
```
```{r}
norm_core_P39 <- transform_sample_counts(core_P39, function(x) x / sum(x) )
ord.core_39.pcoa <- ordinate(norm_core_P39, method="PCoA", distance="bray")
plot_ordination(norm_core_P39, ord.core_39.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P39_rm <- transform_sample_counts(core_P39_rm, function(x) x / sum(x) )
ord.core_39_rm.pcoa <- ordinate(norm_core_P39_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P39_rm, ord.core_39_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P40
```{r}
P_40 <- subset_samples(VCPHY1, Subject=="P40")
P_40_LH<- subset_samples(P_40, Location=="left")
P_40_RH<- subset_samples(P_40, Location=="right")
P_40_SC<- subset_samples(P_40, Location=="scalp")
core_P40LH <- filter_taxa(P_40_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P40RH <- filter_taxa(P_40_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P40SC <- filter_taxa(P_40_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P40 <- merge_phyloseq(core_P40LH , core_P40RH, core_P40SC)

core_rm<- rownames(otu_table(CORE))
core_P40_rm <- taxa_names(core_P40)
core_P40_rm <- core_P40_rm[!(core_P40_rm %in% core_rm)]
core_P40_rm <- prune_taxa(core_P40_rm, core_P40)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P40_rm <- taxa_names(core_P40SC)
core_SC_P40_rm <- core_SC_P40_rm[!(core_SC_P40_rm %in% core_SC_rm)]
core_SC_P40_rm <- prune_taxa(core_SC_P40_rm, core_P40SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P40_hands <- merge_phyloseq(core_P40LH , core_P40RH)
core_HANDS_P40_rm <- taxa_names(core_P40_hands)
core_HANDS_P40_rm <- core_HANDS_P40_rm[!(core_HANDS_P40_rm %in% core_HANDS_rm)]
core_HANDS_P40_rm <- prune_taxa(core_HANDS_P40_rm, core_P40_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P40_rm <- taxa_names(core_P40LH)
core_LH_P40_rm <- core_LH_P40_rm[!(core_LH_P40_rm %in% core_LH_rm)]
core_LH_P40_rm <- prune_taxa(core_LH_P40_rm, core_P40LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P40_rm <- taxa_names(core_P40RH)
core_RH_P40_rm <- core_RH_P40_rm[!(core_RH_P40_rm %in% core_RH_rm)]
core_RH_P40_rm <- prune_taxa(core_RH_P40_rm, core_P40RH)
```
```{r}
norm_core_P40 <- transform_sample_counts(core_P40, function(x) x / sum(x) )
ord.core_40.pcoa <- ordinate(norm_core_P40, method="PCoA", distance="bray")
plot_ordination(norm_core_P40, ord.core_40.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P40_rm <- transform_sample_counts(core_P40_rm, function(x) x / sum(x) )
ord.core_40_rm.pcoa <- ordinate(norm_core_P40_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P40_rm, ord.core_40_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P41
```{r}
P_41 <- subset_samples(VCPHY1, Subject=="P41")
P_41_LH<- subset_samples(P_41, Location=="left")
P_41_RH<- subset_samples(P_41, Location=="right")
P_41_SC<- subset_samples(P_41, Location=="scalp")
core_P41LH <- filter_taxa(P_41_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P41RH <- filter_taxa(P_41_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P41SC <- filter_taxa(P_41_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P41 <- merge_phyloseq(core_P41LH , core_P41RH, core_P41SC)

core_rm<- rownames(otu_table(CORE))
core_P41_rm <- taxa_names(core_P41)
core_P41_rm <- core_P41_rm[!(core_P41_rm %in% core_rm)]
core_P41_rm <- prune_taxa(core_P41_rm, core_P41)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P41_rm <- taxa_names(core_P41SC)
core_SC_P41_rm <- core_SC_P41_rm[!(core_SC_P41_rm %in% core_SC_rm)]
core_SC_P41_rm <- prune_taxa(core_SC_P41_rm, core_P41SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P41_hands <- merge_phyloseq(core_P41LH , core_P41RH)
core_HANDS_P41_rm <- taxa_names(core_P41_hands)
core_HANDS_P41_rm <- core_HANDS_P41_rm[!(core_HANDS_P41_rm %in% core_HANDS_rm)]
core_HANDS_P41_rm <- prune_taxa(core_HANDS_P41_rm, core_P41_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P41_rm <- taxa_names(core_P41LH)
core_LH_P41_rm <- core_LH_P41_rm[!(core_LH_P41_rm %in% core_LH_rm)]
core_LH_P41_rm <- prune_taxa(core_LH_P41_rm, core_P41LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P41_rm <- taxa_names(core_P41RH)
core_RH_P41_rm <- core_RH_P41_rm[!(core_RH_P41_rm %in% core_RH_rm)]
core_RH_P41_rm <- prune_taxa(core_RH_P41_rm, core_P41RH)
```
```{r}
norm_core_P41 <- transform_sample_counts(core_P41, function(x) x / sum(x) )
ord.core_41.pcoa <- ordinate(norm_core_P41, method="PCoA", distance="bray")
plot_ordination(norm_core_P41, ord.core_41.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P41_rm <- transform_sample_counts(core_P41_rm, function(x) x / sum(x) )
ord.core_41_rm.pcoa <- ordinate(norm_core_P41_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P41_rm, ord.core_41_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P42
```{r}
P_42 <- subset_samples(VCPHY1, Subject=="P42")
P_42_LH<- subset_samples(P_42, Location=="left")
P_42_RH<- subset_samples(P_42, Location=="right")
P_42_SC<- subset_samples(P_42, Location=="scalp")
core_P42LH <- filter_taxa(P_42_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P42RH <- filter_taxa(P_42_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P42SC <- filter_taxa(P_42_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P42 <- merge_phyloseq(core_P42LH , core_P42RH, core_P42SC)

core_rm<- rownames(otu_table(CORE))
core_P42_rm <- taxa_names(core_P42)
core_P42_rm <- core_P42_rm[!(core_P42_rm %in% core_rm)]
core_P42_rm <- prune_taxa(core_P42_rm, core_P42)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P42_rm <- taxa_names(core_P42SC)
core_SC_P42_rm <- core_SC_P42_rm[!(core_SC_P42_rm %in% core_SC_rm)]
core_SC_P42_rm <- prune_taxa(core_SC_P42_rm, core_P42SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P42_hands <- merge_phyloseq(core_P42LH , core_P42RH)
core_HANDS_P42_rm <- taxa_names(core_P42_hands)
core_HANDS_P42_rm <- core_HANDS_P42_rm[!(core_HANDS_P42_rm %in% core_HANDS_rm)]
core_HANDS_P42_rm <- prune_taxa(core_HANDS_P42_rm, core_P42_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P42_rm <- taxa_names(core_P42LH)
core_LH_P42_rm <- core_LH_P42_rm[!(core_LH_P42_rm %in% core_LH_rm)]
core_LH_P42_rm <- prune_taxa(core_LH_P42_rm, core_P42LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P42_rm <- taxa_names(core_P42RH)
core_RH_P42_rm <- core_RH_P42_rm[!(core_RH_P42_rm %in% core_RH_rm)]
core_RH_P42_rm <- prune_taxa(core_RH_P42_rm, core_P42RH)
```
```{r}
norm_core_P42 <- transform_sample_counts(core_P42, function(x) x / sum(x) )
ord.core_42.pcoa <- ordinate(norm_core_P42, method="PCoA", distance="bray")
plot_ordination(norm_core_P42, ord.core_42.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P42_rm <- transform_sample_counts(core_P42_rm, function(x) x / sum(x) )
ord.core_42_rm.pcoa <- ordinate(norm_core_P42_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P42_rm, ord.core_42_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#P43
```{r}
P_43 <- subset_samples(VCPHY1, Subject=="P43")
P_43_LH<- subset_samples(P_43, Location=="left")
P_43_RH<- subset_samples(P_43, Location=="right")
P_43_SC<- subset_samples(P_43, Location=="scalp")
core_P43LH <- filter_taxa(P_43_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P43RH <- filter_taxa(P_43_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P43SC <- filter_taxa(P_43_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P43 <- merge_phyloseq(core_P43LH , core_P43RH, core_P43SC)

core_rm<- rownames(otu_table(CORE))
core_P43_rm <- taxa_names(core_P43)
core_P43_rm <- core_P43_rm[!(core_P43_rm %in% core_rm)]
core_P43_rm <- prune_taxa(core_P43_rm, core_P43)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P43_rm <- taxa_names(core_P43SC)
core_SC_P43_rm <- core_SC_P43_rm[!(core_SC_P43_rm %in% core_SC_rm)]
core_SC_P43_rm <- prune_taxa(core_SC_P43_rm, core_P43SC)

core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P43_hands <- merge_phyloseq(core_P43LH , core_P43RH)
core_HANDS_P43_rm <- taxa_names(core_P43_hands)
core_HANDS_P43_rm <- core_HANDS_P43_rm[!(core_HANDS_P43_rm %in% core_HANDS_rm)]
core_HANDS_P43_rm <- prune_taxa(core_HANDS_P43_rm, core_P43_hands)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P43_rm <- taxa_names(core_P43LH)
core_LH_P43_rm <- core_LH_P43_rm[!(core_LH_P43_rm %in% core_LH_rm)]
core_LH_P43_rm <- prune_taxa(core_LH_P43_rm, core_P43LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P43_rm <- taxa_names(core_P43RH)
core_RH_P43_rm <- core_RH_P43_rm[!(core_RH_P43_rm %in% core_RH_rm)]
core_RH_P43_rm <- prune_taxa(core_RH_P43_rm, core_P43RH)
```
```{r}
norm_core_P43 <- transform_sample_counts(core_P43, function(x) x / sum(x) )
ord.core_43.pcoa <- ordinate(norm_core_P43, method="PCoA", distance="bray")
plot_ordination(norm_core_P43, ord.core_43.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
norm_core_P43_rm <- transform_sample_counts(core_P43_rm, function(x) x / sum(x) )
ord.core_43_rm.pcoa <- ordinate(norm_core_P43_rm, method="PCoA", distance="bray")
plot_ordination(norm_core_P43_rm, ord.core_43_rm.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)
```

#Merge each person's core togeher 
```{r}
#Core all locations merge 
core_total_merge <- merge_phyloseq(core_P01, core_P02, core_P03, core_P04, core_P05, core_P06, core_P07, 
                            core_P08, core_P09, core_P10, core_P11, core_P12, core_P13, core_P14,
                            core_P15, core_P16, core_P17, core_P18, core_P19, core_P20, core_P21,
                            core_P22, core_P23, core_P24, core_P25, core_P26, core_P27, core_P28,
                            core_P29, core_P30, core_P31, core_P32, core_P34, core_P35, core_P36,
                            core_P37, core_P38, core_P39, core_P40, core_P41, core_P42, core_P43)
#Core by location merge

#Left Hand
core_LH_merge <- merge_phyloseq(core_P01LH, core_P02LH, core_P03LH, core_P04LH, core_P05LH, core_P06LH, core_P07LH, 
                            core_P08LH, core_P09LH, core_P10LH, core_P11LH, core_P12LH, core_P13LH, core_P14LH,
                            core_P15LH, core_P16LH, core_P17LH, core_P18LH, core_P19LH, core_P20LH, core_P21LH,
                            core_P22LH, core_P23LH, core_P24LH, core_P25LH, core_P26LH, core_P27LH, core_P28LH,
                            core_P29LH, core_P30LH, core_P31LH, core_P32LH, core_P34LH, core_P35LH, core_P36LH,
                            core_P37LH, core_P38LH, core_P39LH, core_P40LH, core_P41LH, core_P42LH, core_P43LH)

#Right Hand
core_RH_merge <- merge_phyloseq(core_P01RH, core_P02RH, core_P03RH, core_P04RH, core_P05RH, core_P06RH, core_P07RH, 
                            core_P08RH, core_P09RH, core_P10RH, core_P11RH, core_P12RH, core_P13RH, core_P14RH,
                            core_P15RH, core_P16RH, core_P17RH, core_P18RH, core_P19RH, core_P20RH, core_P21RH,
                            core_P22RH, core_P23RH, core_P24RH, core_P25RH, core_P26RH, core_P27RH, core_P28RH,
                            core_P29RH, core_P30RH, core_P31RH, core_P32RH, core_P34RH, core_P35RH, core_P36RH,
                            core_P37RH, core_P38RH, core_P39RH, core_P40RH, core_P41RH, core_P42RH, core_P43RH)
#Scalp
core_SC_merge <- merge_phyloseq(core_P01SC, core_P02SC, core_P03SC, core_P04SC, core_P05SC, core_P06SC, core_P07SC, 
                            core_P08SC, core_P09SC, core_P10SC, core_P11SC, core_P12SC, core_P13SC, core_P14SC,
                            core_P15SC, core_P16SC, core_P17SC, core_P18SC, core_P19SC, core_P20SC, core_P21SC,
                            core_P22SC, core_P23SC, core_P24SC, core_P25SC, core_P26SC, core_P27SC, core_P28SC,
                            core_P29SC, core_P30SC, core_P31SC, core_P32SC, core_P34SC, core_P35SC, core_P36SC,
                            core_P37SC, core_P38SC, core_P39SC, core_P40SC, core_P41SC, core_P42SC, core_P43SC)

#core_SC_rm<- rownames(otu_table(CORE_SC))
#core_SC_all_rm <- taxa_names(core_SC_merge)
#core_SC_all_rm <- core_SC_all_rm[!(core_SC_all_rm %in% core_SC_rm)]
#core_SC_all_rm <- prune_taxa(core_SC_all_rm, core_SC_merge)
#Both Hands
core_hands_merge <- merge_phyloseq(core_P01LH, core_P02LH, core_P03LH, core_P04LH, core_P05LH, core_P06LH, core_P07LH, 
                            core_P08LH, core_P09LH, core_P10LH, core_P11LH, core_P12LH, core_P13LH, core_P14LH,
                            core_P15LH, core_P16LH, core_P17LH, core_P18LH, core_P19LH, core_P20LH, core_P21LH,
                            core_P22LH, core_P23LH, core_P24LH, core_P25LH, core_P26LH, core_P27LH, core_P28LH,
                            core_P29LH, core_P30LH, core_P31LH, core_P32LH, core_P34LH, core_P35LH, core_P36LH,
                            core_P37LH, core_P38LH, core_P39LH, core_P40LH, core_P41LH, core_P42LH, core_P43LH,
                            core_P01RH, core_P02RH, core_P03RH, core_P04RH, core_P05RH, core_P06RH, core_P07RH, 
                            core_P08RH, core_P09RH, core_P10RH, core_P11RH, core_P12RH, core_P13RH, core_P14RH,
                            core_P15RH, core_P16RH, core_P17RH, core_P18RH, core_P19RH, core_P20RH, core_P21RH,
                            core_P22RH, core_P23RH, core_P24RH, core_P25RH, core_P26RH, core_P27RH, core_P28RH,
                            core_P29RH, core_P30RH, core_P31RH, core_P32RH, core_P34RH, core_P35RH, core_P36RH,
                            core_P37RH, core_P38RH, core_P39RH, core_P40RH, core_P41RH, core_P42RH, core_P43RH)

core_rm <- merge_phyloseq(core_P01_rm, core_P02_rm, core_P03_rm, core_P04_rm, core_P05_rm, core_P06_rm, core_P07_rm, 
                            core_P08_rm, core_P09_rm, core_P10_rm, core_P11_rm, core_P12_rm, core_P13_rm, core_P14_rm,
                            core_P15_rm, core_P16_rm, core_P17_rm, core_P18_rm, core_P19_rm, core_P20_rm, core_P21_rm,
                            core_P22_rm, core_P23_rm, core_P24_rm, core_P25_rm, core_P26_rm, core_P27_rm, core_P28_rm,
                            core_P29_rm, core_P30_rm, core_P31_rm, core_P32_rm, core_P34_rm, core_P35_rm, core_P36_rm,
                            core_P37_rm, core_P38_rm, core_P39_rm, core_P40_rm, core_P41_rm, core_P42_rm, core_P43_rm)

core_SC_rm <- merge_phyloseq(core_SC_P01_rm, core_SC_P02_rm, core_SC_P03_rm, core_SC_P04_rm, core_SC_P05_rm, core_SC_P06_rm,
                                core_SC_P07_rm, core_SC_P08_rm, core_SC_P09_rm, core_SC_P10_rm, core_SC_P11_rm, core_SC_P12_rm,
                                core_SC_P13_rm, core_SC_P14_rm, core_SC_P15_rm, core_SC_P16_rm, core_SC_P17_rm, core_SC_P18_rm, 
                                core_SC_P19_rm, core_SC_P20_rm, core_SC_P21_rm, core_SC_P22_rm, core_SC_P23_rm, core_SC_P24_rm,
                                core_SC_P25_rm, core_SC_P26_rm, core_SC_P27_rm, core_SC_P28_rm, core_SC_P29_rm, core_SC_P30_rm,
                                core_SC_P31_rm, core_SC_P32_rm, core_SC_P34_rm, core_SC_P35_rm, core_SC_P36_rm, core_SC_P37_rm, 
                                core_SC_P38_rm, core_SC_P39_rm, core_SC_P40_rm, core_SC_P41_rm, core_SC_P42_rm, core_SC_P43_rm)

core_LH_rm <- merge_phyloseq(core_LH_P01_rm, core_LH_P02_rm, core_LH_P03_rm, core_LH_P04_rm, core_LH_P05_rm, core_LH_P06_rm,
                                core_LH_P07_rm, core_LH_P08_rm, core_LH_P09_rm, core_LH_P10_rm, core_LH_P11_rm, core_LH_P12_rm,
                                core_LH_P13_rm, core_LH_P14_rm, core_LH_P15_rm, core_LH_P16_rm, core_LH_P17_rm, core_LH_P18_rm, 
                                core_LH_P19_rm, core_LH_P20_rm, core_LH_P21_rm, core_LH_P22_rm, core_LH_P23_rm, core_LH_P24_rm,
                                core_LH_P25_rm, core_LH_P26_rm, core_LH_P27_rm, core_LH_P28_rm, core_LH_P29_rm, core_LH_P30_rm,
                                core_LH_P31_rm, core_LH_P32_rm, core_LH_P34_rm, core_LH_P35_rm, core_LH_P36_rm, core_LH_P37_rm, 
                                core_LH_P38_rm, core_LH_P39_rm, core_LH_P40_rm, core_LH_P41_rm, core_LH_P42_rm, core_LH_P43_rm)

core_RH_rm <- merge_phyloseq(core_RH_P01_rm, core_RH_P02_rm, core_RH_P03_rm, core_RH_P04_rm, core_RH_P05_rm, core_RH_P06_rm,
                                core_RH_P07_rm, core_RH_P08_rm, core_RH_P09_rm, core_RH_P10_rm, core_RH_P11_rm, core_RH_P12_rm,
                                core_RH_P13_rm, core_RH_P14_rm, core_RH_P15_rm, core_RH_P16_rm, core_RH_P17_rm, core_RH_P18_rm, 
                                core_RH_P19_rm, core_RH_P20_rm, core_RH_P21_rm, core_RH_P22_rm, core_RH_P23_rm, core_RH_P24_rm,
                                core_RH_P25_rm, core_RH_P26_rm, core_RH_P27_rm, core_RH_P28_rm, core_RH_P29_rm, core_RH_P30_rm,
                                core_RH_P31_rm, core_RH_P32_rm, core_RH_P34_rm, core_RH_P35_rm, core_RH_P36_rm, core_RH_P37_rm, 
                                core_RH_P38_rm, core_RH_P39_rm, core_RH_P40_rm, core_RH_P41_rm, core_RH_P42_rm, core_RH_P43_rm)

core_HANDS_rm <- merge_phyloseq(core_HANDS_P01_rm, core_HANDS_P02_rm, core_HANDS_P03_rm, core_HANDS_P04_rm, core_HANDS_P05_rm, core_HANDS_P06_rm,
                                core_HANDS_P07_rm, core_HANDS_P08_rm, core_HANDS_P09_rm, core_HANDS_P10_rm, core_HANDS_P11_rm, core_HANDS_P12_rm,
                                core_HANDS_P13_rm, core_HANDS_P14_rm, core_HANDS_P15_rm, core_HANDS_P16_rm, core_HANDS_P17_rm, core_HANDS_P18_rm, 
                                core_HANDS_P19_rm, core_HANDS_P20_rm, core_HANDS_P21_rm, core_HANDS_P22_rm, core_HANDS_P23_rm, core_HANDS_P24_rm,
                                core_HANDS_P25_rm, core_HANDS_P26_rm, core_HANDS_P27_rm, core_HANDS_P28_rm, core_HANDS_P29_rm, core_HANDS_P30_rm,
                                core_HANDS_P31_rm, core_HANDS_P32_rm, core_HANDS_P34_rm, core_HANDS_P35_rm, core_HANDS_P36_rm, core_HANDS_P37_rm, 
                                core_HANDS_P38_rm, core_HANDS_P39_rm, core_HANDS_P40_rm, core_HANDS_P41_rm, core_HANDS_P42_rm, core_HANDS_P43_rm)

#core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
#core_P01_hands <- merge_phyloseq(core_P01LH , core_P01RH)
#core_HANDS_P01_rm <- taxa_names(core_P01_hands)
#core_HANDS_P01_rm <- core_HANDS_P01_rm[!(core_HANDS_P01_rm %in% core_HANDS_rm)]
#core_HANDS_P01_rm <- prune_taxa(core_HANDS_P01_rm, core_P01_hands)
```

#barplot of top taxa
```{r}
core_total_merge.sum<- tapply(taxa_sums(core_total_merge), tax_table(core_total_merge)[, "family"], sum, na.rm=TRUE)
core_total_merge_top10phyla <-names(sort(core_total_merge.sum, TRUE))[1:10]
core_total_merge_top10 <- prune_taxa((tax_table(core_total_merge)[, "family"] %in% core_total_merge_top10phyla), core_total_merge)
plot_bar(core_total_merge_top10, x="Subject", fill = "family")
```

#100% stacked bargraph
```{r}
core_total_merge2 <- transform_sample_counts(core_total_merge_top10, function(x) x / sum(x) )
plot_bar(core_total_merge2, x="Subject", fill = "family")
```

```{r}
plot_heatmap(core_total_merge_top10, method = "NMDS", distance = "bray")
```

#core merge 
```{r}
# Extract abundance matrix from the phyloseq object
Total_core_OTU = as(otu_table(core_total_merge), "matrix")
# transpose if necessary
Total_core_OTU <- t(Total_core_OTU)
# Coerce to data.frame
Total_core_OTUdf = as.data.frame(Total_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_002_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_03", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_04", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_06", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_07", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_021_13", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_01", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_020_10", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_015_21", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_05", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_020_22", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_02", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_015_20", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_003_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_004_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_005_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_006_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_007_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_008_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_009_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_010_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_011_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_012_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_013_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_014_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_015_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_016_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_017_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_018_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_019_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_020_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_021_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_022_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_023_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_024_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_025_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_026_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_027_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_028_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_029_27", Metadata_core_stat$SampleID),]
Metadata_core_stat <- Metadata_core_stat[!grepl("HV_030_27", Metadata_core_stat$SampleID),]

#alpha diversity measures:
Metadata_core_stat$shannon.vegan <- diversity(Total_core_OTUdf, index = "shannon")
Metadata_core_stat$simpson.vegan <- diversity(Total_core_OTUdf, index = "simpson")
Metadata_core_stat$invsimpson.vegan <- diversity(Total_core_OTUdf, index = "invsimpson")
```
```{r}
model_core_AOV <- aov(shannon.vegan~Gender+Subject/Location+ Subject/Time_Collected+Subject, data = Metadata_core_stat)
summary(model_core_AOV)
```

```{r}
norm_core_groups <- transform_sample_counts(core_total_merge , function(x) x / sum(x) )
ord.core_groups.pcoa <- ordinate(norm_core_groups, method="PCoA", distance="jaccard")
plot_ordination(norm_core_groups, ord.core_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```

```{r}
metadata_core <- as(sample_data(norm_core_groups ), "data.frame")
adonis(phyloseq::distance(norm_core_groups, method="jaccard") ~  Subject+Time_Collected+Location,
       data = metadata_core)


Time3<- subset_samples(VCPHY1, Time_Collected=="3_month")
Time4<- subset_samples(VCPHY1, Time_Collected=="6_month")

sampledf <- data.frame(sample_data(Time3))
norm_core_groups <- transform_sample_counts(Time3 , function(x) x / sum(x) )
adonis(phyloseq::distance(norm_core_groups, method="jaccard") ~  Gender+Subject+Subject/Location
       data = sampledf)
```

```{r}
metadata_core <- as(sample_data(norm_core_groups ), "data.frame")
erie_bray <- phyloseq::distance(norm_core_groups, method="jaccard")
sampledf <- data.frame(sample_data(core_total_merge))

adonis(erie_bray ~ Subject/Time_Collected + Subject/Location, strata = sampledf$Subject,
       data = sampledf)

adonis(erie_bray ~ Subject+ Time_Collected, strata = sampledf$Subject, data = sampledf)
```
#LH Core
```{r}
# Extract abundance matrix from the phyloseq object
LH_core_OTU = as(otu_table(core_LH_merge), "matrix")
# transpose if necessary
LH_core_OTU <- t(LH_core_OTU)
# Coerce to data.frame
LH_core_OTUdf = as.data.frame(LH_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

LH_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_002_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_03", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_04", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_06", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_07", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_021_13", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_01", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_020_10", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_015_21", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_05", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_020_22", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_02", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_015_20", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_003_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_004_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_005_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_006_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_007_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_008_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_009_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_010_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_011_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_012_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_013_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_014_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_015_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_016_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_017_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_018_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_019_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_020_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_021_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_022_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_023_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_024_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_025_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_026_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_027_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_028_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_029_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("HV_030_27", LH_Metadata_core_stat$SampleID),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("right", LH_Metadata_core_stat$Location),]
LH_Metadata_core_stat <- LH_Metadata_core_stat[!grepl("scalp", LH_Metadata_core_stat$Location),]
#alpha diversity measures:
LH_Metadata_core_stat$shannon.vegan <- diversity(LH_core_OTUdf, index = "shannon")
LH_Metadata_core_stat$simpson.vegan <- diversity(LH_core_OTUdf, index = "simpson")
LH_Metadata_core_stat$invsimpson.vegan <- diversity(LH_core_OTUdf, index = "invsimpson")
```

```{r}
model_LH_AOV <- aov(shannon.vegan~Gender+ Subject/Time_Week+Subject, data = LH_Metadata_core_stat)
summary(model_LH_AOV)
```
```{r}
norm_LH_groups <- transform_sample_counts(core_LH_merge , function(x) x / sum(x) )
ord.LH_groups.pcoa <- ordinate(norm_LH_groups, method="PCoA", distance="bray")
plot_ordination(norm_LH_groups, ord.LH_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```

```{r}
metadata_LH_core <- as(sample_data(norm_LH_groups ), "data.frame")
adonis(phyloseq::distance(norm_LH_groups, method="bray") ~  Gender+Subject+Subject/Time_Week,
       data=metadata_LH_core)
```

#RH Core
```{r}
# Extract abundance matrix from the phyloseq object
RH_core_OTU = as(otu_table(core_RH_merge), "matrix")
# transpose if necessary
RH_core_OTU <- t(RH_core_OTU)
# Coerce to data.frame
RH_core_OTUdf = as.data.frame(RH_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

RH_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_002_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_03", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_04", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_06", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_07", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_021_13", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_01", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_020_10", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_015_21", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_05", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_020_22", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_02", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_015_20", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_003_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_004_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_005_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_006_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_007_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_008_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_009_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_010_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_011_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_012_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_013_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_014_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_015_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_016_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_017_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_018_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_019_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_020_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_021_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_022_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_023_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_024_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_025_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_026_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_027_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_028_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_029_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("HV_030_27", RH_Metadata_core_stat$SampleID),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("left", RH_Metadata_core_stat$Location),]
RH_Metadata_core_stat <- RH_Metadata_core_stat[!grepl("scalp", RH_Metadata_core_stat$Location),]
#alpha diversity measures:
RH_Metadata_core_stat$shannon.vegan <- diversity(RH_core_OTUdf, index = "shannon")
RH_Metadata_core_stat$simpson.vegan <- diversity(RH_core_OTUdf, index = "simpson")
RH_Metadata_core_stat$invsimpson.vegan <- diversity(RH_core_OTUdf, index = "invsimpson")
```
```{r}
model_RH_AOV <- aov(shannon.vegan~Gender+ Subject/Time_Week+Subject, data = RH_Metadata_core_stat)
summary(model_RH_AOV)
```
```{r}
norm_RH_groups <- transform_sample_counts(core_RH_merge , function(x) x / sum(x) )
ord.RH_groups.pcoa <- ordinate(norm_RH_groups, method="PCoA", distance="bray")
plot_ordination(norm_RH_groups, ord.RH_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```
```{r}
metadata_RH_core <- as(sample_data(norm_RH_groups ), "data.frame")
adonis(phyloseq::distance(norm_RH_groups, method="bray") ~  Gender+Subject+Subject/Time_Week,
       data=metadata_RH_core)
```

#SC Core
```{r}
# Extract abundance matrix from the phyloseq object
SC_core_OTU = as(otu_table(core_SC_merge), "matrix")
# transpose if necessary
SC_core_OTU <- t(SC_core_OTU)
# Coerce to data.frame
SC_core_OTUdf = as.data.frame(SC_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

SC_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_002_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_03", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_04", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_06", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_07", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_021_13", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_01", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_020_10", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_015_21", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_05", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_020_22", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_02", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_015_20", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_003_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_004_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_005_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_006_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_007_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_008_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_009_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_010_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_011_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_012_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_013_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_014_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_015_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_016_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_017_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_018_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_019_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_020_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_021_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_022_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_023_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_024_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_025_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_026_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_027_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_028_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_029_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("HV_030_27", SC_Metadata_core_stat$SampleID),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("right", SC_Metadata_core_stat$Location),]
SC_Metadata_core_stat <- SC_Metadata_core_stat[!grepl("left", SC_Metadata_core_stat$Location),]
#alpha diversity measures:
SC_Metadata_core_stat$shannon.vegan <- diversity(SC_core_OTUdf, index = "shannon")
SC_Metadata_core_stat$simpson.vegan <- diversity(SC_core_OTUdf, index = "simpson")
SC_Metadata_core_stat$invsimpson.vegan <- diversity(SC_core_OTUdf, index = "invsimpson")
```
```{r}
model_SC_AOV <- aov(shannon.vegan~Gender+ Subject/Time_Week+Subject, data = SC_Metadata_core_stat)
summary(model_SC_AOV)
```

```{r}
norm_SC_groups <- transform_sample_counts(core_SC_merge , function(x) x / sum(x) )
ord.SC_groups.pcoa <- ordinate(norm_SC_groups, method="PCoA", distance="bray")
plot_ordination(norm_SC_groups, ord.SC_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```

```{r}
metadata_SC_core <- as(sample_data(norm_SC_groups ), "data.frame")
adonis(phyloseq::distance(norm_SC_groups, method="bray") ~  Gender+Subject+Subject/Time_Week,
       data=metadata_SC_core)
```

#Hands Core
```{r}
# Extract abundance matrix from the phyloseq object
hands_core_OTU = as(otu_table(core_hands_merge), "matrix")
# transpose if necessary
hands_core_OTU <- t(hands_core_OTU)
# Coerce to data.frame
hands_core_OTUdf = as.data.frame(hands_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

hands_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_002_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_03", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_04", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_06", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_07", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_021_13", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_01", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_020_10", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_015_21", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_05", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_020_22", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_02", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_015_20", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_003_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_004_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_005_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_006_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_007_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_008_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_009_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_010_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_011_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_012_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_013_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_014_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_015_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_016_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_017_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_018_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_019_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_020_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_021_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_022_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_023_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_024_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_025_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_026_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_027_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_028_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_029_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("HV_030_27", hands_Metadata_core_stat$SampleID),]
hands_Metadata_core_stat <- hands_Metadata_core_stat[!grepl("scalp", hands_Metadata_core_stat$Location),]

#alpha diversity measures:
hands_Metadata_core_stat$shannon.vegan <- diversity(hands_core_OTUdf, index = "shannon")
hands_Metadata_core_stat$simpson.vegan <- diversity(hands_core_OTUdf, index = "simpson")
hands_Metadata_core_stat$invsimpson.vegan <- diversity(hands_core_OTUdf, index = "invsimpson")
```
```{r}
model_hands_AOV <- aov(shannon.vegan~Gender+Subject/Location+ Subject/Time_Collected+Subject, data = hands_Metadata_core_stat)
summary(model_hands_AOV)
```

```{r}
norm_hands_groups <- transform_sample_counts(core_hands_merge , function(x) x / sum(x) )
ord.hands_groups.pcoa <- ordinate(norm_hands_groups, method="PCoA", distance="bray")
plot_ordination(norm_hands_groups, ord.hands_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)


erie_bray <- phyloseq::distance(norm_hands_groups, method = "bray")
sampledf <- data.frame(sample_data(core_hands_merge))
adonis(erie_bray ~ Subject, data = sampledf)
# Homogeneity of dispersion test

beta <- betadisper(erie_bray, sampledf$Subject)
permutest(beta)
THSD <- TukeyHSD(beta)



options(max.print = 1000)
getOption("max.print")
#1000

```

```{r}
metadata_hands_core <- as(sample_data(norm_hands_groups ), "data.frame")
adonis(phyloseq::distance(norm_hands_groups, method="bray") ~  Gender+Subject+Subject/Time_Collected+Subject/Location,
       data=metadata_hands_core)
```

#Core Remove
```{r}
# Extract abundance matrix from the phyloseq object
core_rm_core_OTU = as(otu_table(core_rm), "matrix")
# transpose if necessary
core_rm_core_OTU <- t(core_rm_core_OTU)
# Coerce to data.frame
core_rm_core_OTUdf = as.data.frame(core_rm_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

core_rm_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_002_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_03", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_04", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_06", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_07", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_021_13", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_01", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_020_10", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_015_21", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_05", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_020_22", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_02", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_015_20", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_003_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_004_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_005_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_006_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_007_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_008_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_009_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_010_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_011_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_012_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_013_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_014_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_015_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_016_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_017_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_018_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_019_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_020_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_021_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_022_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_023_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_024_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_025_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_026_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_027_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_028_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_029_27", core_rm_Metadata_core_stat$SampleID),]
core_rm_Metadata_core_stat <- core_rm_Metadata_core_stat[!grepl("HV_030_27", core_rm_Metadata_core_stat$SampleID),]


#alpha diversity measures:
core_rm_Metadata_core_stat$shannon.vegan <- diversity(core_rm_core_OTUdf, index = "shannon")
core_rm_Metadata_core_stat$simpson.vegan <- diversity(core_rm_core_OTUdf, index = "simpson")
core_rm_Metadata_core_stat$invsimpson.vegan <- diversity(core_rm_core_OTUdf, index = "invsimpson")
```

```{r}
model_core_rm_AOV <- aov(shannon.vegan~Gender+ Subject/Time_Week+Subject+Subject/Location, data = core_rm_Metadata_core_stat)
summary(model_core_rm_AOV)
```


```{r}
core_rm1 = prune_samples(sample_sums(core_rm)>=1, core_rm)
norm_core_rm_groups <- transform_sample_counts(core_rm1 , function(x) x / sum(x) )
ord.core_rm_groups.pcoa <- ordinate(norm_core_rm_groups, method="PCoA", distance="bray")
plot_ordination(norm_core_rm_groups, ord.core_rm_groups.pcoa, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```

```{r}
metadata_core_rm_core <- as(sample_data(norm_core_rm_groups ), "data.frame")
adonis(phyloseq::distance(norm_core_rm_groups, method="bray") ~  Gender+Subject+Subject/Time_Week+Subject/Location,
       data=metadata_core_rm_core)
```

```{r}
#https://rpubs.com/michberr/randomforestmicrobe

#load libraries
  #library(randomForest)
  #library(ggplot2)
  #library(vegan)
  #library(magrittr)
  #library(scales)
  #library(reshape2)
  #library(phyloseq)
  #library(knitr)
#hands.merge<- ANPHY1 %>% subset_samples(Location != "scalp")

myround <- function(x) { trunc(x + 0.5) }

scale_reads <- function(physeq, n = min(sample_sums(physeq)), round = "floor") {
  
  # transform counts to n
  physeq.scale <- transform_sample_counts(physeq, 
    function(x) {(n * x/sum(x))}
  )
  
  # Pick the rounding functions
  if (round == "floor"){
    otu_table(physeq.scale) <- floor(otu_table(physeq.scale))
  } else if (round == "round"){
    otu_table(physeq.scale) <- myround(otu_table(physeq.scale))
  }
  
  # Prune taxa and return new phyloseq object
  physeq.scale <- prune_taxa(taxa_sums(physeq.scale) > 0, physeq.scale)
  return(physeq.scale)
}

phyloseq_to_adonis <- function(physeq, distmat = NULL, dist = "bray", formula) {
  
  if(!is.null(distmat)) {
    phydist <- distmat
  } else {
    phydist <- phyloseq::distance(physeq, dist)
  }
  
  metadata <- as(sample_data(physeq), "data.frame")
  
  # Adonis test
  f <- reformulate(formula, response = "phydist")
  adonis.test <- adonis(f, data = metadata)
  print(adonis.test)

  # Run homogeneity of dispersion test if there is only 1 variable
  if (grepl("\\+", formula)) {
    l <- list(
      dist = phydist, 
      formula = f, 
      adonis = adonis.test
    )
  } else {
    group <- metadata[ ,formula]
    beta <- betadisper(phydist, group)
    disper.test = permutest(beta)
    print(disper.test)
    
    l <- list(
      dist = phydist, 
      formula = f, 
      adonis = adonis.test, 
      disper = disper.test
    )
  }
  return (l)
}
```

```{r}

core_hands_scale <- scale_reads(core_hands_merge, n=min(sample_sums(core_hands_merge)), round ="round")

# NMDS scores using bray-curtis distance 
set.seed(1)
bray.nmds <- ordinate(physeq = core_hands_merge, method = "NMDS", distance = "bray")

# Plot NMDS and color samples by sites
plot_ordination(
  physeq = core_hands_scale, 
  ordination = bray.nmds, 
  color = "Subject"
) + 
  geom_point(size = 3) + 
  ggtitle("Subject hand viral communities")

```
```{r}
adonis.site <- phyloseq_to_adonis(
  physeq = core_hands_scale, 
  dist = "bray", 
  formula = "Subject"
)
```


```{r}
#library( "DESeq2" )

#https://github.com/joey711/phyloseq/issues/387

diagdds = phyloseq_to_deseq2(core_total_merge, ~ Subject)
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
diagdds2 = DESeq(diagdds, fitType="local")
```


```{r}
res = results(diagdds2, cooksCutoff = FALSE)
res_IBK = res[order(res$padj, na.last = NA), ]
alpha = 0.05
keepOTUs_IBK_ulcer = rownames(res_IBK[res_IBK$padj < alpha, ])[1:68] #looking at top diff asvs between pre and post infection
kosticTrimvs_IBK_ulcer = prune_taxa(keepOTUs_IBK_ulcer, core_total_merge)
kosticTrimvs_IBK_ulcer
norm_desq <- transform_sample_counts(kosticTrimvs_IBK_ulcer , function(x) x / sum(x) )
```
```{r}
#kosticTrimvs_IBK_ulcer_HAND <- subset_samples(kosticTrimvs_IBK_ulcer, !(Location=="scalp"))

# Extract abundance matrix from the phyloseq object
DESEQ_core_OTU = as(otu_table(kosticTrimvs_IBK_ulcer), "matrix")
# transpose if necessary
DESEQ_core_OTU <- t(DESEQ_core_OTU)
# Coerce to data.frame
DESEQ_core_OTUdf = as.data.frame(DESEQ_core_OTU)

#merge metadata file with alpha diversity measures:

#Remove samples from metadatafile that were removed from your phyloseq object (ie control samples, repeated samples, and other samples that are of other persons you are not using in this data set (P33, >P43))
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")

DESEQ_Metadata_core_stat <- Metadata2[!grepl("HV_001_27", Metadata2$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_002_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_03", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_04", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_06", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_07", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_021_13", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_01", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_020_10", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_015_21", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_05", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_020_22", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_02", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_015_20", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_003_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_004_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_005_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_006_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_007_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_008_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_009_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_010_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_011_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_012_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_013_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_014_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_015_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_016_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_017_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_018_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_019_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_020_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_021_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_022_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_023_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_024_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_025_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_026_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_027_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_028_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_029_27", DESEQ_Metadata_core_stat$SampleID),]
DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("HV_030_27", DESEQ_Metadata_core_stat$SampleID),]
#DESEQ_Metadata_core_stat <- DESEQ_Metadata_core_stat[!grepl("scalp", DESEQ_Metadata_core_stat$Location),]
#alpha diversity measures:
DESEQ_Metadata_core_stat$shannon.vegan <- diversity(DESEQ_core_OTUdf, index = "shannon")
DESEQ_Metadata_core_stat$simpson.vegan <- diversity(DESEQ_core_OTUdf, index = "simpson")
DESEQ_Metadata_core_stat$invsimpson.vegan <- diversity(DESEQ_core_OTUdf, index = "invsimpson")
```
#ANOVA of alpha diversity measures using shannon (can be repeated for simpson)
```{r}
model_DESEQ_AOV <- aov(shannon.vegan~Gender+Subject/Location+ Subject/Time_Week+Subject, data = DESEQ_Metadata_core_stat)
summary(model_DESEQ_AOV)
```

```{r}
norm_DESEQ <- transform_sample_counts(kosticTrimvs_IBK_ulcer, function(x) x / sum(x) )
norm_DESEQ <- prune_samples(sample_sums(norm_DESEQ) >= 1, norm_DESEQ)
ord.DESEQ <- ordinate(norm_DESEQ, method="PCoA", distance="bray")
plot_ordination(norm_DESEQ, ord.DESEQ, color = "Subject", shape = "Time_Collected") +
  geom_point(size=1)
```
```{r}
metadata_DESEQ <- as(sample_data(norm_DESEQ ), "data.frame")
adonis(phyloseq::distance(norm_DESEQ, method="bray") ~  Gender+Subject/Location+ Subject/Time_Week+Subject,
       data = metadata_DESEQ)
```


```{r}
norm_kosticTrimvs_IBK_ulcer <-transform_sample_counts(kosticTrimvs_IBK_ulcer, function(x) x / sum(x)) 
test <- otu_table(norm_kosticTrimvs_IBK_ulcer) 
test <- as.data.frame(test)
dim(test)
test2 <- data.frame(t(test[-1]))
test2$SampleID <- row.names(test2)
test_overall <- merge(DESEQ_Metadata_core_stat,test2, by="SampleID")
test_overall2<-test_overall 
names(test_overall2)[names(test_overall2) == "Additional notes"] <- "Additional"
test_overall2 <-  subset(test_overall2, select = -c(SampleID, Raw_seq_name, File_Name, Time_Week, Index_Used, Control, DNase_trt, Month_Sampled, Day_Sampled, Year_Sampled, Date_Sampled, Gender, Seven_Day_Travel_Local, Seven_Day_Travel_Foreign, Time_Since_Last_Washed_Hands, Antibacterial_Soap_Used, Gel_Sanitizer_Use, Time_Since_Gel_Sanitizer, Hand_Lotion_Use, Time_Since_Lotion, Time_Since_Hair_Wash, Swam_in_Chlorine_Pool, Time_Since_Swim, Tanning_Bed_Use, Time_Since_tanning, Animal_Contact, Animal_Type, Additional, shannon.vegan, simpson.vegan, invsimpson.vegan))
```
```{r}
#library(ComplexHeatmap)
#library(circlize)
test_LH<-test_overall2[!(test_overall2$Location=="right"),]
test_LH<-test_LH[!(test_LH$Location=="scalp"),]
test_LH_T1<-test_LH[!(test_LH$Time_Collected=="T_0"),]
test_LH_T1<-test_LH_T1[!(test_LH_T1$Time_Collected=="1_month"),]
test_LH_T1<-test_LH_T1[!(test_LH_T1$Time_Collected=="3_month"),]
test_LH_T1<-test_LH_T1[!(test_LH_T1$Time_Collected=="6_month"),]
test_LH_T1 <-subset(test_LH_T1, select = -c(Location, Time_Collected))
row.names(test_LH_T1) <- test_LH_T1$Subject
test_LH_T1 <-subset(test_LH_T1, select = -c(Subject))
test_LH_T1 <- as.matrix(test_LH_T1)
test_LH_T1<- t(test_LH_T1)
col_fun = colorRamp2(c(0, 0.1), c("white", "#000033"))
col_fun2 = colorRamp2(c(0, 0.1), c("#000033", "#66CCFF"))
Heatmap(test_LH_T1, name = "Relative Abundance", column_title = "Subject", row_title = "Viral Contig", column_title_side = "bottom", row_dend_width = unit(2, "cm"), column_dend_height = unit(2, "cm"),  col = col_fun)
Heatmap(test_LH_T1, name = "Relative Abundance", column_title = "Subject", row_title = "Viral Contig", column_title_side = "bottom", row_dend_width = unit(2, "cm"), column_dend_height = unit(2, "cm"),  col = col_fun2)
Heatmap(test_LH_T1, name = "Relative Abundance", column_title = "Subject", row_title = "Viral Contig", column_title_side = "bottom", row_dend_width = unit(2, "cm"), column_dend_height = unit(2, "cm"))
```

###########____STEPWISE_REGRESSION___#################


#Arcsine square root transform of test sets
```{r}
#load packages 
#library(biom)
  #this no longer is avalible maybe use BIOM.utils or biomformat?
#library(dplyr)
```
```{r}
#Make directory
dir.create("stepwise_reg")

#Differential OTUs
differential_otus_subject<-  otu_table(kosticTrimvs_IBK_ulcer)
differential_otus_subject <- as.data.frame(differential_otus_subject)
differential_otus_subject <- add_rownames(differential_otus_subject, "OTUs")
differential_otus_core<-  otu_table(kosticTrimvs_IBK_ulcer)
differential_otus_core <- as.data.frame(differential_otus_core)
differential_otus_core <- add_rownames(differential_otus_core, "OTUs")

#Heifer trans arcsine transformation
otu_test_subject <- otu_table(VCPHY1)
otu_test_subject<- as.data.frame(otu_test_subject)
test_subject <- sweep(otu_test_subject, 2, colSums(otu_test_subject), FUN = "/")
otu_test_core <- otu_table(core_total_merge)
otu_test_core <- as.data.frame(otu_test_core)
test_core <- sweep(otu_test_core, 2, colSums(otu_test_core), FUN = "/")

#arcsine square root transformation subject
trans.arcsine <- function(x){asin(sign(x) * sqrt(abs(x)))}
test_subject[, 1:612] <- as.data.frame(lapply(test_subject[, 1:612], FUN = function(x) {sapply(x, FUN = trans.arcsine)}))
  # number of samples = 612
test_subject <- add_rownames(test_subject, "OTUs")
test_subject_filtered <- test_subject[which(test_subject$OTUs %in% differential_otus_subject$OTUs),]
test_subject_trans <- as.data.frame(t(test_subject_filtered), stringsAsFactors=F)
colnames(test_subject_trans) <- test_subject_trans[1, ]
test_subject_trans <- test_subject_trans[-1, ]
colnames(test_subject_trans) <- paste("OTU", colnames(test_subject_trans), sep="")
test_subject_final <- add_rownames(test_subject_trans, "SampleID")
write.table(test_subject_final, file="stepwise_reg/test_subject_final.txt", sep="\t", col.names=T, row.names=F, quote=F)


#arcsine square root transformation core
trans.arcsine <- function(x){asin(sign(x) * sqrt(abs(x)))}
test_core[, 1:612] <- as.data.frame(lapply(test_core[, 1:612], FUN = function(x) {sapply(x, FUN = trans.arcsine)}))
  # number of samples = 612
test_core <- add_rownames(test_core, "OTUs")
test_core_filtered <- test_core[which(test_core$OTUs %in% differential_otus_core$OTUs),]
test_core_trans <- as.data.frame(t(test_core_filtered), stringsAsFactors=F)
colnames(test_core_trans) <- test_core_trans[1, ]
test_core_trans <- test_core_trans[-1, ]
colnames(test_core_trans) <- paste("OTU", colnames(test_core_trans), sep="")
test_core_final <- add_rownames(test_core_trans, "SampleID")
write.table(test_core_final, file="stepwise_reg/test_core_final.txt", sep="\t", col.names=T, row.names=F, quote=F)

#Validation set Subject
validation_subject <- test_steer[which(test_steer$OTUs %in% differential_otus_heifer$V1),]

#Heifer validation Core
validation_core <- test_steer[which(test_steer$OTUs %in% differential_otus_heifer$V1),]

```

##############_____DESEQ_TEST_POPULATIONS___############

```{r}
#Left Hand
core_LH_merge <- merge_phyloseq(core_P01LH, core_P02LH, core_P03LH, core_P04LH, core_P05LH, core_P06LH, core_P07LH, 
                            core_P08LH, core_P09LH, core_P10LH, core_P11LH, core_P12LH, core_P13LH, core_P14LH,
                            core_P15LH, core_P16LH, core_P17LH, core_P18LH, core_P19LH, core_P20LH, core_P21LH,
                            core_P22LH, core_P23LH, core_P24LH, core_P25LH, core_P26LH, core_P27LH, core_P28LH,
                            core_P29LH, core_P30LH, core_P31LH, core_P32LH, core_P34LH, core_P35LH, core_P36LH,
                            core_P37LH, core_P38LH, core_P39LH, core_P40LH, core_P41LH, core_P42LH, core_P43LH)

#Right Hand
core_RH_merge <- merge_phyloseq(core_P01RH, core_P02RH, core_P03RH, core_P04RH, core_P05RH, core_P06RH, core_P07RH, 
                            core_P08RH, core_P09RH, core_P10RH, core_P11RH, core_P12RH, core_P13RH, core_P14RH,
                            core_P15RH, core_P16RH, core_P17RH, core_P18RH, core_P19RH, core_P20RH, core_P21RH,
                            core_P22RH, core_P23RH, core_P24RH, core_P25RH, core_P26RH, core_P27RH, core_P28RH,
                            core_P29RH, core_P30RH, core_P31RH, core_P32RH, core_P34RH, core_P35RH, core_P36RH,
                            core_P37RH, core_P38RH, core_P39RH, core_P40RH, core_P41RH, core_P42RH, core_P43RH)
```
#left hand test population
```{r}
diagdds_LH = phyloseq_to_deseq2(core_LH_merge, ~ Subject)
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds_LH), 1, gm_mean)
diagdds_LH = estimateSizeFactors(diagdds_LH, geoMeans = geoMeans)
diagdds2_LH = DESeq(diagdds_LH, fitType="local")
```

```{r}
resLH = results(diagdds2_LH, cooksCutoff = FALSE)
res_LH = resLH[order(resLH$padj, na.last = NA), ]
alpha = 0.05
keepOTUs_LH = rownames(res_LH[res_LH$padj < alpha, ])
write_csv(keepOTUs_LH, 'VCAnntations.csv')
kosticTrimvs_LH = prune_taxa(keepOTUs_LH, core_LH_merge)
kosticTrimvs_LH
norm_desq_LH <- transform_sample_counts(kosticTrimvs_LH , function(x) x / sum(x) )
```


#right hand test population
```{r}
diagdds_RH = phyloseq_to_deseq2(core_RH_merge, ~ Subject)
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds_RH), 1, gm_mean)
diagdds_RH = estimateSizeFactors(diagdds_RH, geoMeans = geoMeans)
diagdds2_RH = DESeq(diagdds_RH, fitType="local")
```

```{r}
resRH = results(diagdds2_RH, cooksCutoff = FALSE)
res_RH = resRH[order(resRH$padj, na.last = NA), ]
alpha = 0.05
keepOTUs_RH = rownames(res_RH[res_RH$padj < alpha, ])
kosticTrimvs_RH = prune_taxa(keepOTUs_RH, core_RH_merge)
kosticTrimvs_RH
otu_tablekosticTrimvs_RH
norm_desq_RH <- transform_sample_counts(kosticTrimvs_RH , function(x) x / sum(x) )


```

```{r}
#Make directory
#dir.create("stepwise_reg")

#Differential OTUs
differential_otus_LH<-  otu_table(kosticTrimvs_LH)
differential_otus_LH <- as.data.frame(differential_otus_LH)
differential_otus_LH <- add_rownames(differential_otus_LH, "OTUs")
differential_otus_RH<-  otu_table(kosticTrimvs_RH)
differential_otus_RH <- as.data.frame(differential_otus_RH)
differential_otus_RH <- add_rownames(differential_otus_RH, "OTUs")

#Heifer LH arcsine transformation
otu_test_LH <- otu_table(core_LH_merge)
otu_test_LH<- as.data.frame(otu_test_LH)
test_LH <- sweep(otu_test_LH, 2, colSums(otu_test_LH), FUN = "/")
otu_test_RH <- otu_table(core_RH_merge)
otu_test_RH <- as.data.frame(otu_test_RH)
test_RH <- sweep(otu_test_RH, 2, colSums(otu_test_RH), FUN = "/")

#arcsine square root transformation LH
trans.arcsine <- function(x){asin(sign(x) * sqrt(abs(x)))}
test_LH[, 1:206] <- as.data.frame(lapply(test_LH[, 1:206], FUN = function(x) {sapply(x, FUN = trans.arcsine)}))
  # number of samples = 206
test_LH <- add_rownames(test_LH, "OTUs")
test_LH_filtered <- test_LH[which(test_LH$OTUs %in% differential_otus_LH$OTUs),]
test_LH_trans <- as.data.frame(t(test_LH_filtered), stringsAsFactors=F)
colnames(test_LH_trans) <- test_LH_trans[1, ]
test_LH_trans <- test_LH_trans[-1, ]
colnames(test_LH_trans) <- paste("OTU", colnames(test_LH_trans), sep="")
test_LH_final <- add_rownames(test_LH_trans, "SampleID")
write.table(test_LH_final, file="stepwise_reg/test_LH_final.txt", sep="\t", col.names=T, row.names=F, quote=F)


#arcsine square root transformation RH,
trans.arcsine <- function(x){asin(sign(x) * sqrt(abs(x)))}
test_RH[, 1:199] <- as.data.frame(lapply(test_RH[, 1:199], FUN = function(x) {sapply(x, FUN = trans.arcsine)}))
  # number of samples = 199
test_RH <- add_rownames(test_RH, "OTUs")
test_RH_filtered <- test_RH[which(test_RH$OTUs %in% differential_otus_RH$OTUs),]
test_RH_trans <- as.data.frame(t(test_RH_filtered), stringsAsFactors=F)
colnames(test_RH_trans) <- test_RH_trans[1, ]
test_RH_trans <- test_RH_trans[-1, ]
colnames(test_RH_trans) <- paste("OTU", colnames(test_RH_trans), sep="")
test_RH_final <- add_rownames(test_RH_trans, "SampleID")
write.table(test_RH_final, file="stepwise_reg/test_RH_final.txt", sep="\t", col.names=T, row.names=F, quote=F)

#Left Hand Validation set 
validation_LH <- test_RH[which(test_RH$OTUs %in% differential_otus_LH$OTUs),]
validation_LH_trans <- as.data.frame(t(validation_LH), stringsAsFactors=F)
colnames(validation_LH_trans) <- validation_LH_trans[1, ]
validation_LH_trans = validation_LH_trans[-1, ]
colnames(validation_LH_trans) <- paste("OTU", colnames(validation_LH_trans), sep="")
validation_LH_final <- add_rownames(validation_LH_trans, "SampleID")
write.table(validation_LH_final, file="stepwise_reg/validation_LH_final.txt", sep="\t", col.names=T, row.names=F, quote=F)


#Right Hand validation set
validation_RH <- test_LH[which(test_LH$OTUs %in% differential_otus_RH$OTUs),]
validation_RH_trans <- as.data.frame(t(validation_RH), stringsAsFactors=F)
colnames(validation_RH_trans) = validation_RH_trans[1, ]
validation_RH_trans = validation_RH_trans[-1, ]
colnames(validation_RH_trans) <- paste("OTU", colnames(validation_RH_trans), sep="")
validation_RH_final <- add_rownames(validation_RH_trans, "SampleID")
write.table(validation_RH_final, file="stepwise_reg/validation_RH_final.txt", sep="\t", col.names=T, row.names=F, quote=F)

```

#Forward stepwise regression for Subject
```{r}
#Load packages
#library(car)

#Create data set for Left Hand cohort 
mapping_LH <- sample_data(core_LH_merge)
mapping_LH <- data.frame(sample_data(core_LH_merge))
row.names(mapping_LH) <- mapping_LH[,1]
View(mapping_LH)
mapping_LH2 <- subset(mapping_LH, select = c(SampleID, Subject, Time_Week))
#Linear models for Time_Collected, Gender, and Subject using ___ as covariates for Left Hand cohort

lm_Subject_LH <- lm(Subject ~ Time_Week, data=mapping_LH2)


```




```{r}
markers <- c("k127_9099273", "k127_807185", "k127_11692508", "k127_9484408", "k127_2554829", "k127_5846858", "k127_3612048", "k127_5791511", "k127_7492625", "k127_2293794")

tax_table(VCPHY2)
VCPHY2


fin1 <- subset_taxa(VCPHY2, contig_id2 == "k127_9099273")
fin2 <- subset_taxa(VCPHY2, contig_id2 == "k127_807185")
fin3 <- subset_taxa(VCPHY2, contig_id2 == "k127_11692508")  
fin4 <- subset_taxa(VCPHY2, contig_id2 == "k127_9484408") 
fin5 <- subset_taxa(VCPHY2, contig_id2 == "k127_2554829") 
fin6 <- subset_taxa(VCPHY2, contig_id2 == "k127_5846858")
fin7 <- subset_taxa(VCPHY2, contig_id2 == "k127_3612048")
fin8 <- subset_taxa(VCPHY2, contig_id2 == "k127_5791511")
fin9 <- subset_taxa(VCPHY2, contig_id2 == "k127_7492625")
fin10 <- subset_taxa(VCPHY2, contig_id2 == "k127_2293794")

final <- merge_phyloseq(fin1, fin2, fin3, fin4, fin5, fin6, fin7, fin8, fin9, fin10)

final01 <- subset_samples(VCPHY2, Subject=="P01")
   otu_table(final01)                         
final42 <- subset_samples(final, Subject=="P42")
   otu_table(final42)                           






```


```{r}
#http://madsalbertsen.github.io/ampvis/examples/ampvis_guide.html
#library(ampvis)
ds<-ANPHY1
dsn <- transform_sample_counts(ANPHY1, function(x) x / sum(x) * 100)
dsr <- rarefy_even_depth(ds, sample.size = 10000, rngseed = 712)
dsf <- filter_taxa(ds, function(x) max(x) >= 10, TRUE)
dsf

amp_rabund(data = dsn,
           tax.aggregate = "Family",
           tax.add = "Phylum",
           scale.seq = 100)
```

```{r}
test <- subset_taxa(P_02, contig_id  == "HV_021_04_k141_17594")

plot_bar(test)
```

```{r}

prevelancedf <- apply(X = otu_table(P_01_LH),
                     MARGIN = 1,
                     FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf <- data.frame(Prevalence = prevelancedf,
                          TotalAbundance = taxa_sums(P_01_LH),
                          tax_table(P_01_LH))
prevelancedf[1:10,]
prevelancedf_stable <- subset(prevelancedf, prevelancedf$Prevalence >= 4)  
```

```{r}
#Subset by person 01
P_01 <- subset_samples(ANPHY1, Subject=="P01")
#Subset person 01 by locations
P_01_LH<- subset_samples(P_01, Location=="left")
P_01_RH<- subset_samples(P_01, Location=="right")
P_01_SC<- subset_samples(P_01, Location=="scalp")
#Find taxa present across 80% of the repeated time points for each location for person 01 
core_P01LH <- filter_taxa(P_01_LH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P01RH <- filter_taxa(P_01_RH, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
core_P01SC <- filter_taxa(P_01_SC, function(x) sum(x > 0) >= (0.8*length(x)), prune = TRUE)
#merge core locations for person 01 together to make an overall core for person 01
core_P01 <- merge_phyloseq(core_P01LH , core_P01RH, core_P01SC)
#Subtract out overall CORE to get just unique core taxa for person 01 
core_rm<- rownames(otu_table(CORE))
core_P01_rm <- taxa_names(core_P01)
core_P01_rm <- core_P01_rm[!(core_P01_rm %in% core_rm)]
core_P01_rm <- prune_taxa(core_P01_rm, core_P01)

core_SC_rm<- rownames(otu_table(CORE_SC))
core_SC_P01_rm <- taxa_names(core_P01SC)
core_SC_P01_rm <- core_SC_P01_rm[!(core_SC_P01_rm %in% core_SC_rm)]
core_SC_P01_rm <- prune_taxa(core_SC_P01_rm, core_P01SC)

core_LH_rm<- rownames(otu_table(CORE_LH))
core_LH_P01_rm <- taxa_names(core_P01LH)
core_LH_P01_rm <- core_LH_P01_rm[!(core_LH_P01_rm %in% core_LH_rm)]
core_LH_P01_rm <- prune_taxa(core_LH_P01_rm, core_P01LH)

core_RH_rm<- rownames(otu_table(CORE_RH))
core_RH_P01_rm <- taxa_names(core_P01RH)
core_RH_P01_rm <- core_RH_P01_rm[!(core_RH_P01_rm %in% core_RH_rm)]
core_RH_P01_rm <- prune_taxa(core_RH_P01_rm, core_P01RH)


core_HANDS_rm<- rownames(otu_table(CORE_HANDS))
core_P01_hands <- merge_phyloseq(core_P01LH , core_P01RH)
core_HANDS_P01_rm <- taxa_names(core_P01_hands)
core_HANDS_P01_rm <- core_HANDS_P01_rm[!(core_HANDS_P01_rm %in% core_HANDS_rm)]
core_HANDS_P01_rm <- prune_taxa(core_HANDS_P01_rm, core_P01_hands)
```
