---
title: "dada2_analysis"
output: html_document
---

```{r setup, include=FALSE}
library(NRES803)
library(tidyverse)
library(GGally)
library(readr)
library(mgcv)
library(broom)
library(gridExtra)
library(lme4)
library(lmerTest)
library(broom.mixed)
library(readr)
library(phyloseq)
library(dplyr)
library(tidyr)
library(ggplot2)
library(phyloseq)
library(tidyverse)
library(microbiome)
library(readr)
library("ggpubr")
library("tidyverse")
library(vegan)
library(ComplexHeatmap)
library(gridBase)
library(metagenomeSeq)
library(bbmle)
library(glmmTMB)
library(circlize)
library(dada2)
```

```{r}
path <- "/Volumes/ermc_drive/Ermc_run2"
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

```{r}
plotQualityProfile(fnFs[1:2])
```

```{r}
plotQualityProfile(fnRs[1:2])
```

```{r}
#quality plot (foward)
plotQualityProfile(fnFs[1:10])
```

```{r}
#The first two reverse reads:
plotQualityProfile(fnRs[1:5])
```

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
```

#filter reads
```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(200,140),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)
head(out)
```

```{r}
saveRDS(out, "out.RDS")
write.table(out, file="out.txt", col.names=T, row.names=T, sep = "\t",quote=F)
```

#Data Statistics after Trimming
```{r ,echo=TRUE}
sum(out[,1]) #total reads in 619345
sum(out[,2]) #total reads out 289025
sum(out[,1]) - sum(out[,2]) #reads lost 330320
sum(out[,2])/sum(out[,1]) # percentage data retained 46.67%
```


```{r}
#to avoid error, due to very low reads
exists <- file.exists(filtFs) & file.exists(filtRs)
filtFs <- filtFs[exists]
filtRs <- filtRs[exists]
#Dereplication
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
#names(derepFs) <- sample.names
#names(derepRs) <- sample.names
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF)
plotErrors(errR)
save(exists, filtFs, filtRs, derepFs, derepRs, errF, errR, file = "runq30.rds")
```

```{r}
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
dadaFs[[1]]
```

```{r}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=T)
head(mergers[[1]])
```

#Make ASV table
```{r}
seqtabAll <- makeSequenceTable(mergers)
dim(seqtabAll) #166 361
table(nchar(getSequences(seqtabAll)))
write.table(seqtabAll, file="seqtabAll.txt", col.names=T, row.names=T, sep = "\t",quote=F)

#remive chimeras
seqtabNoC <- removeBimeraDenovo(seqtabAll)
write.table(seqtabNoC, file="seqtabNoC.txt", col.names=NA, row.names=T, sep = "\t",quote=F)
saveRDS(seqtabNoC, "SEQTAB_nOc.RDS")
SEQTAB <- readRDS("SEQTAB_nOc.RDS")
dim(SEQTAB)
dim(seqtabNoC) #166 306
sum(seqtabNoC)/sum(seqtabAll) #99.75%
t_seqtabNoC= t(seqtabNoC)
dim(t_seqtabNoC)
table(nchar(getSequences(SEQTAB)))
```

```{r}
asv_seqs <- colnames(seqtabNoC)
asv_headers <- vector(dim(seqtabNoC)[2], mode="character")
for (i in 1:dim(seqtabNoC)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep="_")
}

# making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "ASVs.fa")

# count table:
asv_tab <- t(seqtabNoC)
row.names(asv_tab) <- sub(">", "", asv_headers)
asv_tab2<-asv_tab
for ( col in 1:ncol(asv_tab2)){
    colnames(asv_tab2)[col] <-  sub("_F.*", "", colnames(asv_tab2)[col])
}
write.table(asv_tab2, "ASVs_counts.txt", sep="\t", quote=F)
```
#Phyloseq object!
```{r}
Metadata <- read_csv("Sequencing_Data_Info_Sheet.csv")
Metadata3 <- Metadata
Metadata3$Time <- Metadata3$Time_Collected
Metadata3$Time[which(Metadata3$Time == "T_0")] <- "00_Weeks"
Metadata3$Time[which(Metadata3$Time == "2_week")] <- "02_Weeks"
Metadata3$Time[which(Metadata3$Time == "1_month")] <- "04_Weeks"
Metadata3$Time[which(Metadata3$Time == "3_month")] <- "12_Weeks"
Metadata3$Time[which(Metadata3$Time == "6_month")] <- "24_Weeks"
#Metadata3 <-Metadata3[!grepl("HV_003_15", Metadata3$SampleID),]
META <- sample_data(Metadata3)
sample_names(META)= META$PRIMER
#asv_tab2 <- read_delim("ASVs_counts.txt", 
    #"\t", escape_double = FALSE, trim_ws = TRUE,row.names = 1)
asv_tab2 <- read_delim("ASVs_counts.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
counts_tax<-as.data.frame(asv_tab2)
#counts_tax <- tibble::rownames_to_column(asv_tab3, "phylum")
#counts_tax$contig <- counts_tax$phylum
#counts_tax <- subset(counts_tax, select = c(contig,phylum))
#write_csv(counts_tax, 'contig_list.csv')
contigs_list <- read.csv('contig_list.csv', header = TRUE, stringsAsFactors = FALSE, row.names = 1)
contigs_list <- as.matrix(contigs_list)
TAX <- tax_table(contigs_list)
asv_tab2 <- read_delim("ASVs_counts.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
asv_tab2<- as.data.frame(asv_tab2)
row.names(asv_tab2) <- asv_tab2$phylum
asv_tab2 <- subset(asv_tab2, select = -c(phylum))
for ( col in 1:ncol(asv_tab2)){
    colnames(asv_tab2)[col] <-  sub("_F.*", "", colnames(asv_tab2)[col])
}

OTU<-otu_table(asv_tab2, taxa_are_rows=TRUE)


PS<-phyloseq(OTU,META,TAX)


Samples_to_keep<-c("ASV_1","ASV_2","ASV_3","ASV_4","ASV_5","ASV_6","ASV_7","ASV_8","ASV_9","ASV_10","ASV_11","ASV_12","ASV_13","ASV_14","ASV_15","ASV_16","ASV_17","ASV_18","ASV_19","ASV_20","ASV_21","ASV_22","ASV_24","ASV_25","ASV_27","ASV_30","ASV_31","ASV_33","ASV_34","ASV_36","ASV_37","ASV_38","ASV_39","ASV_40","ASV_41","ASV_46","ASV_51","ASV_54","ASV_58","ASV_59","ASV_61","ASV_62","ASV_64","ASV_66","ASV_70","ASV_71","ASV_72","ASV_73","ASV_79","ASV_84","ASV_86","ASV_90","ASV_92","ASV_96","ASV_101","ASV_104","ASV_106","ASV_107","ASV_111","ASV_117","ASV_120","ASV_121","ASV_125","ASV_128","ASV_131","ASV_132","ASV_135","ASV_149","ASV_150","ASV_157","ASV_167","ASV_168","ASV_169","ASV_170","ASV_212","ASV_256","ASV_292","ASV_298","ASV_306")

allTaxa<-taxa_names(PS)
allTaxa <- allTaxa[(allTaxa %in% Samples_to_keep)]
PS1 = prune_taxa(allTaxa, PS)
Samples_toRemove <- c("288", "285", "286", "287")

PS2 <- subset_samples(PS1, !(PRIMER %in% Samples_toRemove))

#norm_PS <- transform_sample_counts(PS, function(x) x / sum(x) )
#allTaxa<-taxa_names(norm_PS)
#allTaxa <- allTaxa[(allTaxa %in% Samples_to_keep)]
#PS1 = prune_taxa(allTaxa, norm_PS)
#Samples_toRemove <- c("288", "285", "286", "287")
#PS2 <- subset_samples(PS1, !(PRIMER %in% Samples_toRemove))
```
PS2 is output phyloseq object

```{r}
PS2_prune= prune_samples(sample_sums(PS2)>=1, PS2)
norm_groups <- transform_sample_counts(PS2_prune, function(x) x / sum(x) )
ord.core_groups.pcoa <- ordinate(norm_groups, method="PCoA", distance="jaccard", na.rm = TRUE, binary = TRUE)
jaccard_p<-plot_ordination(norm_groups, ord.core_groups.pcoa, color = "Subject", shape = "Location") +
  geom_point(size=1)
ord.core_groups.pcoa <- ordinate(norm_groups, method="PCoA", distance="bray", na.rm = TRUE)
bray_p<-plot_ordination(norm_groups, ord.core_groups.pcoa, color = "Subject", shape = "Location") +
  geom_point(size=1)
jaccard_p
bray_p
#ggsave(plot=p,"Figures/beta.pdf",dpi = 600, width = 9,
  #height = 6, units="in",pointsize=12)
```

```{r}
metadata <- as(sample_data(norm_groups), "data.frame")
adonis(phyloseq::distance(norm_groups, method="jaccard", binary = TRUE) ~  Subject+Subject:Location,
       data = metadata)
```
```{r}
metadata <- as(sample_data(norm_groups), "data.frame")
adonis(phyloseq::distance(norm_groups, method="bray", binary = TRUE) ~  Subject+Subject:Location,
       data = metadata)
```
```{r}
damage <- read.csv("damage_snp.csv", quote = "", header = TRUE, stringsAsFactors = FALSE,row.names=1)


damage <- as.matrix(damage)
damage<- tax_table(damage)
PS_damage<-phyloseq(OTU,META,damage)


Samples_to_keep<-c("ASV_1","ASV_2","ASV_3","ASV_4","ASV_5","ASV_6","ASV_7","ASV_8","ASV_9","ASV_10","ASV_11","ASV_12","ASV_13","ASV_14","ASV_15","ASV_16","ASV_17","ASV_18","ASV_19","ASV_20","ASV_21","ASV_22","ASV_24","ASV_25","ASV_27","ASV_30","ASV_31","ASV_33","ASV_34","ASV_36","ASV_37","ASV_38","ASV_39","ASV_40","ASV_41","ASV_46","ASV_51","ASV_54","ASV_58","ASV_59","ASV_61","ASV_62","ASV_64","ASV_66","ASV_70","ASV_71","ASV_72","ASV_73","ASV_79","ASV_84","ASV_86","ASV_90","ASV_92","ASV_96","ASV_101","ASV_104","ASV_106","ASV_107","ASV_111","ASV_117","ASV_120","ASV_121","ASV_125","ASV_128","ASV_131","ASV_132","ASV_135","ASV_149","ASV_150","ASV_157","ASV_167","ASV_168","ASV_169","ASV_170","ASV_212","ASV_256","ASV_292","ASV_298","ASV_306")

allTaxa<-taxa_names(PS_damage)
allTaxa <- allTaxa[(allTaxa %in% Samples_to_keep)]
PS1_damage = prune_taxa(allTaxa, PS_damage)
Samples_toRemove <- c("288", "285", "286", "287")
PS2_damage <- subset_samples(PS1_damage, !(PRIMER %in% Samples_toRemove))


```

```{r}
PS2_damage_prune <- prune_samples(sample_sums(PS2_damage)>=1, PS2_damage)
norm_groups <- transform_sample_counts(PS2_damage_prune, function(x) x / sum(x) )


ord.core_groups.pcoa <- ordinate(norm_groups, method="PCoA", distance="jaccard", na.rm = TRUE, binary = TRUE)

jaccard_p<-plot_ordination(norm_groups, ord.core_groups.pcoa,type="taxa", color = "Del_Mutagenesis") +
  geom_point(size=1)

ord.core_groups.pcoa <- ordinate(norm_groups, method="PCoA", distance="bray", na.rm = TRUE)
bray_p<-plot_ordination(norm_groups, ord.core_groups.pcoa, type="taxa", color = "Del_Mutagenesis") +
  geom_point(size=1)
jaccard_p
bray_p
```
```{r}
sample_data(PS2)
subject_test<-subset(sample_data(PS2), select = c(Subject))
subject_test$contigid<-row.names(subject_test)
row.names(subject_test)=NULL
subject_test[grep("P01", subject_test$Subject), ]
unique(subject_test$Subject)
```

```{r,echo=FALSE}
P01<-c("211","214") 
P02<-c() 
P03<-c()
P04<-c()
P05<-c()
P08<-c()
P09<-c()
P10<-c()
P11<-c()
P13<-c()
P14<-c()
P16<-c()
P18<-c()
P20<-c()
P21<-c()
P22<-c()
P23<-c()
P24<-c()
P25<-c()
P29<-c()
P31<-c()
P36<-c()
P37<-c()
P42<-c()
P43<-c()
P44<-c()
P46<-c()
P47<-c()
P49<-c()
```
















####################################################
```{r}
HMap <- otu_table(PS2) 
HMap <- as.data.frame(HMap)
#HMap[] <- lapply(HMap, function(x) ifelse(x<50, 0, x))
HMAP_meta<-sample_data(PS2)
HMAP_meta <- as.data.frame(HMAP_meta)
HMAP2 <- data.frame(t(HMap[-1]))
HMAP2$SampleID <- row.names(HMAP2)
HMAP_overall <- merge(HMAP_meta,HMAP2, by="row.names")
HMAP_overall2<-HMAP_overall 
HMAP_overall2 <-  subset(HMAP_overall2, select = -c(PLATE, Raw_seq_name, File_Name, Time_Week, Index_Used, Control, DNase_trt, Month_Sampled, Day_Sampled, Year_Sampled, Date_Sampled, Gender, X7_Day_Travel_Local, X7_Day_Travel_Foreign, Time_Since_Last_Washed_Hands, Antibacterial_Soap_Used, Gel_Sanitizer_Use, Time_Since_Gel_Sanitizer, Hand_Lotion_Use, Time_Since_Lotion, Time_Since_Hair_Wash, Swam_in_Chlorine_Pool, Time_Since_Swim, Tanning_Bed_Use, Time_Since_tanning, Animal_Contact, Animal_Type, Additional.notes))
row.names(HMAP_overall2)<- HMAP_overall2$Row.names
```
```{r}
library(ComplexHeatmap)
library(circlize)
HMAP_Subject<-HMAP_overall2
HMAP_Subject <-  subset(HMAP_Subject, select = -c(PRIMER, Row.names, Location, Time_Collected, Time))
library(plyr)
HMAP_Subject2<- ddply(HMAP_Subject,"Subject",numcolwise(sum))
row.names(HMAP_Subject2) <- HMAP_Subject2$Subject
HMAP_Subject2 <-subset(HMAP_Subject2, select = -c(Subject))
HMAP_Subject3 <- as.matrix(HMAP_Subject2)
HMAP_Subject3<- t(HMAP_Subject3)
Heatmap(HMAP_Subject3, name = "Relative Abundance", column_title = "Subject", row_title = "Viral Contig", column_title_side = "bottom", row_dend_width = unit(2, "cm"), column_dend_height = unit(2, "cm"))
#View(HMAP_Subject$ASV_303)
```


```{r}
plot_richness(PS2, x="Subject", measures=c("Shannon", "Simpson"), color="Time_Collected")
plot_richness(PS2, x="Time_Collected", measures=c("Shannon", "Simpson"), color="Subject")
```

```{r}
norm_core_P01 <- transform_sample_counts(PS2, function(x) x / sum(x) )
ord.core_01.pcoa <- ordinate(norm_core_P01, method="PCoA", distance="bray")
plot_ordination(norm_core_P01, ord.core_01.pcoa, color = "Location", shape = "Time_Collected") +
  geom_point(size=5)

```

```{r}
library(ggtree)
library(treeio)
library(ggplotify)

tree<-read.iqtree("guest/210413005502/Tree_ermc_final.phy.treefile")

#plot tree with no editing/additions
ggplot(tree, aes(x, y)) + geom_tree() + theme_tree()
#plot tree with colors and dotted lines
ggtree(tree, color="firebrick", size=2, linetype="dotted")
#plot tree to turn off ladderize form
ggtree(tree, ladderize=FALSE)
#to turn off branch lenght
ggtree(tree, branch.length="none")

```

```{r}
#to see the tree in different layouts to choose the best one:
set.seed(2017-02-16)
tree <- rtree(50)
ggtree(tree)
ggtree(tree, layout="rectangular")
ggtree(tree, layout="slanted")
ggtree(tree, layout="radial")
ggtree(tree, layout="circular")
ggtree(tree, layout="fan", open.angle=120)
ggtree(tree, layout="equal_angle")
ggtree(tree, layout="daylight")
ggtree(tree, branch.length='none')
ggtree(tree, branch.length='none', layout='circular')
ggtree(tree, layout="daylight", branch.length = 'none')
```

```{r}
ggtree(tree) + scale_x_reverse()
ggtree(tree) + coord_flip()
ggtree(tree) + layout_dendrogram()
ggtree(tree, layout='slanted') + coord_flip()
ggtree(tree, layout='slanted', branch.length='none') + layout_dendrogram()
ggtree(tree, layout='circular') + xlim(-10, NA)
#ggtree(tree) + layout_inward_circular()
#ggtree(tree) + layout_inward_circular(xlim=15)
```

```{r}
#adds a tree scale
ggtree(tree) + geom_treescale()
```

```{r}
#set annotations
#tree$tip.label <- paste0('to make the label longer_', tree$tip.label)
p1 <- ggtree(tree) + geom_tiplab()

```

```{r}
tree2<-read.iqtree("guest/210413005502/Tree_ermc_final_asvonly.phy.treefile")
p1 <- ggtree(tree2) + geom_tiplab()

HP<-PS2
HP<- prune_taxa(taxa_sums(HP)>10,HP)


HMAP_Subject2 
HMAP_Subject3

nodeids <- nodeid(tree, tree$node.label[nchar(tree$node.label)>0])


library(ggtreeExtra)
library(ggtree)
library(treeio)
library(tidytree)
library(ggstar)
library(ggplot2)
library(ggnewscale)

tree2<-read.iqtree("guest/210413005502/Tree_ermc_final_asvonly.phy.treefile")
dat1<- HMAP_Subject2
HMAP_Subject3<- t(HMAP_Subject3)


nodeids <- nodeid(tree2, tree2@phylo[["node.label"]][nchar(tree2@phylo[["node.label"]])>4])
nodedf <- data.frame(node=nodeids)
nodelab <- gsub("[\\.0-9]", "", tree2@phylo[["node.label"]][nchar(tree2@phylo[["node.label"]])>4])


tree3 <- read.tree("guest/hmptree.nwk")
dat1 <- read.csv("guest/tippoint_attr.csv")
dat2 <- read.csv("guest/ringheatmap_attr.csv")
dat3 <- read.csv("guest/barplot_attr.csv")
```

```{r}
dat1A<- t(HMAP_Subject2)
dat1A <- subset(HMAP_Subject, select = -c(SampleID))
dat1A<- pivot_longer(dat1A, cols =2:80, names_to = "ID", values_to = "Abundance"  )
dat1A<-as.data.frame(dat1A)
col_order <- c("ID", "Subject", "Abundance")
dat1A <- dat1A[, col_order]
dat2A<-HMAP_overall 
dat2A <-  subset(dat2A, select = -c(PLATE, Raw_seq_name, File_Name, Time_Week, Index_Used, Control, DNase_trt, Month_Sampled, Day_Sampled, Year_Sampled, Date_Sampled, Gender, X7_Day_Travel_Local, X7_Day_Travel_Foreign, Time_Since_Last_Washed_Hands, Antibacterial_Soap_Used, Gel_Sanitizer_Use, Time_Since_Gel_Sanitizer, Hand_Lotion_Use, Time_Since_Lotion, Time_Since_Hair_Wash, Swam_in_Chlorine_Pool, Time_Since_Swim, Tanning_Bed_Use, Time_Since_tanning, Animal_Contact, Animal_Type, Additional.notes, SampleID))
dat2A <-  subset(dat2A, select = -c(PRIMER, Row.names,Time_Collected))
dat2A <- pivot_longer(dat2A, cols =4:82, names_to = "ID", values_to = "Abundance"  )
names(dat1A)[names(dat1A) == "ID"] <- "label"
names(dat2A)[names(dat2A) == "ID"] <- "label"
p1 <- ggtree(tree2) + geom_tiplab() + geom_nodelab()
#p1 <- p1 %<+% dat1A + geom_tippoint(aes(color=Subject))
p <- ggtree(tree2)
p <- p %<+%  geom_facet(panel = "Subject", data = dat1A, geom = ggstance::geom_barh, 
                aes(x = dummy_bar_value, fill = Subject), 
                stat = "identity", width = .6) +
    theme_tree2(legend.position=c(.05, .85))
library(ggnewscale)
p <- ggtree(tree2) + geom_tiplab(align=TRUE)
HMAP_Subject3

library(ggridges)
p <- ggtree(tree2) + geom_tiplab()

gheatmap(tree2, dat1A[, "Subject", drop = FALSE], width = 0.1, color = "black") +
  scale_fill_igv(palette = c("default"),
                           guide=guide_legend(keywidth = 0.3, keyheight = 0.3, order=4))+
         geom_treescale(fontsize=2, linesize=0.3, x=4.9, y=0.1) +
         theme(legend.position=c(0.93, 0.5),
               legend.background=element_rect(fill=NA),
               legend.title=element_text(size=6.5),
               legend.text=element_text(size=4.5),
               legend.spacing.y = unit(0.02, "cm"),
             )



dat2A

dat3A <- subset(dat2A, select = -c(Subject))
#> Scale for 'fill' is already present. Adding another scale for 'fill',
#> which will replace the existing scale.
p<- ggtree(tree2, branch.length='none', layout='circular') +  geom_tiplab(aes(angle=angle))
p<-ggtree(tree2, branch.length='none') +  geom_tiplab()
p1<-p + new_scale_fill() +
         geom_fruit(data=dat2A, geom=geom_tile,
                    mapping=aes(y=label, x=Time, alpha=Abundance, fill=Time),
                    color = "grey50", offset = 0.3 ,size = 0.05)+
         scale_alpha_continuous(range=c(0, 1),
                             guide=guide_legend(keywidth = 0.3, keyheight = 0.3, order=5))+



         geom_fruit(data=dat2A, geom=geom_bar,
                    mapping=aes(y=label, x=Abundance, fill=Location),
                    pwidth=0.38, 
                    orientation="y", 
                    stat="identity",
         ) 



        scale_fill_ucscgb(palette = c("default"),
                           guide=guide_legend(keywidth = 0.3, keyheight = 0.3, order=4))+
         geom_treescale(fontsize=2, linesize=0.3, x=4.9, y=0.1) +
         theme(legend.position=c(0.93, 0.5),
               legend.background=element_rect(fill=NA),
               legend.title=element_text(size=6.5),
               legend.text=element_text(size=4.5),
               legend.spacing.y = unit(0.02, "cm"),
             )

p1 + layout_rectangular() + 
    theme(legend.position=c(.05, .7))
```

```{r}
p2<-ggtree(tree2, branch.length='none') +  geom_tiplab()
p3<-p2 + new_scale_fill() +
         geom_fruit(data=dat2A, geom=geom_bar,
                    mapping=aes(y=label, x=Abundance, fill=Subject),
                    pwidth=0.38, 
                    orientation="y", 
                    stat="identity",
                    offset = 0.13,
                     axis.params=list(
                         axis       = "x",
                         text.size  = 3,
                         hjust      = 0,
                         vjust      = 3,
                         nbreak     = 3,
                     ),
         ) 
        


scale_fill_igv(palette = c("default"),
                           guide=guide_legend(keywidth = 0.3, keyheight = 0.3, order=4))+
         geom_treescale(fontsize=2, linesize=0.3, x=4.9, y=0.1) +
         theme(legend.position=c(0.93, 0.5),
               legend.background=element_rect(fill=NA),
               legend.title=element_text(size=6.5),
               legend.text=element_text(size=4.5),
               legend.spacing.y = unit(0.02, "cm"),
             )

p3 + layout_rectangular() + 
    theme(legend.position=c(.05, .7))
```

```{r}
dat1A
set.seed(2019-10-31)
tr <- rtree(10)
d1 <- data.frame(
    # only some labels match
    label = c(tr$tip.label[sample(5, 5)], "A"),
    value = sample(1:6, 6))
d2 <- data.frame(
    label = rep(tr$tip.label, 5),
    category = rep(LETTERS[1:5], each=10),
    value = rnorm(50, 0, 3)) 
g <- ggtree(tr) + geom_tiplab(align=TRUE)
p1 <- ggplot(d1, aes(label, value)) + geom_col(aes(fill=label)) + 
    geom_text(aes(label=label, y= value+.1)) +
    coord_flip() + theme_tree2() + theme(legend.position='none')
p2 <- ggplot(d2, aes(x=category, y=label)) + 
    geom_tile(aes(fill=value)) + scale_fill_viridis_c() + 
    theme_tree2() 

p2 <- ggplot(dat1A, aes(x=Subject, y=label)) + 
    geom_tile(aes(fill=Abundance)) + scale_fill_viridis_c(option = "plasma") + 
    theme_tree2() 

g <- ggtree(tree2, branch.length='none') + geom_tiplab(align=TRUE)

library(cowplot)
library(aplot)

cowplot::plot_grid(g, p2, ncol=2, align = "v") 

p2 %>% insert_right(g)

dat1A

names(dat1A)[names(dat1A) == "ID"] <- "label"
dat1A<-as.data.frame(dat1A)
library(ggridges)
ggtree(tree2) +
  geom_facet(mapping = aes(x=Abundance,group=label, 
                           fill=Subject),
            data = dat1A, 
            geom = geom_density_ridges,
            panel="Abundance",  
            lwd=.3)


g <- ggtree(tree2, branch.length='none') + geom_tiplab()  
  
library(dplyr)
tip_hporder<-c(get_taxa_name(p))
dat1Asorted%>%
   mutate("label" =  factor("label", levels = tip_hporder)) %>%
  arrange("label")  
write.table(dat2A, "dat2A.txt", sep="\t", quote=F)
dat1Asorted <- dat1A[order(dat1A$label, methid = tip_hporder ),] 
```


```{r}
dat2ALH <- dat2A[grep("left", dat2A$Location), ]
dat2ALH_NST <- subset(dat2ALH, select = -c(Subject,Time,Location))
dat2ALH_NST<- ddply(dat2ALH_NST,"ID",numcolwise(sum))
names(dat2ALH_NST)[names(dat2ALH_NST) == "Abundance"] <- "LH_ABUND"

dat2ARH <- dat2A[grep("right", dat2A$Location), ]
dat2ARH_NST <- subset(dat2ARH, select = -c(Subject,Time,Location))
dat2ARH_NST<- ddply(dat2ARH_NST,"ID",numcolwise(sum))
names(dat2ARH_NST)[names(dat2ARH_NST) == "Abundance"] <- "RH_ABUND"

dat2ASC <- dat2A[grep("scalp", dat2A$Location), ]
dat2ASC_NST <- subset(dat2ASC, select = -c(Subject,Time,Location))
dat2ASC_NST<- ddply(dat2ASC_NST,"ID",numcolwise(sum))
names(dat2ASC_NST)[names(dat2ASC_NST) == "Abundance"] <- "SC_ABUND"

dat2A_Hands<- merge(dat2ALH_NST, dat2ARH_NST, by= "ID")
dat2A_ALL<- merge(dat2A_Hands, dat2ASC_NST, by= "ID")
write.table(dat2A_ALL, file="dat2A_ALL.txt", col.names=T, row.names=T, sep = "\t",quote=F)

```

```{r}
dat2A00 <- dat2A[grep("00_Weeks", dat2A$Time), ]
dat2A00_NST <- subset(dat2A00, select = -c(Subject,Time,Location))
dat2A00_NST<- ddply(dat2A00_NST,"ID",numcolwise(sum))
names(dat2A00_NST)[names(dat2A00_NST) == "Abundance"] <- "00_ABUND"

dat2A02 <- dat2A[grep("02_Weeks", dat2A$Time), ]
dat2A02_NST <- subset(dat2A02, select = -c(Subject,Time,Location))
dat2A02_NST<- ddply(dat2A02_NST,"ID",numcolwise(sum))
names(dat2A02_NST)[names(dat2A02_NST) == "Abundance"] <- "02_ABUND"

dat2A04 <- dat2A[grep("04_Weeks", dat2A$Time), ]
dat2A04_NST <- subset(dat2A04, select = -c(Subject,Time,Location))
dat2A04_NST<- ddply(dat2A04_NST,"ID",numcolwise(sum))
names(dat2A04_NST)[names(dat2A04_NST) == "Abundance"] <- "04_ABUND"

dat2A12 <- dat2A[grep("12_Weeks", dat2A$Time), ]
dat2A12_NST <- subset(dat2A12, select = -c(Subject,Time,Location))
dat2A12_NST<- ddply(dat2A12_NST,"ID",numcolwise(sum))
names(dat2A12_NST)[names(dat2A12_NST) == "Abundance"] <- "12_ABUND"

dat2A24 <- dat2A[grep("24_Weeks", dat2A$Time), ]
dat2A24_NST <- subset(dat2A24, select = -c(Subject,Time,Location))
dat2A24_NST<- ddply(dat2A24_NST,"ID",numcolwise(sum))
names(dat2A24_NST)[names(dat2A24_NST) == "Abundance"] <- "24_ABUND"


dat2A_TIME_ALL1<- merge(dat2A00_NST, dat2A02_NST, by= "ID")
dat2A_TIME_ALL2<- merge(dat2A_TIME_ALL1, dat2A04_NST, by= "ID")
dat2A_TIME_ALL3<- merge(dat2A_TIME_ALL2, dat2A12_NST, by= "ID")
dat2A_TIME_ALL<- merge(dat2A_TIME_ALL3, dat2A24_NST, by= "ID")
write.table(dat2A_TIME_ALL, file="dat2A_TIME_ALL.txt", col.names=T, row.names=T, sep = "\t",quote=F)

```

```{r}
dat1A<- t(HMAP_Subject2)
dat1A<- as.data.frame(dat1A)
write.table(dat2A, file="dat2A.txt", col.names=T, row.names=T, sep = "\t",quote=F)

dat2A->dat3A
dat3A <- dat3A[grep("P01", dat2A$Subject), ]
dat3A_NST <- subset(dat3A, select = -c(Subject))
dat3A_NST<- dat3A_NST%>%pivot_wider(names_from = Time, values_from = Abundance)
dat3A_NST <- subset(dat3A_NST, select = -c(Location))
dat3A_NST<- ddply(dat3A,"ID",numcolwise(sum))
```

```{r}
library(ggthemes)
VIBRANT_AMG_pathways_unmapped_contigs

VIBRANT_AMG_pathways_unmapped_contigs <- VIBRANT_AMG_pathways_unmapped_contigs[order(VIBRANT_AMG_pathways_unmapped_contigs$Metabolism),]
#Turn your 'treatment' column into a character vector
VIBRANT_AMG_pathways_unmapped_contigs$Pathway <- as.character(VIBRANT_AMG_pathways_unmapped_contigs$Pathway)
#Then turn it back into a factor with the levels in the correct order
VIBRANT_AMG_pathways_unmapped_contigs$Pathway <- factor(VIBRANT_AMG_pathways_unmapped_contigs$Pathway, levels=unique(VIBRANT_AMG_pathways_unmapped_contigs$Pathway))

k<-ggplot(data=VIBRANT_AMG_pathways_unmapped_contigs, aes(x=Pathway, y=Total_AMGs, fill=Metabolism)) +
  geom_bar(position = "dodge", stat="identity") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+labs(title="Bar Plot")



ggdotchart(VIBRANT_AMG_pathways_unmapped_contigs, x = "Pathway", y = "Total_AMGs",
           color = "Metabolism",                                # Color by groups
           palette = c("default"), # Custom color palette
           sorting = "descending",                       # Sort value in descending order
           add = "segments",                             # Add segments from y = 0 to dots
           rotate = TRUE,                                # Rotate vertically
           group = "Metabolism",                                # Order by groups
           dot.size = 6,                                 # Large dot size
           label = round(VIBRANT_AMG_pathways_unmapped_contigs$Total_AMGs),                        # Add mpg values as dot labels
           font.label = list(color = "white", size = 9, 
                             vjust = 0.5),               # Adjust label parameters
           ggtheme = theme_pubr()                        # ggplot2 theme
           )

k


theme_Publication <- function(base_size=14, base_family="helvetica") {
      library(grid)
      library(ggthemes)
      (theme_foundation(base_size=base_size, base_family=base_family)
       + theme(plot.title = element_text(face = "bold",
                                         size = rel(1.2), hjust = 0.5),
               text = element_text(),
               panel.background = element_rect(colour = NA),
               plot.background = element_rect(colour = NA),
               panel.border = element_rect(colour = NA),
               axis.title = element_text(face = "bold",size = rel(1)),
               axis.title.y = element_text(angle=90,vjust =2),
               axis.title.x = element_text(vjust = -0.2),
               axis.text = element_text(), 
               axis.line = element_line(colour="black"),
               axis.ticks = element_line(),
               panel.grid.major = element_line(colour="#f0f0f0"),
               panel.grid.minor = element_blank(),
               legend.key = element_rect(colour = NA),
               legend.position = "bottom",
               legend.direction = "horizontal",
               legend.key.size= unit(0.1, "cm"),
               legend.margin = unit(0, "cm"),
               legend.title = element_text(face="italic"),
               plot.margin=unit(c(10,5,5,5),"mm"),
               strip.background=element_rect(colour="#f0f0f0",fill="#f0f0f0"),
               strip.text = element_text(face="bold")
          ))
      
}

scale_fill_Publication <- function(...){
      library(scales)
      discrete_scale("fill","Publication",manual_pal(values = c("#386cb0","#fdb462","#7fc97f","#ef3b2c","#662506","#a6cee3","#fb9a99","#984ea3","#ffff33")), ...)

}

scale_colour_Publication <- function(...){
      library(scales)
      discrete_scale("colour","Publication",manual_pal(values = c("#386cb0","#fdb462","#7fc97f","#ef3b2c","#662506","#a6cee3","#fb9a99","#984ea3","#ffff33")), ...)

}
```


