---
title: "Phage"
output: html_document
---

```{r}
library(NRES803)
library(tidyverse)
library(GGally)
library(readr)
library(mgcv)
library(broom)
library(gridExtra)
library(lme4)
library(lmerTest)
library(broom.mixed)
library(readr)
library(phyloseq)
library(dplyr)
library(tidyr)
library(ggplot2)
library(phyloseq)
library(tidyverse)
library(microbiome)
library(readr)
library("ggpubr")
library("tidyverse")
library(vegan)
library(ComplexHeatmap)
library(gridBase)
library(metagenomeSeq)
library(phyloseq)
library(vegan)
```

```{r}
phage_counts2 <- read_delim("~/Desktop/AMR/phage_counts3.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r}
Anntations <- phage_counts2$contig
Anntations <- as.data.frame(Anntations)
Anntations$phylum = Anntations$V1
Anntations$phylum = Anntations$Anntations
Anntations$AMR_gene <- "NA"
Anntations$AMR_gene[which(Anntations$phylum == "k127_2557225")] <-"PARE"
Anntations$AMR_gene[which(Anntations$phylum == "k127_8634245")] <-"MSRD"
Anntations$AMR_gene[which(Anntations$phylum == "k127_12842510")] <-"ERMC"
Anntations$AMR_gene[which(Anntations$phylum == "k127_6535600")] <-"MLS23S"
Anntations$AMR_gene[which(Anntations$phylum == "k127_2646993")] <-"GYRA"
Anntations$AMR_gene[which(Anntations$phylum == "k127_15107270")] <-"QACF_ANTIPRIME3"
Anntations$AMR_gene[which(Anntations$phylum == "k127_2291")] <-"ERMC2"
write_csv(Anntations, 'Anntations.csv')
PTAX_pre <- read.csv("Anntations.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
PTAX_pre<- as.matrix(PTAX_pre)
PTAX <- tax_table(PTAX_pre)
```

```{r}
phage_counts <- subset(phage_counts2, select = -c(length))
write_csv(phage_counts, 'phage_counts.csv')
Phage_OTU_counts <- read.csv("phage_counts.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
POTU <- otu_table(Phage_OTU_counts, taxa_are_rows = TRUE)
```

```{r}
Metadata2 <- read_csv("~/Desktop/Current_work/OTU_Builber/Sequencing_Data_Info_Sheet3.csv")
#Metadata2= as.matrix(Metadata)
META <- sample_data(Metadata2)
sample_names(META)= META$SampleID
```


```{r}
PPHY <- phyloseq(POTU, PTAX, META)
```
```{r}
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01")
PPHY1 <- subset_samples(PPHY, !(SampleID %in% Samples_toRemove))
```

```{r}
PARE <- subset_taxa(PPHY1, phylum=="k127_2557225" )
plot_bar(PARE, x="Subject", title = "PARE Gene Abundance", facet_grid= "Time_Week")
```


```{r}
MSRD <- subset_taxa(PPHY1, phylum=="k127_8634245" )
plot_bar(MSRD, x="Subject", title = "MSRD Gene Abundance", facet_grid= "Time_Week")
```


```{r}
ERMC <- subset_taxa(PPHY1, phylum=="k127_12842510" )
plot_bar(ERMC, x="Subject", title = "ERMC Gene Abundance", facet_grid= "Time_Week")
```

```{r}
MLS23S <- subset_taxa(PPHY1, phylum=="k127_6535600" )
plot_bar(MLS23S, x="Subject", title = "MLS23S Gene Abundance", facet_grid= "Time_Week")
```

```{r}
GYRA <- subset_taxa(PPHY1, phylum=="k127_2646993" )
plot_bar(GYRA, x="Subject", title = "GYRA Gene Abundance", facet_grid= "Time_Week")
```

```{r}
QACF_ANTIPRIME3 <- subset_taxa(PPHY1, phylum=="k127_15107270" )
plot_bar(QACF_ANTIPRIME3, x="Subject", title = "ANTIPRIME3 and QACF Gene Abundance", facet_grid= "Time_Week")
```

```{r}
ERMC2 <- subset_taxa(PPHY1, phylum=="k127_2291" )
plot_bar(ERMC2, x="Subject", title = "ERMC2 Gene Abundance", facet_grid= "Time_Week")
```

```{r}
AMR_gene_stats <- sample_data(PPHY1)
PAREa <- otu_table(PARE)
PAREa <- t(PAREa)
PARE_gene_stats <- merge(META,PAREa, by= "row.names")
names(PARE_gene_stats)[names(PARE_gene_stats) == "k127_2557225"] <- "PARE"

MSRDa <- otu_table(MSRD)
MSRDa <- t(MSRDa)
MSRD_gene_stats <- merge(META,MSRDa, by= "row.names")
names(MSRD_gene_stats)[names(MSRD_gene_stats) == "k127_8634245"] <- "MSRD"

ERMCa <- otu_table(ERMC)
ERMCa <- t(ERMCa)
ERMC_gene_stats <- merge(META,ERMCa, by= "row.names")
names(ERMC_gene_stats)[names(ERMC_gene_stats) == "k127_12842510"] <- "ERMC"

MLS23Sa <- otu_table(MLS23S)
MLS23Sa <- t(MLS23Sa)
MLS23S_gene_stats <- merge(META,MLS23Sa, by= "row.names")
names(MLS23S_gene_stats)[names(MLS23S_gene_stats) == "k127_6535600"] <- "MLS23S"

GYRAa <- otu_table(GYRA)
GYRAa <- t(GYRAa)
GYRA_gene_stats <- merge(META,GYRAa, by= "row.names")
names(GYRA_gene_stats)[names(GYRA_gene_stats) == "k127_2646993"] <- "GYRA"

QACF_ANTIPRIME3a <- otu_table(QACF_ANTIPRIME3)
QACF_ANTIPRIME3a <- t(QACF_ANTIPRIME3a)
QACF_ANTIPRIME3_gene_stats <- merge(META,QACF_ANTIPRIME3a, by= "row.names")
names(QACF_ANTIPRIME3_gene_stats)[names(QACF_ANTIPRIME3_gene_stats) == "k127_15107270"] <- "QACF_ANTIPRIME3"

ERMC2a <- otu_table(ERMC2)
ERMC2a <- t(ERMC2a)
ERMC2_gene_stats <- merge(META,ERMC2a, by= "row.names")
names(ERMC2_gene_stats)[names(ERMC2_gene_stats) == "k127_2291"] <- "ERMC2"

```

```{r}
PARE_gene_stats_left <- PARE_gene_stats2[!grepl("scalp", PARE_gene_stats2$Location),]
PARE_gene_stats_left <- PARE_gene_stats_left[!grepl("right", PARE_gene_stats_left$Location),]
summary(ss.lm <- lm(PARE ~ Time_Week, PARE_gene_stats_left))
ssBase <- ggplot(PARE_gene_stats_left, aes(x = Time_Week, y = PARE)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Time Since Initial Sampling (week)", y = " Left Hand PARE Abundance")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
PARE_gene_stats2 <- as.data.frame(PARE_gene_stats)
summary(ss.lm <- lm(PARE ~ Time_Week, PARE_gene_stats2))
ssBase <- ggplot(PARE_gene_stats2, aes(x = Time_Week, y = PARE)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```


```{r}
MSRD_gene_stats2 <- as.data.frame(MSRD_gene_stats)
summary(ss.lm <- lm(MSRD ~ Time_Week, MSRD_gene_stats2))
sum
ssBase <- ggplot(MSRD_gene_stats2, aes(x = Time_Week, y = MSRD)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
ERMC_gene_stats2 <- as.data.frame(ERMC_gene_stats)
summary(ss.lm <- lm(ERMC ~ Time_Week, ERMC_gene_stats2))
ssBase <- ggplot(ERMC_gene_stats2, aes(x = Time_Week, y = ERMC)) + geom_point(aes(color = Location)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
ERMC_gene_stats_left <- ERMC_gene_stats2[!grepl("scalp", ERMC_gene_stats2$Location),]
ERMC_gene_stats_left <- ERMC_gene_stats_left[!grepl("right", ERMC_gene_stats_left$Location),]
summary(ss.lm <- lm(ERMC ~ Time_Week, ERMC_gene_stats_left))
ssBase_left <- ggplot(ERMC_gene_stats_left, aes(x = Time_Week, y = ERMC)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Time Since Initial Sampling (week)", y = " Left Hand ERMC Abundance")
ssBase_left + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)

ERMC_gene_stats_right <- ERMC_gene_stats2[!grepl("scalp", ERMC_gene_stats2$Location),]
ERMC_gene_stats_right <- ERMC_gene_stats_right[!grepl("left", ERMC_gene_stats_right$Location),]
summary(ss.lm <- lm(ERMC ~ Time_Week, ERMC_gene_stats_right))
ssBase_right <- ggplot(ERMC_gene_stats_right, aes(x = Time_Week, y = ERMC)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Time Since Initial Sampling (week)", y = "Right Hand ERMC Abundance")
ssBase_right + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)

ERMC_gene_stats_scalp <- ERMC_gene_stats2[!grepl("left", ERMC_gene_stats2$Location),]
ERMC_gene_stats_scalp <- ERMC_gene_stats_scalp[!grepl("right", ERMC_gene_stats_scalp$Location),]
summary(ss.lm <- lm(ERMC ~ Time_Week, ERMC_gene_stats_scalp))
ssBase_scalp <- ggplot(ERMC_gene_stats_scalp, aes(x = Time_Week, y = ERMC)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Time Since Initial Sampling (week)", y = "Scalp Hand ERMC Abundance")
ssBase_scalp + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)

```

```{r}
MLS23S_gene_stats2 <- as.data.frame(MLS23S_gene_stats)
summary(ss.lm <- lm(MLS23S ~ Time_Week, MLS23S_gene_stats2))
ssBase <- ggplot(MLS23S_gene_stats2, aes(x = Time_Week, y = MLS23S)) + geom_point(aes(color = Location)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
GYRA_gene_stats2 <- as.data.frame(GYRA_gene_stats)
summary(ss.lm <- lm(GYRA ~ Time_Week, GYRA_gene_stats2))
ssBase <- ggplot(GYRA_gene_stats2, aes(x = Time_Week, y = GYRA)) + geom_point(aes(color = Location)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
QACF_ANTIPRIME3_gene_stats2 <- as.data.frame(QACF_ANTIPRIME3_gene_stats)
summary(ss.lm <- lm(QACF_ANTIPRIME3 ~ Time_Week, QACF_ANTIPRIME3_gene_stats2))
ssBase <- ggplot(QACF_ANTIPRIME3_gene_stats2, aes(x = Time_Week, y = QACF_ANTIPRIME3)) + geom_point(aes(color = Location)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
ERMC2_gene_stats2 <- as.data.frame(ERMC2_gene_stats)
summary(ss.lm <- lm(ERMC2 ~ Time_Week, ERMC2_gene_stats2))
ssBase <- ggplot(ERMC2_gene_stats2, aes(x = Time_Week, y = ERMC2)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

#alpha diversity
```{r}
#make prevalence and total abundance table for each contig
prevdf_ps= apply(X = otu_table(PPHY1), 
                       MARGIN = ifelse(taxa_are_rows(PPHY1), yes = 1, no = 2), 
                       FUN = function(x){sum(x > 0)})
prevdf_ps <- data.frame(Prevalence= prevdf_ps, TotalAbundance=taxa_sums(PPHY1))
```

```{r}
#get rid of contigs with prevalence less than 1
ps_IBK_run <- rownames(prevdf_ps)[prevdf_ps$Prevalence > 1]
ps_IBK_run
#remove the taxa with prevalence less than 1 from phyloseq object and make new phyloseq object named ps_IBK_analyze
ps_IBK_analyze <- prune_taxa(ps_IBK_run, PPHY1)
ps_IBK_analyze
#seeing how many reads you retained 
sum(otu_table(ps_IBK_analyze))/sum(otu_table(PPHY1)) 
save(ps_IBK_analyze, file = "ps_IBK_analyze.rds")
load("ps_IBK_analyze.rds")
sum(otu_table(ps_IBK_analyze)) 
```

```{r}
#normalize data for beta diversity
norm_IBK  <-  transform_sample_counts(ps_IBK_analyze, function(x) x / sum(x) )
save(norm_IBK, file= "norm_IBK.rds")
load("norm_IBK.rds")
```

```{r}
#rarefy data to a even depth for alpha diversity only
ps_rarefy <- rarefy_even_depth(ps_IBK_analyze, sample.size = min(sample_sums(ps_IBK_analyze)),
  rngseed = T, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
#get alpha diversity measures of Choa1, observed, and shannon
Chao1_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Chao1"))
head(Chao1_obser_rarefy)
obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Observed"))
head(obser_rarefy)
shannon_obser_rarefy= estimate_richness(ps_rarefy, split = TRUE, measures = c("Shannon"))
head(shannon_obser_rarefy)
```
```{r}
#to write observed measurements for excel
#write.table(obser_rarefy, "a.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
#merge alpha diversity measures with metadata
Chao1_obser_rarefy_metadata<- merge(META,Chao1_obser_rarefy, by= "row.names")
head(Chao1_obser_rarefy_metadata)
shannon_obser_rarefy_metadata<- merge(META,shannon_obser_rarefy, by= "row.names")
head(shannon_obser_rarefy_metadata)
```


```{r}
#run alpha diversity measures for Location
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Location", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Location, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Location)
observed_pairwise
#it is noted that using observed there is no significant difference between scalp, left, or right 
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Location)
chao1_pairwise 
#it is noted that using choa1 there is no significant differnce between left and right hand but there is a significant difference from left and scalp and right and scalp
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Location)
shannon_pairwise
#it is noted that using shannon there is no significant differnce between left and right hand but there is a significant difference from left and scalp and right and scalp
```

```{r}
#run alpha diversity measures for Time_Collected
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Time_Collected", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Time_Collected, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Time_Collected)
observed_pairwise
#Significant differences:  2_week&1_month, 2_week&3_month, 2_week&6_month,
#No significant differences: T_0&2_week, T_0&1_month, T_0&3_month, T_0&6_month, 1_month&3_month, 1_month&6_month, 3_month&6_month
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Time_Collected)
chao1_pairwise 
#Significant differences: T_0&3_month, 2_week&1_month, 2_week&3_month, 2_week&6_month, 
#No significant differences: T_0&2_week, T_0&1_month, T_0&6_month, 1_month&3_month, 1_month&6_month, 3_month&6_month 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Time_Collected)
shannon_pairwise
#Significant differences: None. 
#No significant differences 
```

```{r}
#run alpha diversity measures for Month_Sampled
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Month_Sampled", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Month_Sampled, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Month_Sampled)
observed_pairwise
#Significant differences: january&may, july&may, june&may, march&may
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Month_Sampled)
chao1_pairwise 
#Significant differences: may&april, may&august, march&august, may&january, march&july, may&july, june&march, june&may, march&may, march&november, may&october, may&september, 
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Month_Sampled)
shannon_pairwise
#Significant differences: None. may not have worked though since all were 1
```

```{r}
#run alpha diversity measures for Gender
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Gender", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Gender, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Gender)
observed_pairwise
#No significant differences
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Gender)
chao1_pairwise 
#Significant differences
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Gender)
shannon_pairwise
#No significant differences 
```

```{r}
#run alpha diversity measures for Antibacterial_Soap_Used
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Antibacterial_Soap_Used", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Antibacterial_Soap_Used, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Antibacterial_Soap_Used)
observed_pairwise
#Significant differences:  dish soap and all others
#No significant differences: no difference wth any of the others
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Antibacterial_Soap_Used)
chao1_pairwise 
#Significant differences:  dish soap and all others
#No significant differences: no difference wth any of the others
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Antibacterial_Soap_Used)
shannon_pairwise
#Significant differences: None. 
#No significant differences 
```

```{r}
#run alpha diversity measures for Subject
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Subject", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Subject, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Subject)
observed_pairwise
#Significant differences:  dish soap and all others
#No significant differences: no difference wth any of the others
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Subject)
chao1_pairwise 
#Significant differences:  dish soap and all others
#No significant differences: no difference wth any of the others
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Subject)
shannon_pairwise
#Significant differences: None. 
#No significant differences 
```

```{r}
#run alpha diversity measures for Gel_Sanitizer_Use
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Gel_Sanitizer_Use", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Gel_Sanitizer_Use, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Gel_Sanitizer_Use)
observed_pairwise
#Significant differences:  rarely&N, rarely&Y
#No significant differences: no difference with Y&N&Sometimes, Rarely&Sometimes
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Gel_Sanitizer_Use)
chao1_pairwise 
#Significant differences:  None
#No significant differences: All
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Gel_Sanitizer_Use)
shannon_pairwise
#Significant differences: None. 
#No significant differences : All.
```

```{r}
#run alpha diversity measures for Hand_Lotion_Use
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Hand_Lotion_Use", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Hand_Lotion_Use, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Hand_Lotion_Use)
observed_pairwise
#Significant differences:  None
#No significant differences: All
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Hand_Lotion_Use)
chao1_pairwise 
#Significant differences:  None
#No significant differences: All
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Hand_Lotion_Use)
shannon_pairwise
#Significant differences: None. 
#No significant differences : All.
```

```{r}
#run alpha diversity measures for Swam_in_Chlorine_Pool
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Swam_in_Chlorine_Pool", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Swam_in_Chlorine_Pool, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Swam_in_Chlorine_Pool)
observed_pairwise
#Significant differences:  None
#No significant differences: All
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Swam_in_Chlorine_Pool)
chao1_pairwise 
#Significant differences:  None
#No significant differences: All
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Swam_in_Chlorine_Pool)
shannon_pairwise
#Significant differences: None. 
#No significant differences : All.
```

```{r}
#run alpha diversity measures for Tanning_Bed_Use
alpha_meas = c("Observed", "Chao1", "Shannon")
(p <- plot_richness(ps_rarefy, "Tanning_Bed_Use", measures=alpha_meas))
test <- (p + geom_boxplot(data=p$data, aes(x=Tanning_Bed_Use, y=value, color=NULL), alpha=0.1)) 
plot(test)
observed_pairwise <- pairwise.wilcox.test(obser_rarefy$Observed, sample_data(ps_rarefy)$Tanning_Bed_Use)
observed_pairwise
#Significant differences:  None
#No significant differences: All
chao1_pairwise <- pairwise.wilcox.test(Chao1_obser_rarefy$Chao1, sample_data(ps_rarefy)$Tanning_Bed_Use)
chao1_pairwise 
#Significant differences:  None
#No significant differences: All
shannon_pairwise <- pairwise.wilcox.test(shannon_obser_rarefy$Shannon, sample_data(ps_rarefy)$Tanning_Bed_Use)
shannon_pairwise
#Significant differences: None. 
#No significant differences : All.
```

```{r}
norm_PPHY1<- transform_sample_counts(PPHY1, function(x) x / sum(x) )
ord.PPHY1.pcoa <- ordinate(norm_PPHY1, method="PCoA", distance="bray")
plot_ordination(norm_PPHY1, ord.PPHY1.pcoa, color = "Location", axes = c(2,1)) +
  geom_point(size=2) 

```

```{r}
norm_PPHY1<- transform_sample_counts(PPHY1, function(x) x / sum(x) )

metadata_1 <- as(sample_data(norm_PPHY1), "data.frame")

adonis(phyloseq::distance(norm_PPHY1, method="bray") ~  Time_Collected + Gender + Subject,
       data = metadata_1)
```

```{r}
norm_LH<- subset_samples(PPHY1, Location =="left")
metadata_LH <- as(sample_data(norm_LH), "data.frame")

adonis(phyloseq::distance(norm_LH, method="bray") ~  Time_Collected + Gender + Subject,
       data = metadata_LH)
```

```{r}
norm_RH<- subset_samples(PPHY1, Location =="right")
metadata_RH <- as(sample_data(norm_RH), "data.frame")

adonis(phyloseq::distance(norm_RH, method="bray") ~  Time_Collected + Gender + Subject,
       data = metadata_RH)
```

```{r}
norm_SC<- subset_samples(PPHY1, Location =="scalp")
metadata_SC <- as(sample_data(norm_SC), "data.frame")

adonis(phyloseq::distance(norm_SC, method="bray") ~  Time_Collected + Gender + Subject,
       data = metadata_SC)
```

```{r}
norm_AMR  <-  transform_sample_counts(PPHY1, function(x) x / sum(x) )
AMR_list <- c("k127_2557225", "k127_2557225","k127_8634245","k127_12842510","k127_6535600","k127_2646993","k127_15107270","k127_2291")
AMR <- subset_taxa(norm_AMR, (phylum %in% AMR_list))

AMR_heatmap <- AMR %>%
  subset_samples(
    !is.na(Subject)
  )

(p <- plot_heatmap(AMR_heatmap,"Subject"))
```

```{r}
  plot_net(PPHY1, distance = "bray", type = "taxa", 
           maxdist = 0.5, color="AMR_gene")
```

```{r}
ij4 <- make_network(PPHY1, type="samples", distance="bray", max.dist = 0.4, keep.isolates=FALSE)
plot_network(ij4, PPHY1, line_weight=0.6)
```

```{r}
library(phyloseq)
library(igraph)
library(ggnetwork)
ij4 <- make_network(PPHY1, type="taxa", distance="bray", max.dist = 0.3, keep.isolates=FALSE)
plot_network(ij4, PPHY1, line_weight=0.6)
```

```{r}
ij5 <- make_network(PPHY1, type="taxa", distance="bray", max.dist = 0.5, keep.isolates=FALSE)
plot_network(ij4, PPHY1, line_weight=0.6)
```



#CHORD PLOT
```{r}
library("circlize")
PARE_otu<- otu_table
View(PARE_otu_m)
PARE_otu_m <- as.matrix(PARE_otu)
chordDiagram(PARE_otu_m)
```

```{r}
r_name <- c("PARE", "MSRD", "ERMC", "MLS23S", "GYRA", "QACF_ANTIPRIME3")
chord_p <- data.frame(row.names = r_name )
chord_p$P01 <- 0
chord_p$P02 <- 0
chord_p$P03 <- 0
chord_p$P04 <- 0
chord_p$P05 <- 0
chord_p$P06 <- 0
chord_p$P07 <- 0
chord_p$P08 <- 0
chord_p$P09 <- 0
chord_p$P10 <- 0
chord_p$P11 <- 0
chord_p$P12 <- 0
chord_p$P13 <- 0
chord_p$P14 <- 0
chord_p$P15 <- 0
chord_p$P16 <- 0
chord_p$P17 <- 0
chord_p$P18 <- 0
chord_p$P19 <- 0
chord_p$P20 <- 0
chord_p$P21 <- 0
chord_p$P22 <- 0
chord_p$P23 <- 0
chord_p$P24 <- 0
chord_p$P25 <- 0
chord_p$P26 <- 0
chord_p$P27 <- 0
chord_p$P28 <- 0
chord_p$P29 <- 0
chord_p$P30 <- 0
chord_p$P31 <- 0
chord_p$P32 <- 0
chord_p$P34 <- 0
chord_p$P35 <- 0
chord_p$P36 <- 0
chord_p$P37 <- 0
chord_p$P38 <- 0
chord_p$P39 <- 0
chord_p$P40 <- 0
chord_p$P41 <- 0
chord_p$P42 <- 0
chord_p$P43 <- 0
```

```{r}
P01_PARE<- subset_samples(PARE, Subject=="P01")
otu_table(P01_PARE)
chord_p["PARE","P01"] = taxa_sums(P01_PARE)
remove(P01_PARE)
P01_MSRD<- subset_samples(MSRD, Subject=="P01")
otu_table(P01_MSRD)
chord_p["MSRD","P01"] = taxa_sums(P01_MSRD)
remove(P01_MSRD)
P01_MLS23S<- subset_samples(MLS23S, Subject=="P01")
otu_table(P01_MLS23S)
chord_p["MLS23S","P01"] = taxa_sums(P01_MLS23S)
remove(P01_MLS23S)
P01_GYRA<- subset_samples(GYRA, Subject=="P01")
otu_table(P01_GYRA)
chord_p["GYRA","P01"] = taxa_sums(P01_GYRA)
remove(P01_GYRA)
P01_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P01")
otu_table(P01_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P01"] = taxa_sums(P01_QACF_ANTIPRIME3)
remove(P01_QACF_ANTIPRIME3)
P01_ERMC<- subset_samples(ERMC, Subject=="P01")
otu_table(P01_ERMC)
chord_p["ERMC","P01"] = taxa_sums(P01_ERMC)
remove(P01_ERMC)



P02_PARE<- subset_samples(PARE, Subject=="P02")
otu_table(P02_PARE)
chord_p["PARE","P02"] = taxa_sums(P02_PARE)
remove(P02_PARE)
P02_MSRD<- subset_samples(MSRD, Subject=="P02")
otu_table(P02_MSRD)
chord_p["MSRD","P02"] = taxa_sums(P02_MSRD)
remove(P02_MSRD)
P02_MLS23S<- subset_samples(MLS23S, Subject=="P02")
otu_table(P02_MLS23S)
chord_p["MLS23S","P02"] = taxa_sums(P02_MLS23S)
remove(P02_MLS23S)
P02_GYRA<- subset_samples(GYRA, Subject=="P02")
otu_table(P02_GYRA)
chord_p["GYRA","P02"] = taxa_sums(P02_GYRA)
remove(P02_GYRA)
P02_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P02")
otu_table(P02_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P02"] = taxa_sums(P02_QACF_ANTIPRIME3)
remove(P02_QACF_ANTIPRIME3)
P02_ERMC<- subset_samples(ERMC, Subject=="P02")
otu_table(P02_ERMC)
chord_p["ERMC","P02"] = taxa_sums(P02_ERMC)
remove(P02_ERMC)



P03_PARE<- subset_samples(PARE, Subject=="P03")
otu_table(P03_PARE)
chord_p["PARE","P03"] = taxa_sums(P03_PARE)
remove(P03_PARE)
P03_MSRD<- subset_samples(MSRD, Subject=="P03")
otu_table(P03_MSRD)
chord_p["MSRD","P03"] = taxa_sums(P03_MSRD)
remove(P03_MSRD)
P03_MLS23S<- subset_samples(MLS23S, Subject=="P03")
otu_table(P03_MLS23S)
chord_p["MLS23S","P03"] = taxa_sums(P03_MLS23S)
remove(P03_MLS23S)
P03_GYRA<- subset_samples(GYRA, Subject=="P03")
otu_table(P03_GYRA)
chord_p["GYRA","P03"] = taxa_sums(P03_GYRA)
remove(P03_GYRA)
P03_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P03")
otu_table(P03_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P03"] = taxa_sums(P03_QACF_ANTIPRIME3)
remove(P03_QACF_ANTIPRIME3)
P03_ERMC<- subset_samples(ERMC, Subject=="P03")
otu_table(P03_ERMC)
chord_p["ERMC","P03"] = taxa_sums(P03_ERMC)
remove(P03_ERMC)



P04_PARE<- subset_samples(PARE, Subject=="P04")
otu_table(P04_PARE)
chord_p["PARE","P04"] = taxa_sums(P04_PARE)
remove(P04_PARE)
P04_MSRD<- subset_samples(MSRD, Subject=="P04")
otu_table(P04_MSRD)
chord_p["MSRD","P04"] = taxa_sums(P04_MSRD)
remove(P04_MSRD)
P04_MLS23S<- subset_samples(MLS23S, Subject=="P04")
otu_table(P04_MLS23S)
chord_p["MLS23S","P04"] = taxa_sums(P04_MLS23S)
remove(P04_MLS23S)
P04_GYRA<- subset_samples(GYRA, Subject=="P04")
otu_table(P04_GYRA)
chord_p["GYRA","P04"] = taxa_sums(P04_GYRA)
remove(P04_GYRA)
P04_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P04")
otu_table(P04_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P04"] = taxa_sums(P04_QACF_ANTIPRIME3)
remove(P04_QACF_ANTIPRIME3)
P04_ERMC<- subset_samples(ERMC, Subject=="P04")
otu_table(P04_ERMC)
chord_p["ERMC","P04"] = taxa_sums(P04_ERMC)
remove(P04_ERMC)



P05_PARE<- subset_samples(PARE, Subject=="P05")
otu_table(P05_PARE)
chord_p["PARE","P05"] = taxa_sums(P05_PARE)
remove(P05_PARE)
P05_MSRD<- subset_samples(MSRD, Subject=="P05")
otu_table(P05_MSRD)
chord_p["MSRD","P05"] = taxa_sums(P05_MSRD)
remove(P05_MSRD)
P05_MLS23S<- subset_samples(MLS23S, Subject=="P05")
otu_table(P05_MLS23S)
chord_p["MLS23S","P05"] = taxa_sums(P05_MLS23S)
remove(P05_MLS23S)
P05_GYRA<- subset_samples(GYRA, Subject=="P05")
otu_table(P05_GYRA)
chord_p["GYRA","P05"] = taxa_sums(P05_GYRA)
remove(P05_GYRA)
P05_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P05")
otu_table(P05_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P05"] = taxa_sums(P05_QACF_ANTIPRIME3)
remove(P05_QACF_ANTIPRIME3)
P05_ERMC<- subset_samples(ERMC, Subject=="P05")
otu_table(P05_ERMC)
chord_p["ERMC","P05"] = taxa_sums(P05_ERMC)
remove(P05_ERMC)



P06_PARE<- subset_samples(PARE, Subject=="P06")
otu_table(P06_PARE)
chord_p["PARE","P06"] = taxa_sums(P06_PARE)
remove(P06_PARE)
P06_MSRD<- subset_samples(MSRD, Subject=="P06")
otu_table(P06_MSRD)
chord_p["MSRD","P06"] = taxa_sums(P06_MSRD)
remove(P06_MSRD)
P06_MLS23S<- subset_samples(MLS23S, Subject=="P06")
otu_table(P06_MLS23S)
chord_p["MLS23S","P06"] = taxa_sums(P06_MLS23S)
remove(P06_MLS23S)
P06_GYRA<- subset_samples(GYRA, Subject=="P06")
otu_table(P06_GYRA)
chord_p["GYRA","P06"] = taxa_sums(P06_GYRA)
remove(P06_GYRA)
P06_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P06")
otu_table(P06_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P06"] = taxa_sums(P06_QACF_ANTIPRIME3)
remove(P06_QACF_ANTIPRIME3)
P06_ERMC<- subset_samples(ERMC, Subject=="P06")
otu_table(P06_ERMC)
chord_p["ERMC","P06"] = taxa_sums(P06_ERMC)
remove(P06_ERMC)



P07_PARE<- subset_samples(PARE, Subject=="P07")
otu_table(P07_PARE)
chord_p["PARE","P07"] = taxa_sums(P07_PARE)
remove(P07_PARE)
P07_MSRD<- subset_samples(MSRD, Subject=="P07")
otu_table(P07_MSRD)
chord_p["MSRD","P07"] = taxa_sums(P07_MSRD)
remove(P07_MSRD)
P07_MLS23S<- subset_samples(MLS23S, Subject=="P07")
otu_table(P07_MLS23S)
chord_p["MLS23S","P07"] = taxa_sums(P07_MLS23S)
remove(P07_MLS23S)
P07_GYRA<- subset_samples(GYRA, Subject=="P07")
otu_table(P07_GYRA)
chord_p["GYRA","P07"] = taxa_sums(P07_GYRA)
remove(P07_GYRA)
P07_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P07")
otu_table(P07_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P07"] = taxa_sums(P07_QACF_ANTIPRIME3)
remove(P07_QACF_ANTIPRIME3)
P07_ERMC<- subset_samples(ERMC, Subject=="P07")
otu_table(P07_ERMC)
chord_p["ERMC","P07"] = taxa_sums(P07_ERMC)
remove(P07_ERMC)



P08_PARE<- subset_samples(PARE, Subject=="P08")
otu_table(P08_PARE)
chord_p["PARE","P08"] = taxa_sums(P08_PARE)
remove(P08_PARE)
P08_MSRD<- subset_samples(MSRD, Subject=="P08")
otu_table(P08_MSRD)
chord_p["MSRD","P08"] = taxa_sums(P08_MSRD)
remove(P08_MSRD)
P08_MLS23S<- subset_samples(MLS23S, Subject=="P08")
otu_table(P08_MLS23S)
chord_p["MLS23S","P08"] = taxa_sums(P08_MLS23S)
remove(P08_MLS23S)
P08_GYRA<- subset_samples(GYRA, Subject=="P08")
otu_table(P08_GYRA)
chord_p["GYRA","P08"] = taxa_sums(P08_GYRA)
remove(P08_GYRA)
P08_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P08")
otu_table(P08_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P08"] = taxa_sums(P08_QACF_ANTIPRIME3)
remove(P08_QACF_ANTIPRIME3)
P08_ERMC<- subset_samples(ERMC, Subject=="P08")
otu_table(P08_ERMC)
chord_p["ERMC","P08"] = taxa_sums(P08_ERMC)
remove(P08_ERMC)



P09_PARE<- subset_samples(PARE, Subject=="P09")
otu_table(P09_PARE)
chord_p["PARE","P09"] = taxa_sums(P09_PARE)
remove(P09_PARE)
P09_MSRD<- subset_samples(MSRD, Subject=="P09")
otu_table(P09_MSRD)
chord_p["MSRD","P09"] = taxa_sums(P09_MSRD)
remove(P09_MSRD)
P09_MLS23S<- subset_samples(MLS23S, Subject=="P09")
otu_table(P09_MLS23S)
chord_p["MLS23S","P09"] = taxa_sums(P09_MLS23S)
remove(P09_MLS23S)
P09_GYRA<- subset_samples(GYRA, Subject=="P09")
otu_table(P09_GYRA)
chord_p["GYRA","P09"] = taxa_sums(P09_GYRA)
remove(P09_GYRA)
P09_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P09")
otu_table(P09_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P09"] = taxa_sums(P09_QACF_ANTIPRIME3)
remove(P09_QACF_ANTIPRIME3)
P09_ERMC<- subset_samples(ERMC, Subject=="P09")
otu_table(P09_ERMC)
chord_p["ERMC","P09"] = taxa_sums(P09_ERMC)
remove(P09_ERMC)



P10_PARE<- subset_samples(PARE, Subject=="P10")
otu_table(P10_PARE)
chord_p["PARE","P10"] = taxa_sums(P10_PARE)
remove(P10_PARE)
P10_MSRD<- subset_samples(MSRD, Subject=="P10")
otu_table(P10_MSRD)
chord_p["MSRD","P10"] = taxa_sums(P10_MSRD)
remove(P10_MSRD)
P10_MLS23S<- subset_samples(MLS23S, Subject=="P10")
otu_table(P10_MLS23S)
chord_p["MLS23S","P10"] = taxa_sums(P10_MLS23S)
remove(P10_MLS23S)
P10_GYRA<- subset_samples(GYRA, Subject=="P10")
otu_table(P10_GYRA)
chord_p["GYRA","P10"] = taxa_sums(P10_GYRA)
remove(P10_GYRA)
P10_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P10")
otu_table(P10_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P10"] = taxa_sums(P10_QACF_ANTIPRIME3)
remove(P10_QACF_ANTIPRIME3)
P10_ERMC<- subset_samples(ERMC, Subject=="P10")
otu_table(P10_ERMC)
chord_p["ERMC","P10"] = taxa_sums(P10_ERMC)
remove(P10_ERMC)



P11_PARE<- subset_samples(PARE, Subject=="P11")
otu_table(P11_PARE)
chord_p["PARE","P11"] = taxa_sums(P11_PARE)
remove(P11_PARE)
P11_MSRD<- subset_samples(MSRD, Subject=="P11")
otu_table(P11_MSRD)
chord_p["MSRD","P11"] = taxa_sums(P11_MSRD)
remove(P11_MSRD)
P11_MLS23S<- subset_samples(MLS23S, Subject=="P11")
otu_table(P11_MLS23S)
chord_p["MLS23S","P11"] = taxa_sums(P11_MLS23S)
remove(P11_MLS23S)
P11_GYRA<- subset_samples(GYRA, Subject=="P11")
otu_table(P11_GYRA)
chord_p["GYRA","P11"] = taxa_sums(P11_GYRA)
remove(P11_GYRA)
P11_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P11")
otu_table(P11_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P11"] = taxa_sums(P11_QACF_ANTIPRIME3)
remove(P11_QACF_ANTIPRIME3)
P11_ERMC<- subset_samples(ERMC, Subject=="P11")
otu_table(P11_ERMC)
chord_p["ERMC","P11"] = taxa_sums(P11_ERMC)
remove(P11_ERMC)



P12_PARE<- subset_samples(PARE, Subject=="P12")
otu_table(P12_PARE)
chord_p["PARE","P12"] = taxa_sums(P12_PARE)
remove(P12_PARE)
P12_MSRD<- subset_samples(MSRD, Subject=="P12")
otu_table(P12_MSRD)
chord_p["MSRD","P12"] = taxa_sums(P12_MSRD)
remove(P12_MSRD)
P12_MLS23S<- subset_samples(MLS23S, Subject=="P12")
otu_table(P12_MLS23S)
chord_p["MLS23S","P12"] = taxa_sums(P12_MLS23S)
remove(P12_MLS23S)
P12_GYRA<- subset_samples(GYRA, Subject=="P12")
otu_table(P12_GYRA)
chord_p["GYRA","P12"] = taxa_sums(P12_GYRA)
remove(P12_GYRA)
P12_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P12")
otu_table(P12_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P12"] = taxa_sums(P12_QACF_ANTIPRIME3)
remove(P12_QACF_ANTIPRIME3)
P12_ERMC<- subset_samples(ERMC, Subject=="P12")
otu_table(P12_ERMC)
chord_p["ERMC","P12"] = taxa_sums(P12_ERMC)
remove(P12_ERMC)



P13_PARE<- subset_samples(PARE, Subject=="P13")
otu_table(P13_PARE)
chord_p["PARE","P13"] = taxa_sums(P13_PARE)
remove(P13_PARE)
P13_MSRD<- subset_samples(MSRD, Subject=="P13")
otu_table(P13_MSRD)
chord_p["MSRD","P13"] = taxa_sums(P13_MSRD)
remove(P13_MSRD)
P13_MLS23S<- subset_samples(MLS23S, Subject=="P13")
otu_table(P13_MLS23S)
chord_p["MLS23S","P13"] = taxa_sums(P13_MLS23S)
remove(P13_MLS23S)
P13_GYRA<- subset_samples(GYRA, Subject=="P13")
otu_table(P13_GYRA)
chord_p["GYRA","P13"] = taxa_sums(P13_GYRA)
remove(P13_GYRA)
P13_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P13")
otu_table(P13_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P13"] = taxa_sums(P13_QACF_ANTIPRIME3)
remove(P13_QACF_ANTIPRIME3)
P13_ERMC<- subset_samples(ERMC, Subject=="P13")
otu_table(P13_ERMC)
chord_p["ERMC","P13"] = taxa_sums(P13_ERMC)
remove(P13_ERMC)



P14_PARE<- subset_samples(PARE, Subject=="P14")
otu_table(P14_PARE)
chord_p["PARE","P14"] = taxa_sums(P14_PARE)
remove(P14_PARE)
P14_MSRD<- subset_samples(MSRD, Subject=="P14")
otu_table(P14_MSRD)
chord_p["MSRD","P14"] = taxa_sums(P14_MSRD)
remove(P14_MSRD)
P14_MLS23S<- subset_samples(MLS23S, Subject=="P14")
otu_table(P14_MLS23S)
chord_p["MLS23S","P14"] = taxa_sums(P14_MLS23S)
remove(P14_MLS23S)
P14_GYRA<- subset_samples(GYRA, Subject=="P14")
otu_table(P14_GYRA)
chord_p["GYRA","P14"] = taxa_sums(P14_GYRA)
remove(P14_GYRA)
P14_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P14")
otu_table(P14_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P14"] = taxa_sums(P14_QACF_ANTIPRIME3)
remove(P14_QACF_ANTIPRIME3)
P14_ERMC<- subset_samples(ERMC, Subject=="P14")
otu_table(P14_ERMC)
chord_p["ERMC","P14"] = taxa_sums(P14_ERMC)
remove(P14_ERMC)



P15_PARE<- subset_samples(PARE, Subject=="P15")
otu_table(P15_PARE)
chord_p["PARE","P15"] = taxa_sums(P15_PARE)
remove(P15_PARE)
P15_MSRD<- subset_samples(MSRD, Subject=="P15")
otu_table(P15_MSRD)
chord_p["MSRD","P15"] = taxa_sums(P15_MSRD)
remove(P15_MSRD)
P15_MLS23S<- subset_samples(MLS23S, Subject=="P15")
otu_table(P15_MLS23S)
chord_p["MLS23S","P15"] = taxa_sums(P15_MLS23S)
remove(P15_MLS23S)
P15_GYRA<- subset_samples(GYRA, Subject=="P15")
otu_table(P15_GYRA)
chord_p["GYRA","P15"] = taxa_sums(P15_GYRA)
remove(P15_GYRA)
P15_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P15")
otu_table(P15_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P15"] = taxa_sums(P15_QACF_ANTIPRIME3)
remove(P15_QACF_ANTIPRIME3)
P15_ERMC<- subset_samples(ERMC, Subject=="P15")
otu_table(P15_ERMC)
chord_p["ERMC","P15"] = taxa_sums(P15_ERMC)
remove(P15_ERMC)



P16_PARE<- subset_samples(PARE, Subject=="P16")
otu_table(P16_PARE)
chord_p["PARE","P16"] = taxa_sums(P16_PARE)
remove(P16_PARE)
P16_MSRD<- subset_samples(MSRD, Subject=="P16")
otu_table(P16_MSRD)
chord_p["MSRD","P16"] = taxa_sums(P16_MSRD)
remove(P16_MSRD)
P16_MLS23S<- subset_samples(MLS23S, Subject=="P16")
otu_table(P16_MLS23S)
chord_p["MLS23S","P16"] = taxa_sums(P16_MLS23S)
remove(P16_MLS23S)
P16_GYRA<- subset_samples(GYRA, Subject=="P16")
otu_table(P16_GYRA)
chord_p["GYRA","P16"] = taxa_sums(P16_GYRA)
remove(P16_GYRA)
P16_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P16")
otu_table(P16_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P16"] = taxa_sums(P16_QACF_ANTIPRIME3)
remove(P16_QACF_ANTIPRIME3)
P16_ERMC<- subset_samples(ERMC, Subject=="P16")
otu_table(P16_ERMC)
chord_p["ERMC","P16"] = taxa_sums(P16_ERMC)
remove(P16_ERMC)



P17_PARE<- subset_samples(PARE, Subject=="P17")
otu_table(P17_PARE)
chord_p["PARE","P17"] = taxa_sums(P17_PARE)
remove(P17_PARE)
P17_MSRD<- subset_samples(MSRD, Subject=="P17")
otu_table(P17_MSRD)
chord_p["MSRD","P17"] = taxa_sums(P17_MSRD)
remove(P17_MSRD)
P17_MLS23S<- subset_samples(MLS23S, Subject=="P17")
otu_table(P17_MLS23S)
chord_p["MLS23S","P17"] = taxa_sums(P17_MLS23S)
remove(P17_MLS23S)
P17_GYRA<- subset_samples(GYRA, Subject=="P17")
otu_table(P17_GYRA)
chord_p["GYRA","P17"] = taxa_sums(P17_GYRA)
remove(P17_GYRA)
P17_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P17")
otu_table(P17_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P17"] = taxa_sums(P17_QACF_ANTIPRIME3)
remove(P17_QACF_ANTIPRIME3)
P17_ERMC<- subset_samples(ERMC, Subject=="P17")
otu_table(P17_ERMC)
chord_p["ERMC","P17"] = taxa_sums(P17_ERMC)
remove(P17_ERMC)



P18_PARE<- subset_samples(PARE, Subject=="P18")
otu_table(P18_PARE)
chord_p["PARE","P18"] = taxa_sums(P18_PARE)
remove(P18_PARE)
P18_MSRD<- subset_samples(MSRD, Subject=="P18")
otu_table(P18_MSRD)
chord_p["MSRD","P18"] = taxa_sums(P18_MSRD)
remove(P18_MSRD)
P18_MLS23S<- subset_samples(MLS23S, Subject=="P18")
otu_table(P18_MLS23S)
chord_p["MLS23S","P18"] = taxa_sums(P18_MLS23S)
remove(P18_MLS23S)
P18_GYRA<- subset_samples(GYRA, Subject=="P18")
otu_table(P18_GYRA)
chord_p["GYRA","P18"] = taxa_sums(P18_GYRA)
remove(P18_GYRA)
P18_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P18")
otu_table(P18_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P18"] = taxa_sums(P18_QACF_ANTIPRIME3)
remove(P18_QACF_ANTIPRIME3)
P18_ERMC<- subset_samples(ERMC, Subject=="P18")
otu_table(P18_ERMC)
chord_p["ERMC","P18"] = taxa_sums(P18_ERMC)
remove(P18_ERMC)



P19_PARE<- subset_samples(PARE, Subject=="P19")
otu_table(P19_PARE)
chord_p["PARE","P19"] = taxa_sums(P19_PARE)
remove(P19_PARE)
P19_MSRD<- subset_samples(MSRD, Subject=="P19")
otu_table(P19_MSRD)
chord_p["MSRD","P19"] = taxa_sums(P19_MSRD)
remove(P19_MSRD)
P19_MLS23S<- subset_samples(MLS23S, Subject=="P19")
otu_table(P19_MLS23S)
chord_p["MLS23S","P19"] = taxa_sums(P19_MLS23S)
remove(P19_MLS23S)
P19_GYRA<- subset_samples(GYRA, Subject=="P19")
otu_table(P19_GYRA)
chord_p["GYRA","P19"] = taxa_sums(P19_GYRA)
remove(P19_GYRA)
P19_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P19")
otu_table(P19_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P19"] = taxa_sums(P19_QACF_ANTIPRIME3)
remove(P19_QACF_ANTIPRIME3)
P19_ERMC<- subset_samples(ERMC, Subject=="P19")
otu_table(P19_ERMC)
chord_p["ERMC","P19"] = taxa_sums(P19_ERMC)
remove(P19_ERMC)

P20_PARE<- subset_samples(PARE, Subject=="P20")
otu_table(P20_PARE)
chord_p["PARE","P20"] = taxa_sums(P20_PARE)
remove(P20_PARE)
P20_MSRD<- subset_samples(MSRD, Subject=="P20")
otu_table(P20_MSRD)
chord_p["MSRD","P20"] = taxa_sums(P20_MSRD)
remove(P20_MSRD)
P20_MLS23S<- subset_samples(MLS23S, Subject=="P20")
otu_table(P20_MLS23S)
chord_p["MLS23S","P20"] = taxa_sums(P20_MLS23S)
remove(P20_MLS23S)
P20_GYRA<- subset_samples(GYRA, Subject=="P20")
otu_table(P20_GYRA)
chord_p["GYRA","P20"] = taxa_sums(P20_GYRA)
remove(P20_GYRA)
P20_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P20")
otu_table(P20_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P20"] = taxa_sums(P20_QACF_ANTIPRIME3)
remove(P20_QACF_ANTIPRIME3)
P20_ERMC<- subset_samples(ERMC, Subject=="P20")
otu_table(P20_ERMC)
chord_p["ERMC","P20"] = taxa_sums(P20_ERMC)
remove(P20_ERMC)

P21_PARE<- subset_samples(PARE, Subject=="P21")
otu_table(P21_PARE)
chord_p["PARE","P21"] = taxa_sums(P21_PARE)
remove(P21_PARE)
P21_MSRD<- subset_samples(MSRD, Subject=="P21")
otu_table(P21_MSRD)
chord_p["MSRD","P21"] = taxa_sums(P21_MSRD)
remove(P21_MSRD)
P21_MLS23S<- subset_samples(MLS23S, Subject=="P21")
otu_table(P21_MLS23S)
chord_p["MLS23S","P21"] = taxa_sums(P21_MLS23S)
remove(P21_MLS23S)
P21_GYRA<- subset_samples(GYRA, Subject=="P21")
otu_table(P21_GYRA)
chord_p["GYRA","P21"] = taxa_sums(P21_GYRA)
remove(P21_GYRA)
P21_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P21")
otu_table(P21_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P21"] = taxa_sums(P21_QACF_ANTIPRIME3)
remove(P21_QACF_ANTIPRIME3)
P21_ERMC<- subset_samples(ERMC, Subject=="P21")
otu_table(P21_ERMC)
chord_p["ERMC","P21"] = taxa_sums(P21_ERMC)
remove(P21_ERMC)



P22_PARE<- subset_samples(PARE, Subject=="P22")
otu_table(P22_PARE)
chord_p["PARE","P22"] = taxa_sums(P22_PARE)
remove(P22_PARE)
P22_MSRD<- subset_samples(MSRD, Subject=="P22")
otu_table(P22_MSRD)
chord_p["MSRD","P22"] = taxa_sums(P22_MSRD)
remove(P22_MSRD)
P22_MLS23S<- subset_samples(MLS23S, Subject=="P22")
otu_table(P22_MLS23S)
chord_p["MLS23S","P22"] = taxa_sums(P22_MLS23S)
remove(P22_MLS23S)
P22_GYRA<- subset_samples(GYRA, Subject=="P22")
otu_table(P22_GYRA)
chord_p["GYRA","P22"] = taxa_sums(P22_GYRA)
remove(P22_GYRA)
P22_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P22")
otu_table(P22_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P22"] = taxa_sums(P22_QACF_ANTIPRIME3)
remove(P22_QACF_ANTIPRIME3)
P22_ERMC<- subset_samples(ERMC, Subject=="P22")
otu_table(P22_ERMC)
chord_p["ERMC","P22"] = taxa_sums(P22_ERMC)
remove(P22_ERMC)



P23_PARE<- subset_samples(PARE, Subject=="P23")
otu_table(P23_PARE)
chord_p["PARE","P23"] = taxa_sums(P23_PARE)
remove(P23_PARE)
P23_MSRD<- subset_samples(MSRD, Subject=="P23")
otu_table(P23_MSRD)
chord_p["MSRD","P23"] = taxa_sums(P23_MSRD)
remove(P23_MSRD)
P23_MLS23S<- subset_samples(MLS23S, Subject=="P23")
otu_table(P23_MLS23S)
chord_p["MLS23S","P23"] = taxa_sums(P23_MLS23S)
remove(P23_MLS23S)
P23_GYRA<- subset_samples(GYRA, Subject=="P23")
otu_table(P23_GYRA)
chord_p["GYRA","P23"] = taxa_sums(P23_GYRA)
remove(P23_GYRA)
P23_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P23")
otu_table(P23_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P23"] = taxa_sums(P23_QACF_ANTIPRIME3)
remove(P23_QACF_ANTIPRIME3)
P23_ERMC<- subset_samples(ERMC, Subject=="P23")
otu_table(P23_ERMC)
chord_p["ERMC","P23"] = taxa_sums(P23_ERMC)
remove(P23_ERMC)



P24_PARE<- subset_samples(PARE, Subject=="P24")
otu_table(P24_PARE)
chord_p["PARE","P24"] = taxa_sums(P24_PARE)
remove(P24_PARE)
P24_MSRD<- subset_samples(MSRD, Subject=="P24")
otu_table(P24_MSRD)
chord_p["MSRD","P24"] = taxa_sums(P24_MSRD)
remove(P24_MSRD)
P24_MLS23S<- subset_samples(MLS23S, Subject=="P24")
otu_table(P24_MLS23S)
chord_p["MLS23S","P24"] = taxa_sums(P24_MLS23S)
remove(P24_MLS23S)
P24_GYRA<- subset_samples(GYRA, Subject=="P24")
otu_table(P24_GYRA)
chord_p["GYRA","P24"] = taxa_sums(P24_GYRA)
remove(P24_GYRA)
P24_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P24")
otu_table(P24_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P24"] = taxa_sums(P24_QACF_ANTIPRIME3)
remove(P24_QACF_ANTIPRIME3)
P24_ERMC<- subset_samples(ERMC, Subject=="P24")
otu_table(P24_ERMC)
chord_p["ERMC","P24"] = taxa_sums(P24_ERMC)
remove(P24_ERMC)



P25_PARE<- subset_samples(PARE, Subject=="P25")
otu_table(P25_PARE)
chord_p["PARE","P25"] = taxa_sums(P25_PARE)
remove(P25_PARE)
P25_MSRD<- subset_samples(MSRD, Subject=="P25")
otu_table(P25_MSRD)
chord_p["MSRD","P25"] = taxa_sums(P25_MSRD)
remove(P25_MSRD)
P25_MLS23S<- subset_samples(MLS23S, Subject=="P25")
otu_table(P25_MLS23S)
chord_p["MLS23S","P25"] = taxa_sums(P25_MLS23S)
remove(P25_MLS23S)
P25_GYRA<- subset_samples(GYRA, Subject=="P25")
otu_table(P25_GYRA)
chord_p["GYRA","P25"] = taxa_sums(P25_GYRA)
remove(P25_GYRA)
P25_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P25")
otu_table(P25_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P25"] = taxa_sums(P25_QACF_ANTIPRIME3)
remove(P25_QACF_ANTIPRIME3)
P25_ERMC<- subset_samples(ERMC, Subject=="P25")
otu_table(P25_ERMC)
chord_p["ERMC","P25"] = taxa_sums(P25_ERMC)
remove(P25_ERMC)



P26_PARE<- subset_samples(PARE, Subject=="P26")
otu_table(P26_PARE)
chord_p["PARE","P26"] = taxa_sums(P26_PARE)
remove(P26_PARE)
P26_MSRD<- subset_samples(MSRD, Subject=="P26")
otu_table(P26_MSRD)
chord_p["MSRD","P26"] = taxa_sums(P26_MSRD)
remove(P26_MSRD)
P26_MLS23S<- subset_samples(MLS23S, Subject=="P26")
otu_table(P26_MLS23S)
chord_p["MLS23S","P26"] = taxa_sums(P26_MLS23S)
remove(P26_MLS23S)
P26_GYRA<- subset_samples(GYRA, Subject=="P26")
otu_table(P26_GYRA)
chord_p["GYRA","P26"] = taxa_sums(P26_GYRA)
remove(P26_GYRA)
P26_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P26")
otu_table(P26_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P26"] = taxa_sums(P26_QACF_ANTIPRIME3)
remove(P26_QACF_ANTIPRIME3)
P26_ERMC<- subset_samples(ERMC, Subject=="P26")
otu_table(P26_ERMC)
chord_p["ERMC","P26"] = taxa_sums(P26_ERMC)
remove(P26_ERMC)



P27_PARE<- subset_samples(PARE, Subject=="P27")
otu_table(P27_PARE)
chord_p["PARE","P27"] = taxa_sums(P27_PARE)
remove(P27_PARE)
P27_MSRD<- subset_samples(MSRD, Subject=="P27")
otu_table(P27_MSRD)
chord_p["MSRD","P27"] = taxa_sums(P27_MSRD)
remove(P27_MSRD)
P27_MLS23S<- subset_samples(MLS23S, Subject=="P27")
otu_table(P27_MLS23S)
chord_p["MLS23S","P27"] = taxa_sums(P27_MLS23S)
remove(P27_MLS23S)
P27_GYRA<- subset_samples(GYRA, Subject=="P27")
otu_table(P27_GYRA)
chord_p["GYRA","P27"] = taxa_sums(P27_GYRA)
remove(P27_GYRA)
P27_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P27")
otu_table(P27_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P27"] = taxa_sums(P27_QACF_ANTIPRIME3)
remove(P27_QACF_ANTIPRIME3)
P27_ERMC<- subset_samples(ERMC, Subject=="P27")
otu_table(P27_ERMC)
chord_p["ERMC","P27"] = taxa_sums(P27_ERMC)
remove(P27_ERMC)



P28_PARE<- subset_samples(PARE, Subject=="P28")
otu_table(P28_PARE)
chord_p["PARE","P28"] = taxa_sums(P28_PARE)
remove(P28_PARE)
P28_MSRD<- subset_samples(MSRD, Subject=="P28")
otu_table(P28_MSRD)
chord_p["MSRD","P28"] = taxa_sums(P28_MSRD)
remove(P28_MSRD)
P28_MLS23S<- subset_samples(MLS23S, Subject=="P28")
otu_table(P28_MLS23S)
chord_p["MLS23S","P28"] = taxa_sums(P28_MLS23S)
remove(P28_MLS23S)
P28_GYRA<- subset_samples(GYRA, Subject=="P28")
otu_table(P28_GYRA)
chord_p["GYRA","P28"] = taxa_sums(P28_GYRA)
remove(P28_GYRA)
P28_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P28")
otu_table(P28_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P28"] = taxa_sums(P28_QACF_ANTIPRIME3)
remove(P28_QACF_ANTIPRIME3)
P28_ERMC<- subset_samples(ERMC, Subject=="P28")
otu_table(P28_ERMC)
chord_p["ERMC","P28"] = taxa_sums(P28_ERMC)
remove(P28_ERMC)



P29_PARE<- subset_samples(PARE, Subject=="P29")
otu_table(P29_PARE)
chord_p["PARE","P29"] = taxa_sums(P29_PARE)
remove(P29_PARE)
P29_MSRD<- subset_samples(MSRD, Subject=="P29")
otu_table(P29_MSRD)
chord_p["MSRD","P29"] = taxa_sums(P29_MSRD)
remove(P29_MSRD)
P29_MLS23S<- subset_samples(MLS23S, Subject=="P29")
otu_table(P29_MLS23S)
chord_p["MLS23S","P29"] = taxa_sums(P29_MLS23S)
remove(P29_MLS23S)
P29_GYRA<- subset_samples(GYRA, Subject=="P29")
otu_table(P29_GYRA)
chord_p["GYRA","P29"] = taxa_sums(P29_GYRA)
remove(P29_GYRA)
P29_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P29")
otu_table(P29_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P29"] = taxa_sums(P29_QACF_ANTIPRIME3)
remove(P29_QACF_ANTIPRIME3)
P29_ERMC<- subset_samples(ERMC, Subject=="P29")
otu_table(P29_ERMC)
chord_p["ERMC","P29"] = taxa_sums(P29_ERMC)
remove(P29_ERMC)



P30_PARE<- subset_samples(PARE, Subject=="P30")
otu_table(P30_PARE)
chord_p["PARE","P30"] = taxa_sums(P30_PARE)
remove(P30_PARE)
P30_MSRD<- subset_samples(MSRD, Subject=="P30")
otu_table(P30_MSRD)
chord_p["MSRD","P30"] = taxa_sums(P30_MSRD)
remove(P30_MSRD)
P30_MLS23S<- subset_samples(MLS23S, Subject=="P30")
otu_table(P30_MLS23S)
chord_p["MLS23S","P30"] = taxa_sums(P30_MLS23S)
remove(P30_MLS23S)
P30_GYRA<- subset_samples(GYRA, Subject=="P30")
otu_table(P30_GYRA)
chord_p["GYRA","P30"] = taxa_sums(P30_GYRA)
remove(P30_GYRA)
P30_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P30")
otu_table(P30_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P30"] = taxa_sums(P30_QACF_ANTIPRIME3)
remove(P30_QACF_ANTIPRIME3)
P30_ERMC<- subset_samples(ERMC, Subject=="P30")
otu_table(P30_ERMC)
chord_p["ERMC","P30"] = taxa_sums(P30_ERMC)
remove(P30_ERMC)



P31_PARE<- subset_samples(PARE, Subject=="P31")
otu_table(P31_PARE)
chord_p["PARE","P31"] = taxa_sums(P31_PARE)
remove(P31_PARE)
P31_MSRD<- subset_samples(MSRD, Subject=="P31")
otu_table(P31_MSRD)
chord_p["MSRD","P31"] = taxa_sums(P31_MSRD)
remove(P31_MSRD)
P31_MLS23S<- subset_samples(MLS23S, Subject=="P31")
otu_table(P31_MLS23S)
chord_p["MLS23S","P31"] = taxa_sums(P31_MLS23S)
remove(P31_MLS23S)
P31_GYRA<- subset_samples(GYRA, Subject=="P31")
otu_table(P31_GYRA)
chord_p["GYRA","P31"] = taxa_sums(P31_GYRA)
remove(P31_GYRA)
P31_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P31")
otu_table(P31_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P31"] = taxa_sums(P31_QACF_ANTIPRIME3)
remove(P31_QACF_ANTIPRIME3)
P31_ERMC<- subset_samples(ERMC, Subject=="P31")
otu_table(P31_ERMC)
chord_p["ERMC","P31"] = taxa_sums(P31_ERMC)
remove(P31_ERMC)



P32_PARE<- subset_samples(PARE, Subject=="P32")
otu_table(P32_PARE)
chord_p["PARE","P32"] = taxa_sums(P32_PARE)
remove(P32_PARE)
P32_MSRD<- subset_samples(MSRD, Subject=="P32")
otu_table(P32_MSRD)
chord_p["MSRD","P32"] = taxa_sums(P32_MSRD)
remove(P32_MSRD)
P32_MLS23S<- subset_samples(MLS23S, Subject=="P32")
otu_table(P32_MLS23S)
chord_p["MLS23S","P32"] = taxa_sums(P32_MLS23S)
remove(P32_MLS23S)
P32_GYRA<- subset_samples(GYRA, Subject=="P32")
otu_table(P32_GYRA)
chord_p["GYRA","P32"] = taxa_sums(P32_GYRA)
remove(P32_GYRA)
P32_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P32")
otu_table(P32_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P32"] = taxa_sums(P32_QACF_ANTIPRIME3)
remove(P32_QACF_ANTIPRIME3)
P32_ERMC<- subset_samples(ERMC, Subject=="P32")
otu_table(P32_ERMC)
chord_p["ERMC","P32"] = taxa_sums(P32_ERMC)
remove(P32_ERMC)



P34_PARE<- subset_samples(PARE, Subject=="P34")
otu_table(P34_PARE)
chord_p["PARE","P34"] = taxa_sums(P34_PARE)
remove(P34_PARE)
P34_MSRD<- subset_samples(MSRD, Subject=="P34")
otu_table(P34_MSRD)
chord_p["MSRD","P34"] = taxa_sums(P34_MSRD)
remove(P34_MSRD)
P34_MLS23S<- subset_samples(MLS23S, Subject=="P34")
otu_table(P34_MLS23S)
chord_p["MLS23S","P34"] = taxa_sums(P34_MLS23S)
remove(P34_MLS23S)
P34_GYRA<- subset_samples(GYRA, Subject=="P34")
otu_table(P34_GYRA)
chord_p["GYRA","P34"] = taxa_sums(P34_GYRA)
remove(P34_GYRA)
P34_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P34")
otu_table(P34_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P34"] = taxa_sums(P34_QACF_ANTIPRIME3)
remove(P34_QACF_ANTIPRIME3)
P34_ERMC<- subset_samples(ERMC, Subject=="P34")
otu_table(P34_ERMC)
chord_p["ERMC","P34"] = taxa_sums(P34_ERMC)
remove(P34_ERMC)



P35_PARE<- subset_samples(PARE, Subject=="P35")
otu_table(P35_PARE)
chord_p["PARE","P35"] = taxa_sums(P35_PARE)
remove(P35_PARE)
P35_MSRD<- subset_samples(MSRD, Subject=="P35")
otu_table(P35_MSRD)
chord_p["MSRD","P35"] = taxa_sums(P35_MSRD)
remove(P35_MSRD)
P35_MLS23S<- subset_samples(MLS23S, Subject=="P35")
otu_table(P35_MLS23S)
chord_p["MLS23S","P35"] = taxa_sums(P35_MLS23S)
remove(P35_MLS23S)
P35_GYRA<- subset_samples(GYRA, Subject=="P35")
otu_table(P35_GYRA)
chord_p["GYRA","P35"] = taxa_sums(P35_GYRA)
remove(P35_GYRA)
P35_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P35")
otu_table(P35_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P35"] = taxa_sums(P35_QACF_ANTIPRIME3)
remove(P35_QACF_ANTIPRIME3)
P35_ERMC<- subset_samples(ERMC, Subject=="P35")
otu_table(P35_ERMC)
chord_p["ERMC","P35"] = taxa_sums(P35_ERMC)
remove(P35_ERMC)



P36_PARE<- subset_samples(PARE, Subject=="P36")
otu_table(P36_PARE)
chord_p["PARE","P36"] = taxa_sums(P36_PARE)
remove(P36_PARE)
P36_MSRD<- subset_samples(MSRD, Subject=="P36")
otu_table(P36_MSRD)
chord_p["MSRD","P36"] = taxa_sums(P36_MSRD)
remove(P36_MSRD)
P36_MLS23S<- subset_samples(MLS23S, Subject=="P36")
otu_table(P36_MLS23S)
chord_p["MLS23S","P36"] = taxa_sums(P36_MLS23S)
remove(P36_MLS23S)
P36_GYRA<- subset_samples(GYRA, Subject=="P36")
otu_table(P36_GYRA)
chord_p["GYRA","P36"] = taxa_sums(P36_GYRA)
remove(P36_GYRA)
P36_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P36")
otu_table(P36_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P36"] = taxa_sums(P36_QACF_ANTIPRIME3)
remove(P36_QACF_ANTIPRIME3)
P36_ERMC<- subset_samples(ERMC, Subject=="P36")
otu_table(P36_ERMC)
chord_p["ERMC","P36"] = taxa_sums(P36_ERMC)
remove(P36_ERMC)



P37_PARE<- subset_samples(PARE, Subject=="P37")
otu_table(P37_PARE)
chord_p["PARE","P37"] = taxa_sums(P37_PARE)
remove(P37_PARE)
P37_MSRD<- subset_samples(MSRD, Subject=="P37")
otu_table(P37_MSRD)
chord_p["MSRD","P37"] = taxa_sums(P37_MSRD)
remove(P37_MSRD)
P37_MLS23S<- subset_samples(MLS23S, Subject=="P37")
otu_table(P37_MLS23S)
chord_p["MLS23S","P37"] = taxa_sums(P37_MLS23S)
remove(P37_MLS23S)
P37_GYRA<- subset_samples(GYRA, Subject=="P37")
otu_table(P37_GYRA)
chord_p["GYRA","P37"] = taxa_sums(P37_GYRA)
remove(P37_GYRA)
P37_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P37")
otu_table(P37_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P37"] = taxa_sums(P37_QACF_ANTIPRIME3)
remove(P37_QACF_ANTIPRIME3)
P37_ERMC<- subset_samples(ERMC, Subject=="P37")
otu_table(P37_ERMC)
chord_p["ERMC","P37"] = taxa_sums(P37_ERMC)
remove(P37_ERMC)



P38_PARE<- subset_samples(PARE, Subject=="P38")
otu_table(P38_PARE)
chord_p["PARE","P38"] = taxa_sums(P38_PARE)
remove(P38_PARE)
P38_MSRD<- subset_samples(MSRD, Subject=="P38")
otu_table(P38_MSRD)
chord_p["MSRD","P38"] = taxa_sums(P38_MSRD)
remove(P38_MSRD)
P38_MLS23S<- subset_samples(MLS23S, Subject=="P38")
otu_table(P38_MLS23S)
chord_p["MLS23S","P38"] = taxa_sums(P38_MLS23S)
remove(P38_MLS23S)
P38_GYRA<- subset_samples(GYRA, Subject=="P38")
otu_table(P38_GYRA)
chord_p["GYRA","P38"] = taxa_sums(P38_GYRA)
remove(P38_GYRA)
P38_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P38")
otu_table(P38_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P38"] = taxa_sums(P38_QACF_ANTIPRIME3)
remove(P38_QACF_ANTIPRIME3)
P38_ERMC<- subset_samples(ERMC, Subject=="P38")
otu_table(P38_ERMC)
chord_p["ERMC","P38"] = taxa_sums(P38_ERMC)
remove(P38_ERMC)



P39_PARE<- subset_samples(PARE, Subject=="P39")
otu_table(P39_PARE)
chord_p["PARE","P39"] = taxa_sums(P39_PARE)
remove(P39_PARE)
P39_MSRD<- subset_samples(MSRD, Subject=="P39")
otu_table(P39_MSRD)
chord_p["MSRD","P39"] = taxa_sums(P39_MSRD)
remove(P39_MSRD)
P39_MLS23S<- subset_samples(MLS23S, Subject=="P39")
otu_table(P39_MLS23S)
chord_p["MLS23S","P39"] = taxa_sums(P39_MLS23S)
remove(P39_MLS23S)
P39_GYRA<- subset_samples(GYRA, Subject=="P39")
otu_table(P39_GYRA)
chord_p["GYRA","P39"] = taxa_sums(P39_GYRA)
remove(P39_GYRA)
P39_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P39")
otu_table(P39_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P39"] = taxa_sums(P39_QACF_ANTIPRIME3)
remove(P39_QACF_ANTIPRIME3)
P39_ERMC<- subset_samples(ERMC, Subject=="P39")
otu_table(P39_ERMC)
chord_p["ERMC","P39"] = taxa_sums(P39_ERMC)
remove(P39_ERMC)



P40_PARE<- subset_samples(PARE, Subject=="P40")
otu_table(P40_PARE)
chord_p["PARE","P40"] = taxa_sums(P40_PARE)
remove(P40_PARE)
P40_MSRD<- subset_samples(MSRD, Subject=="P40")
otu_table(P40_MSRD)
chord_p["MSRD","P40"] = taxa_sums(P40_MSRD)
remove(P40_MSRD)
P40_MLS23S<- subset_samples(MLS23S, Subject=="P40")
otu_table(P40_MLS23S)
chord_p["MLS23S","P40"] = taxa_sums(P40_MLS23S)
remove(P40_MLS23S)
P40_GYRA<- subset_samples(GYRA, Subject=="P40")
otu_table(P40_GYRA)
chord_p["GYRA","P40"] = taxa_sums(P40_GYRA)
remove(P40_GYRA)
P40_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P40")
otu_table(P40_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P40"] = taxa_sums(P40_QACF_ANTIPRIME3)
remove(P40_QACF_ANTIPRIME3)
P40_ERMC<- subset_samples(ERMC, Subject=="P40")
otu_table(P40_ERMC)
chord_p["ERMC","P40"] = taxa_sums(P40_ERMC)
remove(P40_ERMC)



P41_PARE<- subset_samples(PARE, Subject=="P41")
otu_table(P41_PARE)
chord_p["PARE","P41"] = taxa_sums(P41_PARE)
remove(P41_PARE)
P41_MSRD<- subset_samples(MSRD, Subject=="P41")
otu_table(P41_MSRD)
chord_p["MSRD","P41"] = taxa_sums(P41_MSRD)
remove(P41_MSRD)
P41_MLS23S<- subset_samples(MLS23S, Subject=="P41")
otu_table(P41_MLS23S)
chord_p["MLS23S","P41"] = taxa_sums(P41_MLS23S)
remove(P41_MLS23S)
P41_GYRA<- subset_samples(GYRA, Subject=="P41")
otu_table(P41_GYRA)
chord_p["GYRA","P41"] = taxa_sums(P41_GYRA)
remove(P41_GYRA)
P41_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P41")
otu_table(P41_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P41"] = taxa_sums(P41_QACF_ANTIPRIME3)
remove(P41_QACF_ANTIPRIME3)
P41_ERMC<- subset_samples(ERMC, Subject=="P41")
otu_table(P41_ERMC)
chord_p["ERMC","P41"] = taxa_sums(P41_ERMC)
remove(P41_ERMC)



P42_PARE<- subset_samples(PARE, Subject=="P42")
otu_table(P42_PARE)
chord_p["PARE","P42"] = taxa_sums(P42_PARE)
remove(P42_PARE)
P42_MSRD<- subset_samples(MSRD, Subject=="P42")
otu_table(P42_MSRD)
chord_p["MSRD","P42"] = taxa_sums(P42_MSRD)
remove(P42_MSRD)
P42_MLS23S<- subset_samples(MLS23S, Subject=="P42")
otu_table(P42_MLS23S)
chord_p["MLS23S","P42"] = taxa_sums(P42_MLS23S)
remove(P42_MLS23S)
P42_GYRA<- subset_samples(GYRA, Subject=="P42")
otu_table(P42_GYRA)
chord_p["GYRA","P42"] = taxa_sums(P42_GYRA)
remove(P42_GYRA)
P42_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P42")
otu_table(P42_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P42"] = taxa_sums(P42_QACF_ANTIPRIME3)
remove(P42_QACF_ANTIPRIME3)
P42_ERMC<- subset_samples(ERMC, Subject=="P42")
otu_table(P42_ERMC)
chord_p["ERMC","P42"] = taxa_sums(P42_ERMC)
remove(P42_ERMC)



P43_PARE<- subset_samples(PARE, Subject=="P43")
otu_table(P43_PARE)
chord_p["PARE","P43"] = taxa_sums(P43_PARE)
remove(P43_PARE)
P43_MSRD<- subset_samples(MSRD, Subject=="P43")
otu_table(P43_MSRD)
chord_p["MSRD","P43"] = taxa_sums(P43_MSRD)
remove(P43_MSRD)
P43_MLS23S<- subset_samples(MLS23S, Subject=="P43")
otu_table(P43_MLS23S)
chord_p["MLS23S","P43"] = taxa_sums(P43_MLS23S)
remove(P43_MLS23S)
P43_GYRA<- subset_samples(GYRA, Subject=="P43")
otu_table(P43_GYRA)
chord_p["GYRA","P43"] = taxa_sums(P43_GYRA)
remove(P43_GYRA)
P43_QACF_ANTIPRIME3<- subset_samples(QACF_ANTIPRIME3, Subject=="P43")
otu_table(P43_QACF_ANTIPRIME3)
chord_p["QACF_ANTIPRIME3","P43"] = taxa_sums(P43_QACF_ANTIPRIME3)
remove(P43_QACF_ANTIPRIME3)
P43_ERMC<- subset_samples(ERMC, Subject=="P43")
otu_table(P43_ERMC)
chord_p["ERMC","P43"] = taxa_sums(P43_ERMC)
remove(P43_ERMC)
```
```{r}
chord_p$AMR_Gene <- rownames(chord_p)
chord_p_fix <- pivot_longer(chord_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
# parameters
circos.clear()
circos.par(start.degree = 90, gap.degree = 1, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))


# Base plot
chordDiagram(
  x = chord_p, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

```
```{r}

#parameters
circos.clear()
circos.par(start.degree = -40, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))
plot.new()
circle_size = unit(1, "snpc") # snpc unit gives you a square region

pushViewport(viewport(x = 0, y = 0.5, width = circle_size, height = circle_size,
    just = c("left", "center")))
par(omi = gridOMI(), new = TRUE)


#legend_col <- distinctColorPalette(k = length(unique(chord_p_fix$AMR_Gene)), altCol = FALSE, runTsne = FALSE)
#legend_col
# Base plot
#png("images/overall2.png", units = "in", height = 7, width = 5, res = 500)
chordDiagram(
  x = chord_p_fix, 
  transparency = 0.25,
  directional = 1,
  grid.col  = c(ERMC = "blue", PARE = "red", MSRD = "green", MLS23S = "black", GYRA = "purple", QACF_ANTIPRIME3 = "orange"),
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(1)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)

lgd_col = Legend(
labels = c("ERMC", "PARE", "MSRD", "MLS23S", "GYRA", "QACF_ANTIPRIME3" ),                 
title = "AMR Gene",                 
type = "point",
background = unique(grid.col))
upViewport()
draw(lgd_col, x = circle_size, just = "left")
#dev.off()

```

```{r}
ERMC_p <- chord_p[!(row.names(chord_p) %in% c("PARE","MSRD","MLS23S","GYRA","QACF_ANTIPRIME3")), ]
ERMC_p$AMR_Gene <- rownames(ERMC_p)
ERMC_p_fix <- pivot_longer(ERMC_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
QACF_ANTIPRIME3_gene_stats2 <- as.data.frame(QACF_ANTIPRIME3_gene_stats)
summary(ss.lm <- lm(QACF_ANTIPRIME3 ~ Time_Week, QACF_ANTIPRIME3_gene_stats2))
ssBase <- ggplot(QACF_ANTIPRIME3_gene_stats2, aes(x = Time_Week, y = QACF_ANTIPRIME3)) + geom_point(aes(color = Location)) + 
    scale_color_discrete() + labs(x = "Days with no sleep", y = "Reaction time [ms]")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -55, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/ERMC.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = ERMC_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```

```{r}
PARE_p <- chord_p[!(row.names(chord_p) %in% c("ERMC","MSRD","MLS23S","GYRA","QACF_ANTIPRIME3")), ]
PARE_p$AMR_Gene <- rownames(PARE_p)
PARE_p_fix <- pivot_longer(PARE_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -55, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/PARE.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = PARE_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```

```{r}
MSRD_p <- chord_p[!(row.names(chord_p) %in% c("ERMC","PARE","MLS23S","GYRA","QACF_ANTIPRIME3")), ]
MSRD_p$AMR_Gene <- rownames(MSRD_p)
MSRD_p_fix <- pivot_longer(MSRD_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -55, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/MSRD.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = MSRD_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```

```{r}
MLS23S_p <- chord_p[!(row.names(chord_p) %in% c("ERMC","PARE","MSRD","GYRA","QACF_ANTIPRIME3")), ]
MLS23S_p$AMR_Gene <- rownames(MLS23S_p)
MLS23S_p_fix <- pivot_longer(MLS23S_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -55, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/MLS23S.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = MLS23S_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```

```{r}
GYRA_p <- chord_p[!(row.names(chord_p) %in% c("ERMC","PARE","MSRD","MLS23S", "QACF_ANTIPRIME3")), ]
GYRA_p$AMR_Gene <- rownames(GYRA_p)
GYRA_p_fix <- pivot_longer(GYRA_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -55, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/GYRA_chord.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = GYRA_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)

# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```

```{r}
QACF_ANTIPRIME3_p <- chord_p[!(row.names(chord_p) %in% c("ERMC","PARE","MSRD","MLS23S", "GYRA")), ]
QACF_ANTIPRIME3_p$AMR_Gene <- rownames(QACF_ANTIPRIME3_p)
QACF_ANTIPRIME3_p_fix <- pivot_longer(QACF_ANTIPRIME3_p, cols =1:42, names_to = "Subject", values_to = "Abundance"  )
```

```{r}
#parameters
circos.clear()
circos.par(start.degree = -40, gap.degree = 5, track.margin = c(-0.1, 0.1), points.overflow.warning = FALSE)
par(mar = rep(0, 4))

png("images/QACF_chord.png", units = "in", height = 7, width = 5, res = 500)
# Base plot
chordDiagram(
  x = QACF_ANTIPRIME3_p_fix, 
  transparency = 0.25,
  directional = 1,
  direction.type = c("arrows", "diffHeight"), 
  diffHeight  = -0.04,
  annotationTrack = "grid", 
  annotationTrackHeight = c(0.05, 0.1),
  link.arr.type = "big.arrow", 
  link.sort = TRUE, 
  link.largest.ontop = TRUE)


# Add text and axis
circos.trackPlotRegion(
  track.index = 1, 
  bg.border = NA, 
  panel.fun = function(x, y) {
    
    xlim = get.cell.meta.data("xlim")
    sector.index = get.cell.meta.data("sector.index")
    
    # Add names to the sector. 
    circos.text(
      x = mean(xlim), 
      y = 2, 
      labels = sector.index, 
      adj = c(0, degree(5)), 
      facing = "clockwise",
      cex = 0.6
      )
  }
)
dev.off()
```


```{r}
library("RSQLite")
library(D3GB)
?
gb <- genomebrowser(getAssemblyFromFasta("ERMC.fa"), dir = "ERMC_phage_genomebrowser")
genome_addSequence(gb,"ERMC.fa")

genome_addGFF(gb,"ERMC.gff3")

```
