---
title: "Time_Abundance_paper1"
output: html_document
---


```{r setup,message=FALSE}
library(readr)
library(dplyr)
library(tidyr)
library(ggplot2)
library(phyloseq)
library(cowplot)
```
## Phyloseq Object Generation for Diversity Analysis
This is the same phyloseq object titled Profile that was generated in the 14_Within_vs_Between_Dissimilarity_Fig6.Rmd script

Metadata for all Phyloseq objects produced in this script
```{r}
Metadata2 <- read_csv("~/Desktop/Current_work/OTU_Builber/Sequencing_Data_Info_Sheet3.csv")
Metadata3 <- Metadata2
Metadata3$Time <- Metadata3$Time_Collected
Metadata3$Time[which(Metadata3$Time == "T_0")] <- "00_Weeks"
Metadata3$Time[which(Metadata3$Time == "2_week")] <- "02_Weeks"
Metadata3$Time[which(Metadata3$Time == "1_month")] <- "04_Weeks"
Metadata3$Time[which(Metadata3$Time == "3_month")] <- "12_Weeks"
Metadata3$Time[which(Metadata3$Time == "6_month")] <- "24_Weeks"
META <- sample_data(Metadata3)
sample_names(META)= META$SampleID
```
Make Phyloseq object for stable Set A CheckV identified and annotated viral species Makers 
```{r,message=FALSE,warning=FALSE,results="hide"}
CHeckV_counts <- read.csv("counts2_bowtie.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
CHeckV_counts  <- subset(CHeckV_counts , select = -c(length))
tax_ann <- read.csv("vc_tax.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
CHeckV_counts2 <- merge(tax_ann, CHeckV_counts, by.x = "row.names", by.y = "row.names")
CHeckV_counts2<- subset (CHeckV_counts2, select = -c(Superkingdom, Realm, Kingdom, Phylum, Class, Family, Order,Subfamily, Genus))
CHeckV_spec<-c("Autographa californica multiple nucleopolyhedrovirus","Autographa californica nucleopolyhedrovirus","Staphylococcus phage St 134","Unclassified Caudovirales","Unclassified Homo sapiens like virus","Unclassified Siphoviridae","Uncultured Caudovirales phage")
CHeckV_counts2<-subset(CHeckV_counts2, Species %in% CHeckV_spec)
CHeckV_counts2<-ddply(CHeckV_counts2,"Species",numcolwise(sum))
row.names(CHeckV_counts2)<- CHeckV_counts2$Species
CHeckV_counts2<- subset (CHeckV_counts2, select = -c(Species))
CHeckV_OTU <- otu_table(CHeckV_counts2, taxa_are_rows = TRUE)
tax_ann <- read.csv("vc_tax.csv", quote = "", header = TRUE, stringsAsFactors = FALSE,row.names=NULL)
tax_ann<-subset(tax_ann, Species %in% CHeckV_spec)
tax_ann<- subset (tax_ann, select = -c(contig_id))
tax_ann<-unique(tax_ann)
remv<-c("4","70")
tax_ann<-subset(tax_ann, !row.names(tax_ann) %in% remv)
row.names(tax_ann)<-tax_ann$Species
CHeckV_tax <-as.matrix(CHeckV_tax)
CHeckV_tax <- tax_table(CHeckV_tax)
CHeckV_PHY <- phyloseq(CHeckV_OTU, CHeckV_tax, META)
```

Make Phyloseq object for stable Set B Caudovirales Makers 
```{r,message=FALSE,warning=FALSE,results="hide"}
CAUDO_counts <- read.csv("counts2_CAUDO.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
CAUDO_tax <- read.csv("CAUDO_tax.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
CAUDO_OTU <- merge(CAUDO_tax, CAUDO_counts, by.x="row.names", by.y="row.names")
CAUDO_OTU<- subset (CAUDO_OTU, select = -c(Superkingdom, Realm, Kingdom, Phylum, Class, Order, Family, Subfamily, Genus))
CAUDO_spec<-c("Bacillus phage Stitch","Caudovirales sp.","Escherichia phage T7","Escherichia virus DE3","Escherichia virus Lambda","Escherichia virus T7","Escherichia virus TH38","Marine virus AFVG_250M104","Marine virus AFVG_25M557","Staphylococcus phage HOB 14.1.R1","Staphylococcus phage PhiSepi-HH3","Staphylococcus phage SPbeta-like","Staphylococcus phage vB_SauH_DELF3","Staphylococcus virus IPLA7","Staphylococcus virus IPLAC1C","Staphylococcus virus PH15","Staphylococcus virus vB_SepS_456","Staphylococcus virus vB_SepS_459","Staphylococcus virus vB_SepS_E72","Streptococcus phage phi-SC181","Streptococcus phage phiJH1301-2","Uncultured Caudovirales phage")
CAUDO_OTU<-subset(CAUDO_OTU, Species %in% CAUDO_spec)
CAUDO_OTU<-ddply(CAUDO_OTU,"Species",numcolwise(sum))
row.names(CAUDO_OTU)<-CAUDO_OTU$Species
CAUDO_OTU<- subset (CAUDO_OTU, select = -c(Species))
CAUDO_OTU <- otu_table(CAUDO_OTU, taxa_are_rows = TRUE)
CAUDO_tax <- read.csv("CAUDO_tax.csv", quote = "", header = TRUE, stringsAsFactors = FALSE)
CAUDO_tax<-subset(CAUDO_tax, Species %in% CAUDO_spec)
CAUDO_tax<- subset (CAUDO_tax, select = -c(ContigID))
CAUDO_tax<-unique(CAUDO_tax)
row.names(CAUDO_tax)<-CAUDO_tax$Species
CAUDO_tax <-as.matrix(CAUDO_tax)
CAUDO_tax <- tax_table(CAUDO_tax)
CAUDO_PHY <- phyloseq(CAUDO_OTU, CAUDO_tax, META)
```

Make Phyloseq object for stable Set B Papillomaviridae Makers 
```{r,message=FALSE,warning=FALSE,results="hide"}
PAP_counts <- read.csv("papilloma/PAPcounts.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
PAP_tax <- read.csv("papilloma/papilloma_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
PAP_tax <- subset(PAP_tax, select = -c(Serotype,Isolate))
PAP_OTU <- merge(PAP_tax, PAP_counts, by.x="row.names", by.y="row.names")
PAP_OTU<- subset (PAP_OTU, select = -c(Superkingdom, Realm, Kingdom, Phylum, Class, Order, Family, Subfamily, Genus))
PAP_spec<-c("Alphapapillomavirus 5","Betapapillomavirus 1","Betapapillomavirus 2","Betapapillomavirus 3","Gammapapillomavirus 12","Gammapapillomavirus 22","Gammapapillomavirus 7","Gammapapillomavirus 8")
PAP_OTU<-subset(PAP_OTU, Species %in% PAP_spec)
PAP_OTU<-ddply(PAP_OTU,"Species",numcolwise(sum))
row.names(PAP_OTU)<-PAP_OTU$Species
PAP_OTU<- subset (PAP_OTU, select = -c(Species))
PAP_OTU <- otu_table(PAP_OTU, taxa_are_rows = TRUE)
PAP_tax <- read.csv("papilloma/papilloma_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE)
PAP_tax <- subset(PAP_tax, select = -c(Serotype,Isolate))
PAP_tax<-subset(PAP_tax, Species %in% PAP_spec)
PAP_tax<- subset (PAP_tax, select = -c(ContigID))
PAP_tax<-unique(PAP_tax)
remv<-c("1341")
PAP_tax<-subset(PAP_tax, !row.names(PAP_tax) %in% remv)
row.names(PAP_tax)<-PAP_tax$Species
PAP_tax <-as.matrix(PAP_tax)
PAP_tax <- tax_table(PAP_tax)
PAP_PHY <- phyloseq(PAP_OTU, PAP_tax, META)
```

Make Phyloseq object for stable Set B Baculoviridae Makers 
```{r,message=FALSE,warning=FALSE,results="hide"}
BAC_counts <- read.csv("Baculo/counts2_baculo_edited.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
BAC_tax <- read.csv("Baculo/baculo_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
BAC_tax <- subset(BAC_tax, select = -c(Isolate))
BAC_OTU <- merge(BAC_tax, BAC_counts, by.x="row.names", by.y="row.names")
BAC_OTU<- subset (BAC_OTU, select = -c(Superkingdom, Realm, Kingdom, Phylum, Class, Order, Family, Subfamily, Genus))
BAC_spec<-c("Autographa californica multiple nucleopolyhedrovirus","Malacosoma neustria nucleopolyhedrovirus","Unclassified Baculoviridae")
BAC_OTU<-subset(BAC_OTU, Species %in% BAC_spec)
BAC_OTU<-ddply(BAC_OTU,"Species",numcolwise(sum))
row.names(BAC_OTU)<-BAC_OTU$Species
BAC_OTU<- subset (BAC_OTU, select = -c(Species))
BAC_OTU <- otu_table(BAC_OTU, taxa_are_rows = TRUE)
BAC_tax <- read.csv("Baculo/baculo_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
BAC_tax <- subset(BAC_tax, select = -c(Isolate))
BAC_tax<-subset(BAC_tax, Species %in% BAC_spec)
BAC_tax<-unique(BAC_tax)
row.names(BAC_tax)<-BAC_tax$Species
BAC_tax <-as.matrix(BAC_tax)
BAC_tax <- tax_table(BAC_tax)
BAC_PHY <- phyloseq(BAC_OTU, BAC_tax, META)
```

Make Phyloseq object for stable Set B Genomoviridae Makers 
```{r,message=FALSE,warning=FALSE,results="hide"}
GEN_counts <- read.csv("Genomovir/counts2_genomovir_edited.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
GEN_tax <- read.csv("Genomovir/genomovir_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
GEN_tax <- subset(GEN_tax, select = -c(Isolate))
GEN_OTU <- merge(GEN_tax, GEN_counts, by.x="row.names", by.y="row.names")
GEN_OTU<- subset (GEN_OTU, select = -c(Superkingdom, Realm, Kingdom, Phylum, Class, Order, Family, Subfamily, Genus))
GEN_spec<-c("Genomoviridae sp.")
GEN_OTU<-subset(GEN_OTU, Species %in% GEN_spec)
GEN_OTU<-ddply(GEN_OTU,"Species",numcolwise(sum))
row.names(GEN_OTU)<-GEN_OTU$Species
GEN_OTU<- subset (GEN_OTU, select = -c(Species))
GEN_OTU <- otu_table(GEN_OTU, taxa_are_rows = TRUE)
GEN_tax <- read.csv("Genomovir/genomovir_taxa.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
GEN_tax <- subset(GEN_tax, select = -c(Isolate))
GEN_tax<-subset(GEN_tax, Species %in% GEN_spec)
GEN_tax<-unique(GEN_tax)
row.names(GEN_tax)<-GEN_tax$Species
GEN_tax <-as.matrix(GEN_tax)
GEN_tax <- tax_table(GEN_tax)
GEN_PHY <- phyloseq(GEN_OTU, GEN_tax, META)
```

Make Phyloseq object for stable Set C contigs
```{r,message=FALSE,warning=FALSE,results="hide"}
Contigs_counts <- read.csv("counts2_bowtie.csv", quote = "", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
Contigs_counts  <- subset(Contigs_counts , select = -c(length))
CONTIG_spec<-c("k127_12278877","k127_1246730","k127_12842510","k127_12954","k127_13186497","k127_13257999","k127_14008563","k127_14874687","k127_3126920","k127_3443640","k127_4343132","k127_5189512","k127_5608134","k127_5988390","k127_6026611","k127_639643","k127_6473642","k127_699139","k127_8568361","k127_9807842","k127_9848036")
Contigs_counts<-subset(Contigs_counts, row.names(Contigs_counts) %in% CONTIG_spec)
CONTIG_OTU <- otu_table(Contigs_counts, taxa_are_rows = TRUE)
Contigs_tax <- Contigs_counts
Contigs_tax$Superkingdom<-"Viruses"
Contigs_tax$Realm <- row.names(Contigs_tax)
Contigs_tax <- subset(Contigs_tax, select = c(Superkingdom,Realm))
Contigs_tax$Kingdom<-Contigs_tax$Realm
Contigs_tax$Phylum <-Contigs_tax$Realm
Contigs_tax$Class<-Contigs_tax$Realm
Contigs_tax$Order<-Contigs_tax$Realm
Contigs_tax$Family<-Contigs_tax$Realm
Contigs_tax$Subfamily<-Contigs_tax$Realm
Contigs_tax$Genus<-Contigs_tax$Realm
Contigs_tax$Species<-Contigs_tax$Realm
Contigs_tax<-subset(Contigs_tax, row.names(Contigs_tax) %in% CONTIG_spec)
Contigs_tax <-as.matrix(Contigs_tax)
Contigs_tax <- tax_table(Contigs_tax)
CONTIG_PHY <- phyloseq(CONTIG_OTU, Contigs_tax, META)
```

Remove repeat samples and samples not used in this study
```{r,message=FALSE,warning=FALSE,results="hide"}
Samples_toRemove <- c("HV_001_27", "HV_002_27", "HV_003_27", "HV_004_27", "HV_005_27", "HV_006_27", "HV_007_27", "HV_008_27",
                      "HV_009_27", "HV_010_27", "HV_011_27", "HV_012_27", "HV_013_27", "HV_014_27", "HV_015_27", "HV_016_27",
                      "HV_017_27", "HV_018_27", "HV_019_27", "HV_020_27", "HV_021_27", "HV_022_27", "HV_023_27", "HV_024_27",
                      "HV_025_27", "HV_026_27", "HV_027_27", "HV_028_27", "HV_029_27", "HV_030_27", "HV_015_20", "HV_015_21", 
                      "HV_020_22", "HV_027_02", "HV_027_03", "HV_027_04", "HV_027_05", "HV_027_06", "HV_027_07", "HV_020_10",
                      "HV_021_13", "HV_027_01","HV_003_15")
CAUDO_PHY2 <- subset_samples(CAUDO_PHY, !(SampleID %in% Samples_toRemove))
CHeckV_PHY2 <- subset_samples(CHeckV_PHY, !(SampleID %in% Samples_toRemove))
PAP_PHY2 <- subset_samples(PAP_PHY, !(SampleID %in% Samples_toRemove))
BAC_PHY2 <- subset_samples(BAC_PHY, !(SampleID %in% Samples_toRemove))
GEN_PHY2 <- subset_samples(GEN_PHY, !(SampleID %in% Samples_toRemove))
CONTIG_PHY2 <- subset_samples(CONTIG_PHY, !(SampleID %in% Samples_toRemove))
```
Merge Phyloseq objects to make an overall profile (all three sets) Phyloseq object
```{r,message=FALSE,warning=FALSE,results="hide"}
Profile<- merge_phyloseq(CAUDO_PHY2,CHeckV_PHY2,PAP_PHY2,BAC_PHY2,GEN_PHY2,CONTIG_PHY2)
```

```{r}
SetA_Profile<-CHeckV_PHY2
SetA_Profile_glom <- tax_glom(SetA_Profile, "Species")
SetA_Profile_norm <- transform_sample_counts(SetA_Profile_glom, function(x) x / sum(x) )
SetB_Profile<- merge_phyloseq(CAUDO_PHY2,PAP_PHY2,BAC_PHY2,GEN_PHY2)
SetB_Profile_glom <- tax_glom(SetB_Profile, "Species")
SetB_Profile_norm <- transform_sample_counts(SetB_Profile_glom, function(x) x / sum(x) )
SetC_Profile<-CONTIG_PHY2
SetC_Profile_glom <- tax_glom(SetC_Profile, "Species")
SetC_Profile_norm <- transform_sample_counts(SetC_Profile_glom, function(x) x / sum(x) )
```
CHeckV_spec<-c("Autographa californica multiple nucleopolyhedrovirus","Autographa californica nucleopolyhedrovirus","Staphylococcus phage St 134","Unclassified Caudovirales","Unclassified Homo sapiens like virus","Unclassified Siphoviridae","Uncultured Caudovirales phage")

```{r}
SetA_Marker1 <- subset_taxa(SetA_Profile, Species=="Autographa californica multiple nucleopolyhedrovirus" )
SetA_Marker1a<-psmelt(SetA_Marker1)
SetA_Marker1a<-as.data.frame(SetA_Marker1a)
SetA_Marker1a_plot<-ggplot(SetA_Marker1a, aes(x=Time_Week,y=Abundance,color=Subject))+geom_line()+geom_point() +facet_grid(vars(Location))
SetA_Marker1a_plot


SetA_Marker2 <- subset_taxa(SetA_Profile, Species=="Autographa californica nucleopolyhedrovirus" )
SetA_Marker2a<-psmelt(SetA_Marker2)
SetA_Marker2a<-as.data.frame(SetA_Marker2a)
SetA_Marker2a_plot<-ggplot(SetA_Marker2a, aes(x=Time_Week,y=Abundance,color=Subject))+geom_line()+geom_point() +facet_grid(vars(Location))
SetA_Marker2a_plot

Profile_norm <- transform_sample_counts(Profile, function(x) x / sum(x) )
Profile_glom2 <- subset_samples(Profile_norm , Subject=="P01" )
Profile_glom2<- psmelt(Profile_glom2)
ggplot(Profile_glom2, aes(x=Time_Week,y=Abundance,color=OTU))+geom_line()+geom_point() +facet_grid(vars(Location))


```





PAREa <- t(PAREa)
PARE_gene_stats <- merge(META,PAREa, by= "row.names")
names(PARE_gene_stats)[names(PARE_gene_stats) == "k127_2557225"] <- "PARE"
```


```{r}
AMR_gene_stats <- sample_data(PPHY1)
PAREa <- otu_table(PARE)
PAREa <- t(PAREa)
PARE_gene_stats <- merge(META,PAREa, by= "row.names")
names(PARE_gene_stats)[names(PARE_gene_stats) == "k127_2557225"] <- "PARE"

MSRDa <- otu_table(MSRD)
MSRDa <- t(MSRDa)
MSRD_gene_stats <- merge(META,MSRDa, by= "row.names")
names(MSRD_gene_stats)[names(MSRD_gene_stats) == "k127_8634245"] <- "MSRD"

ERMCa <- otu_table(ERMC)
ERMCa <- t(ERMCa)
ERMC_gene_stats <- merge(META,ERMCa, by= "row.names")
names(ERMC_gene_stats)[names(ERMC_gene_stats) == "k127_12842510"] <- "ERMC"

MLS23Sa <- otu_table(MLS23S)
MLS23Sa <- t(MLS23Sa)
MLS23S_gene_stats <- merge(META,MLS23Sa, by= "row.names")
names(MLS23S_gene_stats)[names(MLS23S_gene_stats) == "k127_6535600"] <- "MLS23S"

GYRAa <- otu_table(GYRA)
GYRAa <- t(GYRAa)
GYRA_gene_stats <- merge(META,GYRAa, by= "row.names")
names(GYRA_gene_stats)[names(GYRA_gene_stats) == "k127_2646993"] <- "GYRA"

QACF_ANTIPRIME3a <- otu_table(QACF_ANTIPRIME3)
QACF_ANTIPRIME3a <- t(QACF_ANTIPRIME3a)
QACF_ANTIPRIME3_gene_stats <- merge(META,QACF_ANTIPRIME3a, by= "row.names")
names(QACF_ANTIPRIME3_gene_stats)[names(QACF_ANTIPRIME3_gene_stats) == "k127_15107270"] <- "QACF_ANTIPRIME3"

ERMC2a <- otu_table(ERMC2)
ERMC2a <- t(ERMC2a)
ERMC2_gene_stats <- merge(META,ERMC2a, by= "row.names")
names(ERMC2_gene_stats)[names(ERMC2_gene_stats) == "k127_2291"] <- "ERMC2"

```

```{r}
PARE_gene_stats_left <- PARE_gene_stats2[!grepl("scalp", PARE_gene_stats2$Location),]
PARE_gene_stats_left <- PARE_gene_stats_left[!grepl("right", PARE_gene_stats_left$Location),]
summary(ss.lm <- lm(PARE ~ Time_Week, PARE_gene_stats_left))
ssBase <- ggplot(PARE_gene_stats_left, aes(x = Time_Week, y = PARE)) + geom_point(aes(color = Subject)) + 
    scale_color_discrete() + labs(x = "Time Since Initial Sampling (week)", y = " Left Hand PARE Abundance")
ssBase + geom_smooth(method = "lm") + facet_wrap(~Subject) + 
    guides(color = FALSE)
```


